{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Patient ID 7614"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'gplots'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'reshape2'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:tidyr':\n",
      "\n",
      "    smiths\n",
      "\n",
      "\n",
      "corrplot 0.92 loaded\n",
      "\n",
      "Loading required package: plotly\n",
      "\n",
      "\n",
      "Attaching package: 'plotly'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:ggplot2':\n",
      "\n",
      "    last_plot\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "The following object is masked from 'package:graphics':\n",
      "\n",
      "    layout\n",
      "\n",
      "\n",
      "Loading required package: viridis\n",
      "\n",
      "Loading required package: viridisLite\n",
      "\n",
      "\n",
      "======================\n",
      "Welcome to heatmaply version 1.4.2\n",
      "\n",
      "Type citation('heatmaply') for how to cite the package.\n",
      "Type ?heatmaply for the main documentation.\n",
      "\n",
      "The github page is: https://github.com/talgalili/heatmaply/\n",
      "Please submit your suggestions and bug-reports at: https://github.com/talgalili/heatmaply/issues\n",
      "You may ask questions at stackoverflow, use the r and heatmaply tags: \n",
      "\t https://stackoverflow.com/questions/tagged/heatmaply\n",
      "======================\n",
      "\n",
      "\n",
      "Loading required package: usethis\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.15.3\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite either one:\n",
      "- Gu, Z. Complex Heatmap Visualization. iMeta 2022.\n",
      "- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "    genomic data. Bioinformatics 2016.\n",
      "\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments\n",
      "   in the original pheatmap() are identically supported in the new function. You \n",
      "   can still use the original function by explicitly calling pheatmap::pheatmap().\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'ComplexHeatmap'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:plotly':\n",
      "\n",
      "    add_heatmap\n",
      "\n",
      "\n",
      "The following object is masked from 'package:pheatmap':\n",
      "\n",
      "    pheatmap\n",
      "\n",
      "\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggcorrplot)\n",
    "library(gplots)\n",
    "library(reshape2)\n",
    "library(corrplot)\n",
    "library(grid)\n",
    "library(pheatmap)\n",
    "library(ggdendro)\n",
    "library(heatmaply)\n",
    "library(devtools) \n",
    "library(ComplexHeatmap) #ComplexHeatmap cite \"Gu, Z. (2022) Complex Heatmap Visualization, iMeta. DOI: 10.1002/imt2.43.\"\n",
    "library(circlize)\n",
    "library(dendsort)\n",
    "\n",
    "filterdata <- read.table(\"/Users/rahulnedunuri/Documents/GleesonLab/Mosaic Variant Projects/7614_final_summary_anno3.txt\" ,header=T,sep=\"\\t\")\n",
    "annotated <- read.csv(\"/Users/rahulnedunuri/Documents/GleesonLab/Mosaic Variant Projects/287_variant_annotation_bulk_sort_COUPTF2_TBR1_DLX1_CBH.csv\" ,header=T,sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 42</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>CHROM</th><th scope=col>POS</th><th scope=col>REF</th><th scope=col>ALT</th><th scope=col>CHROM.POS.REF.ALT</th><th scope=col>ANNO</th><th scope=col>GENE</th><th scope=col>GNOMAD_FREQ</th><th scope=col>REPEAT_MASKER</th><th scope=col>...</th><th scope=col>Punch</th><th scope=col>Cell_Type</th><th scope=col>Location</th><th scope=col>AntPost</th><th scope=col>DorVen</th><th scope=col>Organ</th><th scope=col>Note</th><th scope=col>Bulk_main</th><th scope=col>Sort_main</th><th scope=col>CTX_BG_HIP_relationship</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>...</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>7614-B-L-T</td><td>1</td><td> 4262091</td><td>T</td><td>A</td><td>1-4262091-T-A </td><td>intergenic</td><td>LINC01346(dist=249448),LINC01777(dist=210020)</td><td>6.270e-02</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>7614-B-L-T</td><td>1</td><td> 5745160</td><td>G</td><td>A</td><td>1-5745160-G-A </td><td>intergenic</td><td>MIR4417(dist=120957),MIR4689(dist=177572)    </td><td>0.000e+00</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>7614-B-L-T</td><td>1</td><td> 5786487</td><td>C</td><td>T</td><td>1-5786487-C-T </td><td>intergenic</td><td>MIR4417(dist=162284),MIR4689(dist=136245)    </td><td>6.494e-05</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>7614-B-L-T</td><td>1</td><td>12559193</td><td>G</td><td>T</td><td>1-12559193-G-T</td><td>intronic  </td><td>VPS13D                                       </td><td>0.000e+00</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>7614-B-L-T</td><td>1</td><td>19931070</td><td>C</td><td>A</td><td>1-19931070-C-A</td><td>intronic  </td><td>MINOS1,MINOS1-NBL1                           </td><td>0.000e+00</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>7614-B-L-T</td><td>1</td><td>23121461</td><td>C</td><td>G</td><td>1-23121461-C-G</td><td>intronic  </td><td>EPHB2                                        </td><td>7.944e-01</td><td>0</td><td>...</td><td>NA</td><td>Bulk</td><td>NA</td><td>Post</td><td>Ven</td><td>CTX</td><td>LARGE</td><td>1</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 42\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & ID & CHROM & POS & REF & ALT & CHROM.POS.REF.ALT & ANNO & GENE & GNOMAD\\_FREQ & REPEAT\\_MASKER & ... & Punch & Cell\\_Type & Location & AntPost & DorVen & Organ & Note & Bulk\\_main & Sort\\_main & CTX\\_BG\\_HIP\\_relationship\\\\\n",
       "  & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & ... & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 7614-B-L-T & 1 &  4262091 & T & A & 1-4262091-T-A  & intergenic & LINC01346(dist=249448),LINC01777(dist=210020) & 6.270e-02 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\t2 & 7614-B-L-T & 1 &  5745160 & G & A & 1-5745160-G-A  & intergenic & MIR4417(dist=120957),MIR4689(dist=177572)     & 0.000e+00 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\t3 & 7614-B-L-T & 1 &  5786487 & C & T & 1-5786487-C-T  & intergenic & MIR4417(dist=162284),MIR4689(dist=136245)     & 6.494e-05 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\t4 & 7614-B-L-T & 1 & 12559193 & G & T & 1-12559193-G-T & intronic   & VPS13D                                        & 0.000e+00 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\t5 & 7614-B-L-T & 1 & 19931070 & C & A & 1-19931070-C-A & intronic   & MINOS1,MINOS1-NBL1                            & 0.000e+00 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\t6 & 7614-B-L-T & 1 & 23121461 & C & G & 1-23121461-C-G & intronic   & EPHB2                                         & 7.944e-01 & 0 & ... & NA & Bulk & NA & Post & Ven & CTX & LARGE & 1 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 42\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | CHROM &lt;chr&gt; | POS &lt;int&gt; | REF &lt;chr&gt; | ALT &lt;chr&gt; | CHROM.POS.REF.ALT &lt;chr&gt; | ANNO &lt;chr&gt; | GENE &lt;chr&gt; | GNOMAD_FREQ &lt;dbl&gt; | REPEAT_MASKER &lt;dbl&gt; | ... ... | Punch &lt;int&gt; | Cell_Type &lt;chr&gt; | Location &lt;chr&gt; | AntPost &lt;chr&gt; | DorVen &lt;chr&gt; | Organ &lt;chr&gt; | Note &lt;chr&gt; | Bulk_main &lt;int&gt; | Sort_main &lt;int&gt; | CTX_BG_HIP_relationship &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 7614-B-L-T | 1 |  4262091 | T | A | 1-4262091-T-A  | intergenic | LINC01346(dist=249448),LINC01777(dist=210020) | 6.270e-02 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "| 2 | 7614-B-L-T | 1 |  5745160 | G | A | 1-5745160-G-A  | intergenic | MIR4417(dist=120957),MIR4689(dist=177572)     | 0.000e+00 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "| 3 | 7614-B-L-T | 1 |  5786487 | C | T | 1-5786487-C-T  | intergenic | MIR4417(dist=162284),MIR4689(dist=136245)     | 6.494e-05 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "| 4 | 7614-B-L-T | 1 | 12559193 | G | T | 1-12559193-G-T | intronic   | VPS13D                                        | 0.000e+00 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "| 5 | 7614-B-L-T | 1 | 19931070 | C | A | 1-19931070-C-A | intronic   | MINOS1,MINOS1-NBL1                            | 0.000e+00 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "| 6 | 7614-B-L-T | 1 | 23121461 | C | G | 1-23121461-C-G | intronic   | EPHB2                                         | 7.944e-01 | 0 | ... | NA | Bulk | NA | Post | Ven | CTX | LARGE | 1 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  ID         CHROM POS      REF ALT CHROM.POS.REF.ALT ANNO      \n",
       "1 7614-B-L-T 1      4262091 T   A   1-4262091-T-A     intergenic\n",
       "2 7614-B-L-T 1      5745160 G   A   1-5745160-G-A     intergenic\n",
       "3 7614-B-L-T 1      5786487 C   T   1-5786487-C-T     intergenic\n",
       "4 7614-B-L-T 1     12559193 G   T   1-12559193-G-T    intronic  \n",
       "5 7614-B-L-T 1     19931070 C   A   1-19931070-C-A    intronic  \n",
       "6 7614-B-L-T 1     23121461 C   G   1-23121461-C-G    intronic  \n",
       "  GENE                                          GNOMAD_FREQ REPEAT_MASKER ...\n",
       "1 LINC01346(dist=249448),LINC01777(dist=210020) 6.270e-02   0             ...\n",
       "2 MIR4417(dist=120957),MIR4689(dist=177572)     0.000e+00   0             ...\n",
       "3 MIR4417(dist=162284),MIR4689(dist=136245)     6.494e-05   0             ...\n",
       "4 VPS13D                                        0.000e+00   0             ...\n",
       "5 MINOS1,MINOS1-NBL1                            0.000e+00   0             ...\n",
       "6 EPHB2                                         7.944e-01   0             ...\n",
       "  Punch Cell_Type Location AntPost DorVen Organ Note  Bulk_main Sort_main\n",
       "1 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "2 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "3 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "4 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "5 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "6 NA    Bulk      NA       Post    Ven    CTX   LARGE 1         0        \n",
       "  CTX_BG_HIP_relationship\n",
       "1 0                      \n",
       "2 0                      \n",
       "3 0                      \n",
       "4 0                      \n",
       "5 0                      \n",
       "6 0                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 314</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>variant_list</th><th scope=col>sample_number_presented</th><th scope=col>CTX</th><th scope=col>CB</th><th scope=col>Heart</th><th scope=col>Kidney</th><th scope=col>BG</th><th scope=col>THAL</th><th scope=col>Liver</th><th scope=col>...</th><th scope=col>BG_HIP_shared_CBH</th><th scope=col>CTX_HIP_shared_CBH</th><th scope=col>CTX_only_CBH</th><th scope=col>BG_only_CBH</th><th scope=col>GP_only_CBH</th><th scope=col>HIP_only_CBH</th><th scope=col>IN_only_CBH</th><th scope=col>TBR_only_CBH</th><th scope=col>TBR_BR_CBH</th><th scope=col>DLX_BR_CBH</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>1-31691734-G-A </td><td>1-31691734-G-A </td><td> 3</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>...</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>1</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>1-45675887-A-C </td><td>1-45675887-A-C </td><td> 1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>...</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>1-64512024-C-T </td><td>1-64512024-C-T </td><td> 3</td><td>1</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>...</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>1-98426749-G-A </td><td>1-98426749-G-A </td><td> 7</td><td>1</td><td>0</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>...</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>1-99581604-C-G </td><td>1-99581604-C-G </td><td> 1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>...</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>1-106934533-C-T</td><td>1-106934533-C-T</td><td>32</td><td>1</td><td>1</td><td>0</td><td>1</td><td>1</td><td>1</td><td>0</td><td>...</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 314\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & X & variant\\_list & sample\\_number\\_presented & CTX & CB & Heart & Kidney & BG & THAL & Liver & ... & BG\\_HIP\\_shared\\_CBH & CTX\\_HIP\\_shared\\_CBH & CTX\\_only\\_CBH & BG\\_only\\_CBH & GP\\_only\\_CBH & HIP\\_only\\_CBH & IN\\_only\\_CBH & TBR\\_only\\_CBH & TBR\\_BR\\_CBH & DLX\\_BR\\_CBH\\\\\n",
       "  & <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ... & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & 1-31691734-G-A  & 1-31691734-G-A  &  3 & 1 & 0 & 0 & 0 & 0 & 1 & 0 & ... & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 1 & 1 & 0\\\\\n",
       "\t2 & 1-45675887-A-C  & 1-45675887-A-C  &  1 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & ... & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1\\\\\n",
       "\t3 & 1-64512024-C-T  & 1-64512024-C-T  &  3 & 1 & 0 & 0 & 0 & 1 & 0 & 0 & ... & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1\\\\\n",
       "\t4 & 1-98426749-G-A  & 1-98426749-G-A  &  7 & 1 & 0 & 0 & 0 & 1 & 0 & 0 & ... & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1\\\\\n",
       "\t5 & 1-99581604-C-G  & 1-99581604-C-G  &  1 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & ... & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t6 & 1-106934533-C-T & 1-106934533-C-T & 32 & 1 & 1 & 0 & 1 & 1 & 1 & 0 & ... & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 314\n",
       "\n",
       "| <!--/--> | X &lt;chr&gt; | variant_list &lt;chr&gt; | sample_number_presented &lt;int&gt; | CTX &lt;int&gt; | CB &lt;int&gt; | Heart &lt;int&gt; | Kidney &lt;int&gt; | BG &lt;int&gt; | THAL &lt;int&gt; | Liver &lt;int&gt; | ... ... | BG_HIP_shared_CBH &lt;int&gt; | CTX_HIP_shared_CBH &lt;int&gt; | CTX_only_CBH &lt;int&gt; | BG_only_CBH &lt;int&gt; | GP_only_CBH &lt;int&gt; | HIP_only_CBH &lt;int&gt; | IN_only_CBH &lt;int&gt; | TBR_only_CBH &lt;int&gt; | TBR_BR_CBH &lt;int&gt; | DLX_BR_CBH &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 1-31691734-G-A  | 1-31691734-G-A  |  3 | 1 | 0 | 0 | 0 | 0 | 1 | 0 | ... | 0 | 0 | 0 | 0 | 0 | 1 | 0 | 1 | 1 | 0 |\n",
       "| 2 | 1-45675887-A-C  | 1-45675887-A-C  |  1 | 0 | 0 | 0 | 0 | 1 | 0 | 0 | ... | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 |\n",
       "| 3 | 1-64512024-C-T  | 1-64512024-C-T  |  3 | 1 | 0 | 0 | 0 | 1 | 0 | 0 | ... | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 |\n",
       "| 4 | 1-98426749-G-A  | 1-98426749-G-A  |  7 | 1 | 0 | 0 | 0 | 1 | 0 | 0 | ... | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 |\n",
       "| 5 | 1-99581604-C-G  | 1-99581604-C-G  |  1 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | ... | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 6 | 1-106934533-C-T | 1-106934533-C-T | 32 | 1 | 1 | 0 | 1 | 1 | 1 | 0 | ... | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 1 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  X               variant_list    sample_number_presented CTX CB Heart Kidney\n",
       "1 1-31691734-G-A  1-31691734-G-A   3                      1   0  0     0     \n",
       "2 1-45675887-A-C  1-45675887-A-C   1                      0   0  0     0     \n",
       "3 1-64512024-C-T  1-64512024-C-T   3                      1   0  0     0     \n",
       "4 1-98426749-G-A  1-98426749-G-A   7                      1   0  0     0     \n",
       "5 1-99581604-C-G  1-99581604-C-G   1                      1   0  0     0     \n",
       "6 1-106934533-C-T 1-106934533-C-T 32                      1   1  0     1     \n",
       "  BG THAL Liver ... BG_HIP_shared_CBH CTX_HIP_shared_CBH CTX_only_CBH\n",
       "1 0  1    0     ... 0                 0                  0           \n",
       "2 1  0    0     ... 0                 0                  0           \n",
       "3 1  0    0     ... 1                 0                  0           \n",
       "4 1  0    0     ... 0                 0                  0           \n",
       "5 0  0    0     ... 0                 0                  0           \n",
       "6 1  1    0     ... 0                 0                  0           \n",
       "  BG_only_CBH GP_only_CBH HIP_only_CBH IN_only_CBH TBR_only_CBH TBR_BR_CBH\n",
       "1 0           0           1            0           1            1         \n",
       "2 0           0           0            0           0            1         \n",
       "3 0           0           0            0           0            1         \n",
       "4 0           0           0            0           0            1         \n",
       "5 0           0           0            0           0            0         \n",
       "6 0           0           0            0           0            1         \n",
       "  DLX_BR_CBH\n",
       "1 0         \n",
       "2 1         \n",
       "3 1         \n",
       "4 1         \n",
       "5 0         \n",
       "6 1         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ID'</li><li>'CHROM'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'CHROM.POS.REF.ALT'</li><li>'ANNO'</li><li>'GENE'</li><li>'GNOMAD_FREQ'</li><li>'REPEAT_MASKER'</li><li>'SEGDUP'</li><li>'HOMOPOLYMER'</li><li>'REF_SEQ'</li><li>'DINUCLEOTIDE'</li><li>'NEAR_INDEL'</li><li>'UCSC_RPMSK'</li><li>'REF_COUNT'</li><li>'ALT_COUNT'</li><li>'MAF'</li><li>'LOWER_CI'</li><li>'UPPER_CI'</li><li>'CI_IS_GREATER'</li><li>'NORMAL_REF_COUNT'</li><li>'NORMAL_ALT_COUNT'</li><li>'NORMAL_MAF'</li><li>'NORMAL_LOWER_CI'</li><li>'NORMAL_UPPER_CI'</li><li>'NORMAL_CI_IS_GREATER'</li><li>'Mosaic'</li><li>'Mosaic_shared_btw_WGS_Amp'</li><li>'Hemisphere'</li><li>'Region'</li><li>'Punch'</li><li>'Cell_Type'</li><li>'Location'</li><li>'AntPost'</li><li>'DorVen'</li><li>'Organ'</li><li>'Note'</li><li>'Bulk_main'</li><li>'Sort_main'</li><li>'CTX_BG_HIP_relationship'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ID'\n",
       "\\item 'CHROM'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'CHROM.POS.REF.ALT'\n",
       "\\item 'ANNO'\n",
       "\\item 'GENE'\n",
       "\\item 'GNOMAD\\_FREQ'\n",
       "\\item 'REPEAT\\_MASKER'\n",
       "\\item 'SEGDUP'\n",
       "\\item 'HOMOPOLYMER'\n",
       "\\item 'REF\\_SEQ'\n",
       "\\item 'DINUCLEOTIDE'\n",
       "\\item 'NEAR\\_INDEL'\n",
       "\\item 'UCSC\\_RPMSK'\n",
       "\\item 'REF\\_COUNT'\n",
       "\\item 'ALT\\_COUNT'\n",
       "\\item 'MAF'\n",
       "\\item 'LOWER\\_CI'\n",
       "\\item 'UPPER\\_CI'\n",
       "\\item 'CI\\_IS\\_GREATER'\n",
       "\\item 'NORMAL\\_REF\\_COUNT'\n",
       "\\item 'NORMAL\\_ALT\\_COUNT'\n",
       "\\item 'NORMAL\\_MAF'\n",
       "\\item 'NORMAL\\_LOWER\\_CI'\n",
       "\\item 'NORMAL\\_UPPER\\_CI'\n",
       "\\item 'NORMAL\\_CI\\_IS\\_GREATER'\n",
       "\\item 'Mosaic'\n",
       "\\item 'Mosaic\\_shared\\_btw\\_WGS\\_Amp'\n",
       "\\item 'Hemisphere'\n",
       "\\item 'Region'\n",
       "\\item 'Punch'\n",
       "\\item 'Cell\\_Type'\n",
       "\\item 'Location'\n",
       "\\item 'AntPost'\n",
       "\\item 'DorVen'\n",
       "\\item 'Organ'\n",
       "\\item 'Note'\n",
       "\\item 'Bulk\\_main'\n",
       "\\item 'Sort\\_main'\n",
       "\\item 'CTX\\_BG\\_HIP\\_relationship'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ID'\n",
       "2. 'CHROM'\n",
       "3. 'POS'\n",
       "4. 'REF'\n",
       "5. 'ALT'\n",
       "6. 'CHROM.POS.REF.ALT'\n",
       "7. 'ANNO'\n",
       "8. 'GENE'\n",
       "9. 'GNOMAD_FREQ'\n",
       "10. 'REPEAT_MASKER'\n",
       "11. 'SEGDUP'\n",
       "12. 'HOMOPOLYMER'\n",
       "13. 'REF_SEQ'\n",
       "14. 'DINUCLEOTIDE'\n",
       "15. 'NEAR_INDEL'\n",
       "16. 'UCSC_RPMSK'\n",
       "17. 'REF_COUNT'\n",
       "18. 'ALT_COUNT'\n",
       "19. 'MAF'\n",
       "20. 'LOWER_CI'\n",
       "21. 'UPPER_CI'\n",
       "22. 'CI_IS_GREATER'\n",
       "23. 'NORMAL_REF_COUNT'\n",
       "24. 'NORMAL_ALT_COUNT'\n",
       "25. 'NORMAL_MAF'\n",
       "26. 'NORMAL_LOWER_CI'\n",
       "27. 'NORMAL_UPPER_CI'\n",
       "28. 'NORMAL_CI_IS_GREATER'\n",
       "29. 'Mosaic'\n",
       "30. 'Mosaic_shared_btw_WGS_Amp'\n",
       "31. 'Hemisphere'\n",
       "32. 'Region'\n",
       "33. 'Punch'\n",
       "34. 'Cell_Type'\n",
       "35. 'Location'\n",
       "36. 'AntPost'\n",
       "37. 'DorVen'\n",
       "38. 'Organ'\n",
       "39. 'Note'\n",
       "40. 'Bulk_main'\n",
       "41. 'Sort_main'\n",
       "42. 'CTX_BG_HIP_relationship'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ID\"                        \"CHROM\"                    \n",
       " [3] \"POS\"                       \"REF\"                      \n",
       " [5] \"ALT\"                       \"CHROM.POS.REF.ALT\"        \n",
       " [7] \"ANNO\"                      \"GENE\"                     \n",
       " [9] \"GNOMAD_FREQ\"               \"REPEAT_MASKER\"            \n",
       "[11] \"SEGDUP\"                    \"HOMOPOLYMER\"              \n",
       "[13] \"REF_SEQ\"                   \"DINUCLEOTIDE\"             \n",
       "[15] \"NEAR_INDEL\"                \"UCSC_RPMSK\"               \n",
       "[17] \"REF_COUNT\"                 \"ALT_COUNT\"                \n",
       "[19] \"MAF\"                       \"LOWER_CI\"                 \n",
       "[21] \"UPPER_CI\"                  \"CI_IS_GREATER\"            \n",
       "[23] \"NORMAL_REF_COUNT\"          \"NORMAL_ALT_COUNT\"         \n",
       "[25] \"NORMAL_MAF\"                \"NORMAL_LOWER_CI\"          \n",
       "[27] \"NORMAL_UPPER_CI\"           \"NORMAL_CI_IS_GREATER\"     \n",
       "[29] \"Mosaic\"                    \"Mosaic_shared_btw_WGS_Amp\"\n",
       "[31] \"Hemisphere\"                \"Region\"                   \n",
       "[33] \"Punch\"                     \"Cell_Type\"                \n",
       "[35] \"Location\"                  \"AntPost\"                  \n",
       "[37] \"DorVen\"                    \"Organ\"                    \n",
       "[39] \"Note\"                      \"Bulk_main\"                \n",
       "[41] \"Sort_main\"                 \"CTX_BG_HIP_relationship\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X'</li><li>'variant_list'</li><li>'sample_number_presented'</li><li>'CTX'</li><li>'CB'</li><li>'Heart'</li><li>'Kidney'</li><li>'BG'</li><li>'THAL'</li><li>'Liver'</li><li>'L'</li><li>'R'</li><li>'PF'</li><li>'F'</li><li>'P'</li><li>'O'</li><li>'T'</li><li>'Ins'</li><li>'Cau'</li><li>'Put'</li><li>'GP'</li><li>'Bulk'</li><li>'DLX1'</li><li>'TBR1'</li><li>'COUPTF2'</li><li>'NeuN'</li><li>'LHX2'</li><li>'Olig2'</li><li>'PU1'</li><li>'DARPP32M.NeuNP'</li><li>'DARPP32P.NeuNP'</li><li>'Cardiomyocyte'</li><li>'Brain_only'</li><li>'CTX_only'</li><li>'CTX_BG_only'</li><li>'BG_only'</li><li>'THAL_only'</li><li>'Left_only'</li><li>'Right_only'</li><li>'L_BR_PF_only'</li><li>'L_BR_F_only'</li><li>'L_BR_P_only'</li><li>'L_BR_O_only'</li><li>'L_BR_Ins_only'</li><li>'R_BR_PF_only'</li><li>'R_BR_F_only'</li><li>'R_BR_P_only'</li><li>'R_BR_O_only'</li><li>'R_BR_Ins_only'</li><li>'IN_only'</li><li>'COUP_only'</li><li>'TBR_only'</li><li>'TBR_BR'</li><li>'DLX_BR'</li><li>'COUP_BR'</li><li>'MSN_BR'</li><li>'nonMSN_BR'</li><li>'sample_number_presented_sort'</li><li>'CTX_sort'</li><li>'CB_sort'</li><li>'Heart_sort'</li><li>'Kidney_sort'</li><li>'BG_sort'</li><li>'THAL_sort'</li><li>'Liver_sort'</li><li>'L_sort'</li><li>'R_sort'</li><li>'PF_sort'</li><li>'F_sort'</li><li>'P_sort'</li><li>'O_sort'</li><li>'T_sort'</li><li>'Ins_sort'</li><li>'Cau_sort'</li><li>'Put_sort'</li><li>'GP_sort'</li><li>'Bulk_sort'</li><li>'DLX1_sort'</li><li>'TBR1_sort'</li><li>'COUPTF2_sort'</li><li>'NeuN_sort'</li><li>'LHX2_sort'</li><li>'Olig2_sort'</li><li>'PU1_sort'</li><li>'DARPP32M.NeuNP_sort'</li><li>'DARPP32P.NeuNP_sort'</li><li>'Cardiomyocyte_sort'</li><li>'Brain_only_sort'</li><li>'CTX_only_sort'</li><li>'CTX_BG_only_sort'</li><li>'BG_only_sort'</li><li>'THAL_only_sort'</li><li>'Left_only_sort'</li><li>'Right_only_sort'</li><li>'L_BR_PF_only_sort'</li><li>'L_BR_F_only_sort'</li><li>'L_BR_P_only_sort'</li><li>'L_BR_O_only_sort'</li><li>'L_BR_Ins_only_sort'</li><li>'R_BR_PF_only_sort'</li><li>'R_BR_F_only_sort'</li><li>'R_BR_P_only_sort'</li><li>'R_BR_O_only_sort'</li><li>'R_BR_Ins_only_sort'</li><li>'IN_only_sort'</li><li>'COUP_only_sort'</li><li>'TBR_only_sort'</li><li>'TBR_BR_sort'</li><li>'DLX_BR_sort'</li><li>'COUP_BR_sort'</li><li>'MSN_BR_sort'</li><li>'nonMSN_BR_sort'</li><li>'COUPTF2_enriched_sort'</li><li>'sample_number_presented_COUPTF2'</li><li>'CTX_COUPTF2'</li><li>'CB_COUPTF2'</li><li>'Heart_COUPTF2'</li><li>'Kidney_COUPTF2'</li><li>'BG_COUPTF2'</li><li>'THAL_COUPTF2'</li><li>'Liver_COUPTF2'</li><li>'L_COUPTF2'</li><li>'R_COUPTF2'</li><li>'PF_COUPTF2'</li><li>'F_COUPTF2'</li><li>'P_COUPTF2'</li><li>'O_COUPTF2'</li><li>'T_COUPTF2'</li><li>'Ins_COUPTF2'</li><li>'Cau_COUPTF2'</li><li>'Put_COUPTF2'</li><li>'GP_COUPTF2'</li><li>'Bulk_COUPTF2'</li><li>'DLX1_COUPTF2'</li><li>'TBR1_COUPTF2'</li><li>'COUPTF2_COUPTF2'</li><li>'NeuN_COUPTF2'</li><li>'LHX2_COUPTF2'</li><li>'Olig2_COUPTF2'</li><li>'PU1_COUPTF2'</li><li>'DARPP32M.NeuNP_COUPTF2'</li><li>'DARPP32P.NeuNP_COUPTF2'</li><li>'Cardiomyocyte_COUPTF2'</li><li>'Brain_only_COUPTF2'</li><li>'CTX_only_COUPTF2'</li><li>'CTX_BG_only_COUPTF2'</li><li>'BG_only_COUPTF2'</li><li>'THAL_only_COUPTF2'</li><li>'Left_only_COUPTF2'</li><li>'Right_only_COUPTF2'</li><li>'L_BR_PF_only_COUPTF2'</li><li>'L_BR_F_only_COUPTF2'</li><li>'L_BR_P_only_COUPTF2'</li><li>'L_BR_O_only_COUPTF2'</li><li>'L_BR_Ins_only_COUPTF2'</li><li>'R_BR_PF_only_COUPTF2'</li><li>'R_BR_F_only_COUPTF2'</li><li>'R_BR_P_only_COUPTF2'</li><li>'R_BR_O_only_COUPTF2'</li><li>'R_BR_Ins_only_COUPTF2'</li><li>'IN_only_COUPTF2'</li><li>'COUP_only_COUPTF2'</li><li>'TBR_only_COUPTF2'</li><li>'TBR_BR_COUPTF2'</li><li>'DLX_BR_COUPTF2'</li><li>'COUP_BR_COUPTF2'</li><li>'MSN_BR_COUPTF2'</li><li>'nonMSN_BR_COUPTF2'</li><li>'COUPTF2_enriched_COUPTF2'</li><li>'sample_number_presented_TBR1'</li><li>'CTX_TBR1'</li><li>'CB_TBR1'</li><li>'Heart_TBR1'</li><li>'Kidney_TBR1'</li><li>'BG_TBR1'</li><li>'THAL_TBR1'</li><li>'Liver_TBR1'</li><li>'L_TBR1'</li><li>'R_TBR1'</li><li>'PF_TBR1'</li><li>'F_TBR1'</li><li>'P_TBR1'</li><li>'O_TBR1'</li><li>'T_TBR1'</li><li>'Ins_TBR1'</li><li>'Cau_TBR1'</li><li>'Put_TBR1'</li><li>'GP_TBR1'</li><li>'Bulk_TBR1'</li><li>'DLX1_TBR1'</li><li>'TBR1_TBR1'</li><li>'COUPTF2_TBR1'</li><li>'NeuN_TBR1'</li><li>'LHX2_TBR1'</li><li>'Olig2_TBR1'</li><li>'PU1_TBR1'</li><li>'DARPP32M.NeuNP_TBR1'</li><li>'DARPP32P.NeuNP_TBR1'</li><li>'Cardiomyocyte_TBR1'</li><li>'Brain_only_TBR1'</li><li>'CTX_only_TBR1'</li><li>'CTX_BG_only_TBR1'</li><li>'BG_only_TBR1'</li><li>'THAL_only_TBR1'</li><li>'Left_only_TBR1'</li><li>'Right_only_TBR1'</li><li>'L_BR_PF_only_TBR1'</li><li>'L_BR_F_only_TBR1'</li><li>'L_BR_P_only_TBR1'</li><li>'L_BR_O_only_TBR1'</li><li>'L_BR_Ins_only_TBR1'</li><li>'R_BR_PF_only_TBR1'</li><li>'R_BR_F_only_TBR1'</li><li>'R_BR_P_only_TBR1'</li><li>'R_BR_O_only_TBR1'</li><li>'R_BR_Ins_only_TBR1'</li><li>'IN_only_TBR1'</li><li>'COUP_only_TBR1'</li><li>'TBR_only_TBR1'</li><li>'TBR_BR_TBR1'</li><li>'DLX_BR_TBR1'</li><li>'COUP_BR_TBR1'</li><li>'MSN_BR_TBR1'</li><li>'nonMSN_BR_TBR1'</li><li>'sample_number_presented_DLX1'</li><li>'CTX_DLX1'</li><li>'CB_DLX1'</li><li>'Heart_DLX1'</li><li>'Kidney_DLX1'</li><li>'BG_DLX1'</li><li>'THAL_DLX1'</li><li>'Liver_DLX1'</li><li>'L_DLX1'</li><li>'R_DLX1'</li><li>'PF_DLX1'</li><li>'F_DLX1'</li><li>'P_DLX1'</li><li>'O_DLX1'</li><li>'T_DLX1'</li><li>'Ins_DLX1'</li><li>'Cau_DLX1'</li><li>'Put_DLX1'</li><li>'GP_DLX1'</li><li>'Bulk_DLX1'</li><li>'DLX1_DLX1'</li><li>'TBR1_DLX1'</li><li>'COUPTF2_DLX1'</li><li>'NeuN_DLX1'</li><li>'LHX2_DLX1'</li><li>'Olig2_DLX1'</li><li>'PU1_DLX1'</li><li>'DARPP32M.NeuNP_DLX1'</li><li>'DARPP32P.NeuNP_DLX1'</li><li>'Cardiomyocyte_DLX1'</li><li>'Brain_only_DLX1'</li><li>'CTX_only_DLX1'</li><li>'CTX_BG_only_DLX1'</li><li>'BG_only_DLX1'</li><li>'THAL_only_DLX1'</li><li>'Left_only_DLX1'</li><li>'Right_only_DLX1'</li><li>'L_BR_PF_only_DLX1'</li><li>'L_BR_F_only_DLX1'</li><li>'L_BR_P_only_DLX1'</li><li>'L_BR_O_only_DLX1'</li><li>'L_BR_Ins_only_DLX1'</li><li>'R_BR_PF_only_DLX1'</li><li>'R_BR_F_only_DLX1'</li><li>'R_BR_P_only_DLX1'</li><li>'R_BR_O_only_DLX1'</li><li>'R_BR_Ins_only_DLX1'</li><li>'IN_only_DLX1'</li><li>'COUP_only_DLX1'</li><li>'TBR_only_DLX1'</li><li>'TBR_BR_DLX1'</li><li>'DLX_BR_DLX1'</li><li>'COUP_BR_DLX1'</li><li>'MSN_BR_DLX1'</li><li>'nonMSN_BR_DLX1'</li><li>'sample_number_presented_CBH'</li><li>'CTX_CBH'</li><li>'CB_CBH'</li><li>'Heart_CBH'</li><li>'Kidney_CBH'</li><li>'BG_CBH'</li><li>'THAL_CBH'</li><li>'HIP_CBH'</li><li>'Liver_CBH'</li><li>'PF_CBH'</li><li>'F_CBH'</li><li>'P_CBH'</li><li>'O_CBH'</li><li>'T_CBH'</li><li>'Ins_CBH'</li><li>'Cau_CBH'</li><li>'Put_CBH'</li><li>'GP_CBH'</li><li>'CA1_CBH'</li><li>'CA3_CBH'</li><li>'DG_CBH'</li><li>'DLX1_CBH'</li><li>'TBR1_CBH'</li><li>'Brain_only_CBH'</li><li>'CTX_BG_shared_CBH'</li><li>'BG_HIP_shared_CBH'</li><li>'CTX_HIP_shared_CBH'</li><li>'CTX_only_CBH'</li><li>'BG_only_CBH'</li><li>'GP_only_CBH'</li><li>'HIP_only_CBH'</li><li>'IN_only_CBH'</li><li>'TBR_only_CBH'</li><li>'TBR_BR_CBH'</li><li>'DLX_BR_CBH'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X'\n",
       "\\item 'variant\\_list'\n",
       "\\item 'sample\\_number\\_presented'\n",
       "\\item 'CTX'\n",
       "\\item 'CB'\n",
       "\\item 'Heart'\n",
       "\\item 'Kidney'\n",
       "\\item 'BG'\n",
       "\\item 'THAL'\n",
       "\\item 'Liver'\n",
       "\\item 'L'\n",
       "\\item 'R'\n",
       "\\item 'PF'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'O'\n",
       "\\item 'T'\n",
       "\\item 'Ins'\n",
       "\\item 'Cau'\n",
       "\\item 'Put'\n",
       "\\item 'GP'\n",
       "\\item 'Bulk'\n",
       "\\item 'DLX1'\n",
       "\\item 'TBR1'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'NeuN'\n",
       "\\item 'LHX2'\n",
       "\\item 'Olig2'\n",
       "\\item 'PU1'\n",
       "\\item 'DARPP32M.NeuNP'\n",
       "\\item 'DARPP32P.NeuNP'\n",
       "\\item 'Cardiomyocyte'\n",
       "\\item 'Brain\\_only'\n",
       "\\item 'CTX\\_only'\n",
       "\\item 'CTX\\_BG\\_only'\n",
       "\\item 'BG\\_only'\n",
       "\\item 'THAL\\_only'\n",
       "\\item 'Left\\_only'\n",
       "\\item 'Right\\_only'\n",
       "\\item 'L\\_BR\\_PF\\_only'\n",
       "\\item 'L\\_BR\\_F\\_only'\n",
       "\\item 'L\\_BR\\_P\\_only'\n",
       "\\item 'L\\_BR\\_O\\_only'\n",
       "\\item 'L\\_BR\\_Ins\\_only'\n",
       "\\item 'R\\_BR\\_PF\\_only'\n",
       "\\item 'R\\_BR\\_F\\_only'\n",
       "\\item 'R\\_BR\\_P\\_only'\n",
       "\\item 'R\\_BR\\_O\\_only'\n",
       "\\item 'R\\_BR\\_Ins\\_only'\n",
       "\\item 'IN\\_only'\n",
       "\\item 'COUP\\_only'\n",
       "\\item 'TBR\\_only'\n",
       "\\item 'TBR\\_BR'\n",
       "\\item 'DLX\\_BR'\n",
       "\\item 'COUP\\_BR'\n",
       "\\item 'MSN\\_BR'\n",
       "\\item 'nonMSN\\_BR'\n",
       "\\item 'sample\\_number\\_presented\\_sort'\n",
       "\\item 'CTX\\_sort'\n",
       "\\item 'CB\\_sort'\n",
       "\\item 'Heart\\_sort'\n",
       "\\item 'Kidney\\_sort'\n",
       "\\item 'BG\\_sort'\n",
       "\\item 'THAL\\_sort'\n",
       "\\item 'Liver\\_sort'\n",
       "\\item 'L\\_sort'\n",
       "\\item 'R\\_sort'\n",
       "\\item 'PF\\_sort'\n",
       "\\item 'F\\_sort'\n",
       "\\item 'P\\_sort'\n",
       "\\item 'O\\_sort'\n",
       "\\item 'T\\_sort'\n",
       "\\item 'Ins\\_sort'\n",
       "\\item 'Cau\\_sort'\n",
       "\\item 'Put\\_sort'\n",
       "\\item 'GP\\_sort'\n",
       "\\item 'Bulk\\_sort'\n",
       "\\item 'DLX1\\_sort'\n",
       "\\item 'TBR1\\_sort'\n",
       "\\item 'COUPTF2\\_sort'\n",
       "\\item 'NeuN\\_sort'\n",
       "\\item 'LHX2\\_sort'\n",
       "\\item 'Olig2\\_sort'\n",
       "\\item 'PU1\\_sort'\n",
       "\\item 'DARPP32M.NeuNP\\_sort'\n",
       "\\item 'DARPP32P.NeuNP\\_sort'\n",
       "\\item 'Cardiomyocyte\\_sort'\n",
       "\\item 'Brain\\_only\\_sort'\n",
       "\\item 'CTX\\_only\\_sort'\n",
       "\\item 'CTX\\_BG\\_only\\_sort'\n",
       "\\item 'BG\\_only\\_sort'\n",
       "\\item 'THAL\\_only\\_sort'\n",
       "\\item 'Left\\_only\\_sort'\n",
       "\\item 'Right\\_only\\_sort'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_sort'\n",
       "\\item 'L\\_BR\\_F\\_only\\_sort'\n",
       "\\item 'L\\_BR\\_P\\_only\\_sort'\n",
       "\\item 'L\\_BR\\_O\\_only\\_sort'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_sort'\n",
       "\\item 'R\\_BR\\_PF\\_only\\_sort'\n",
       "\\item 'R\\_BR\\_F\\_only\\_sort'\n",
       "\\item 'R\\_BR\\_P\\_only\\_sort'\n",
       "\\item 'R\\_BR\\_O\\_only\\_sort'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_sort'\n",
       "\\item 'IN\\_only\\_sort'\n",
       "\\item 'COUP\\_only\\_sort'\n",
       "\\item 'TBR\\_only\\_sort'\n",
       "\\item 'TBR\\_BR\\_sort'\n",
       "\\item 'DLX\\_BR\\_sort'\n",
       "\\item 'COUP\\_BR\\_sort'\n",
       "\\item 'MSN\\_BR\\_sort'\n",
       "\\item 'nonMSN\\_BR\\_sort'\n",
       "\\item 'COUPTF2\\_enriched\\_sort'\n",
       "\\item 'sample\\_number\\_presented\\_COUPTF2'\n",
       "\\item 'CTX\\_COUPTF2'\n",
       "\\item 'CB\\_COUPTF2'\n",
       "\\item 'Heart\\_COUPTF2'\n",
       "\\item 'Kidney\\_COUPTF2'\n",
       "\\item 'BG\\_COUPTF2'\n",
       "\\item 'THAL\\_COUPTF2'\n",
       "\\item 'Liver\\_COUPTF2'\n",
       "\\item 'L\\_COUPTF2'\n",
       "\\item 'R\\_COUPTF2'\n",
       "\\item 'PF\\_COUPTF2'\n",
       "\\item 'F\\_COUPTF2'\n",
       "\\item 'P\\_COUPTF2'\n",
       "\\item 'O\\_COUPTF2'\n",
       "\\item 'T\\_COUPTF2'\n",
       "\\item 'Ins\\_COUPTF2'\n",
       "\\item 'Cau\\_COUPTF2'\n",
       "\\item 'Put\\_COUPTF2'\n",
       "\\item 'GP\\_COUPTF2'\n",
       "\\item 'Bulk\\_COUPTF2'\n",
       "\\item 'DLX1\\_COUPTF2'\n",
       "\\item 'TBR1\\_COUPTF2'\n",
       "\\item 'COUPTF2\\_COUPTF2'\n",
       "\\item 'NeuN\\_COUPTF2'\n",
       "\\item 'LHX2\\_COUPTF2'\n",
       "\\item 'Olig2\\_COUPTF2'\n",
       "\\item 'PU1\\_COUPTF2'\n",
       "\\item 'DARPP32M.NeuNP\\_COUPTF2'\n",
       "\\item 'DARPP32P.NeuNP\\_COUPTF2'\n",
       "\\item 'Cardiomyocyte\\_COUPTF2'\n",
       "\\item 'Brain\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_BG\\_only\\_COUPTF2'\n",
       "\\item 'BG\\_only\\_COUPTF2'\n",
       "\\item 'THAL\\_only\\_COUPTF2'\n",
       "\\item 'Left\\_only\\_COUPTF2'\n",
       "\\item 'Right\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_F\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_P\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_O\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_PF\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_F\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_P\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_O\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_COUPTF2'\n",
       "\\item 'IN\\_only\\_COUPTF2'\n",
       "\\item 'COUP\\_only\\_COUPTF2'\n",
       "\\item 'TBR\\_only\\_COUPTF2'\n",
       "\\item 'TBR\\_BR\\_COUPTF2'\n",
       "\\item 'DLX\\_BR\\_COUPTF2'\n",
       "\\item 'COUP\\_BR\\_COUPTF2'\n",
       "\\item 'MSN\\_BR\\_COUPTF2'\n",
       "\\item 'nonMSN\\_BR\\_COUPTF2'\n",
       "\\item 'COUPTF2\\_enriched\\_COUPTF2'\n",
       "\\item 'sample\\_number\\_presented\\_TBR1'\n",
       "\\item 'CTX\\_TBR1'\n",
       "\\item 'CB\\_TBR1'\n",
       "\\item 'Heart\\_TBR1'\n",
       "\\item 'Kidney\\_TBR1'\n",
       "\\item 'BG\\_TBR1'\n",
       "\\item 'THAL\\_TBR1'\n",
       "\\item 'Liver\\_TBR1'\n",
       "\\item 'L\\_TBR1'\n",
       "\\item 'R\\_TBR1'\n",
       "\\item 'PF\\_TBR1'\n",
       "\\item 'F\\_TBR1'\n",
       "\\item 'P\\_TBR1'\n",
       "\\item 'O\\_TBR1'\n",
       "\\item 'T\\_TBR1'\n",
       "\\item 'Ins\\_TBR1'\n",
       "\\item 'Cau\\_TBR1'\n",
       "\\item 'Put\\_TBR1'\n",
       "\\item 'GP\\_TBR1'\n",
       "\\item 'Bulk\\_TBR1'\n",
       "\\item 'DLX1\\_TBR1'\n",
       "\\item 'TBR1\\_TBR1'\n",
       "\\item 'COUPTF2\\_TBR1'\n",
       "\\item 'NeuN\\_TBR1'\n",
       "\\item 'LHX2\\_TBR1'\n",
       "\\item 'Olig2\\_TBR1'\n",
       "\\item 'PU1\\_TBR1'\n",
       "\\item 'DARPP32M.NeuNP\\_TBR1'\n",
       "\\item 'DARPP32P.NeuNP\\_TBR1'\n",
       "\\item 'Cardiomyocyte\\_TBR1'\n",
       "\\item 'Brain\\_only\\_TBR1'\n",
       "\\item 'CTX\\_only\\_TBR1'\n",
       "\\item 'CTX\\_BG\\_only\\_TBR1'\n",
       "\\item 'BG\\_only\\_TBR1'\n",
       "\\item 'THAL\\_only\\_TBR1'\n",
       "\\item 'Left\\_only\\_TBR1'\n",
       "\\item 'Right\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_F\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_P\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_O\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_PF\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_F\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_P\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_O\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_TBR1'\n",
       "\\item 'IN\\_only\\_TBR1'\n",
       "\\item 'COUP\\_only\\_TBR1'\n",
       "\\item 'TBR\\_only\\_TBR1'\n",
       "\\item 'TBR\\_BR\\_TBR1'\n",
       "\\item 'DLX\\_BR\\_TBR1'\n",
       "\\item 'COUP\\_BR\\_TBR1'\n",
       "\\item 'MSN\\_BR\\_TBR1'\n",
       "\\item 'nonMSN\\_BR\\_TBR1'\n",
       "\\item 'sample\\_number\\_presented\\_DLX1'\n",
       "\\item 'CTX\\_DLX1'\n",
       "\\item 'CB\\_DLX1'\n",
       "\\item 'Heart\\_DLX1'\n",
       "\\item 'Kidney\\_DLX1'\n",
       "\\item 'BG\\_DLX1'\n",
       "\\item 'THAL\\_DLX1'\n",
       "\\item 'Liver\\_DLX1'\n",
       "\\item 'L\\_DLX1'\n",
       "\\item 'R\\_DLX1'\n",
       "\\item 'PF\\_DLX1'\n",
       "\\item 'F\\_DLX1'\n",
       "\\item 'P\\_DLX1'\n",
       "\\item 'O\\_DLX1'\n",
       "\\item 'T\\_DLX1'\n",
       "\\item 'Ins\\_DLX1'\n",
       "\\item 'Cau\\_DLX1'\n",
       "\\item 'Put\\_DLX1'\n",
       "\\item 'GP\\_DLX1'\n",
       "\\item 'Bulk\\_DLX1'\n",
       "\\item 'DLX1\\_DLX1'\n",
       "\\item 'TBR1\\_DLX1'\n",
       "\\item 'COUPTF2\\_DLX1'\n",
       "\\item 'NeuN\\_DLX1'\n",
       "\\item 'LHX2\\_DLX1'\n",
       "\\item 'Olig2\\_DLX1'\n",
       "\\item 'PU1\\_DLX1'\n",
       "\\item 'DARPP32M.NeuNP\\_DLX1'\n",
       "\\item 'DARPP32P.NeuNP\\_DLX1'\n",
       "\\item 'Cardiomyocyte\\_DLX1'\n",
       "\\item 'Brain\\_only\\_DLX1'\n",
       "\\item 'CTX\\_only\\_DLX1'\n",
       "\\item 'CTX\\_BG\\_only\\_DLX1'\n",
       "\\item 'BG\\_only\\_DLX1'\n",
       "\\item 'THAL\\_only\\_DLX1'\n",
       "\\item 'Left\\_only\\_DLX1'\n",
       "\\item 'Right\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_F\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_P\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_O\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_PF\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_F\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_P\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_O\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_DLX1'\n",
       "\\item 'IN\\_only\\_DLX1'\n",
       "\\item 'COUP\\_only\\_DLX1'\n",
       "\\item 'TBR\\_only\\_DLX1'\n",
       "\\item 'TBR\\_BR\\_DLX1'\n",
       "\\item 'DLX\\_BR\\_DLX1'\n",
       "\\item 'COUP\\_BR\\_DLX1'\n",
       "\\item 'MSN\\_BR\\_DLX1'\n",
       "\\item 'nonMSN\\_BR\\_DLX1'\n",
       "\\item 'sample\\_number\\_presented\\_CBH'\n",
       "\\item 'CTX\\_CBH'\n",
       "\\item 'CB\\_CBH'\n",
       "\\item 'Heart\\_CBH'\n",
       "\\item 'Kidney\\_CBH'\n",
       "\\item 'BG\\_CBH'\n",
       "\\item 'THAL\\_CBH'\n",
       "\\item 'HIP\\_CBH'\n",
       "\\item 'Liver\\_CBH'\n",
       "\\item 'PF\\_CBH'\n",
       "\\item 'F\\_CBH'\n",
       "\\item 'P\\_CBH'\n",
       "\\item 'O\\_CBH'\n",
       "\\item 'T\\_CBH'\n",
       "\\item 'Ins\\_CBH'\n",
       "\\item 'Cau\\_CBH'\n",
       "\\item 'Put\\_CBH'\n",
       "\\item 'GP\\_CBH'\n",
       "\\item 'CA1\\_CBH'\n",
       "\\item 'CA3\\_CBH'\n",
       "\\item 'DG\\_CBH'\n",
       "\\item 'DLX1\\_CBH'\n",
       "\\item 'TBR1\\_CBH'\n",
       "\\item 'Brain\\_only\\_CBH'\n",
       "\\item 'CTX\\_BG\\_shared\\_CBH'\n",
       "\\item 'BG\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_only\\_CBH'\n",
       "\\item 'BG\\_only\\_CBH'\n",
       "\\item 'GP\\_only\\_CBH'\n",
       "\\item 'HIP\\_only\\_CBH'\n",
       "\\item 'IN\\_only\\_CBH'\n",
       "\\item 'TBR\\_only\\_CBH'\n",
       "\\item 'TBR\\_BR\\_CBH'\n",
       "\\item 'DLX\\_BR\\_CBH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X'\n",
       "2. 'variant_list'\n",
       "3. 'sample_number_presented'\n",
       "4. 'CTX'\n",
       "5. 'CB'\n",
       "6. 'Heart'\n",
       "7. 'Kidney'\n",
       "8. 'BG'\n",
       "9. 'THAL'\n",
       "10. 'Liver'\n",
       "11. 'L'\n",
       "12. 'R'\n",
       "13. 'PF'\n",
       "14. 'F'\n",
       "15. 'P'\n",
       "16. 'O'\n",
       "17. 'T'\n",
       "18. 'Ins'\n",
       "19. 'Cau'\n",
       "20. 'Put'\n",
       "21. 'GP'\n",
       "22. 'Bulk'\n",
       "23. 'DLX1'\n",
       "24. 'TBR1'\n",
       "25. 'COUPTF2'\n",
       "26. 'NeuN'\n",
       "27. 'LHX2'\n",
       "28. 'Olig2'\n",
       "29. 'PU1'\n",
       "30. 'DARPP32M.NeuNP'\n",
       "31. 'DARPP32P.NeuNP'\n",
       "32. 'Cardiomyocyte'\n",
       "33. 'Brain_only'\n",
       "34. 'CTX_only'\n",
       "35. 'CTX_BG_only'\n",
       "36. 'BG_only'\n",
       "37. 'THAL_only'\n",
       "38. 'Left_only'\n",
       "39. 'Right_only'\n",
       "40. 'L_BR_PF_only'\n",
       "41. 'L_BR_F_only'\n",
       "42. 'L_BR_P_only'\n",
       "43. 'L_BR_O_only'\n",
       "44. 'L_BR_Ins_only'\n",
       "45. 'R_BR_PF_only'\n",
       "46. 'R_BR_F_only'\n",
       "47. 'R_BR_P_only'\n",
       "48. 'R_BR_O_only'\n",
       "49. 'R_BR_Ins_only'\n",
       "50. 'IN_only'\n",
       "51. 'COUP_only'\n",
       "52. 'TBR_only'\n",
       "53. 'TBR_BR'\n",
       "54. 'DLX_BR'\n",
       "55. 'COUP_BR'\n",
       "56. 'MSN_BR'\n",
       "57. 'nonMSN_BR'\n",
       "58. 'sample_number_presented_sort'\n",
       "59. 'CTX_sort'\n",
       "60. 'CB_sort'\n",
       "61. 'Heart_sort'\n",
       "62. 'Kidney_sort'\n",
       "63. 'BG_sort'\n",
       "64. 'THAL_sort'\n",
       "65. 'Liver_sort'\n",
       "66. 'L_sort'\n",
       "67. 'R_sort'\n",
       "68. 'PF_sort'\n",
       "69. 'F_sort'\n",
       "70. 'P_sort'\n",
       "71. 'O_sort'\n",
       "72. 'T_sort'\n",
       "73. 'Ins_sort'\n",
       "74. 'Cau_sort'\n",
       "75. 'Put_sort'\n",
       "76. 'GP_sort'\n",
       "77. 'Bulk_sort'\n",
       "78. 'DLX1_sort'\n",
       "79. 'TBR1_sort'\n",
       "80. 'COUPTF2_sort'\n",
       "81. 'NeuN_sort'\n",
       "82. 'LHX2_sort'\n",
       "83. 'Olig2_sort'\n",
       "84. 'PU1_sort'\n",
       "85. 'DARPP32M.NeuNP_sort'\n",
       "86. 'DARPP32P.NeuNP_sort'\n",
       "87. 'Cardiomyocyte_sort'\n",
       "88. 'Brain_only_sort'\n",
       "89. 'CTX_only_sort'\n",
       "90. 'CTX_BG_only_sort'\n",
       "91. 'BG_only_sort'\n",
       "92. 'THAL_only_sort'\n",
       "93. 'Left_only_sort'\n",
       "94. 'Right_only_sort'\n",
       "95. 'L_BR_PF_only_sort'\n",
       "96. 'L_BR_F_only_sort'\n",
       "97. 'L_BR_P_only_sort'\n",
       "98. 'L_BR_O_only_sort'\n",
       "99. 'L_BR_Ins_only_sort'\n",
       "100. 'R_BR_PF_only_sort'\n",
       "101. 'R_BR_F_only_sort'\n",
       "102. 'R_BR_P_only_sort'\n",
       "103. 'R_BR_O_only_sort'\n",
       "104. 'R_BR_Ins_only_sort'\n",
       "105. 'IN_only_sort'\n",
       "106. 'COUP_only_sort'\n",
       "107. 'TBR_only_sort'\n",
       "108. 'TBR_BR_sort'\n",
       "109. 'DLX_BR_sort'\n",
       "110. 'COUP_BR_sort'\n",
       "111. 'MSN_BR_sort'\n",
       "112. 'nonMSN_BR_sort'\n",
       "113. 'COUPTF2_enriched_sort'\n",
       "114. 'sample_number_presented_COUPTF2'\n",
       "115. 'CTX_COUPTF2'\n",
       "116. 'CB_COUPTF2'\n",
       "117. 'Heart_COUPTF2'\n",
       "118. 'Kidney_COUPTF2'\n",
       "119. 'BG_COUPTF2'\n",
       "120. 'THAL_COUPTF2'\n",
       "121. 'Liver_COUPTF2'\n",
       "122. 'L_COUPTF2'\n",
       "123. 'R_COUPTF2'\n",
       "124. 'PF_COUPTF2'\n",
       "125. 'F_COUPTF2'\n",
       "126. 'P_COUPTF2'\n",
       "127. 'O_COUPTF2'\n",
       "128. 'T_COUPTF2'\n",
       "129. 'Ins_COUPTF2'\n",
       "130. 'Cau_COUPTF2'\n",
       "131. 'Put_COUPTF2'\n",
       "132. 'GP_COUPTF2'\n",
       "133. 'Bulk_COUPTF2'\n",
       "134. 'DLX1_COUPTF2'\n",
       "135. 'TBR1_COUPTF2'\n",
       "136. 'COUPTF2_COUPTF2'\n",
       "137. 'NeuN_COUPTF2'\n",
       "138. 'LHX2_COUPTF2'\n",
       "139. 'Olig2_COUPTF2'\n",
       "140. 'PU1_COUPTF2'\n",
       "141. 'DARPP32M.NeuNP_COUPTF2'\n",
       "142. 'DARPP32P.NeuNP_COUPTF2'\n",
       "143. 'Cardiomyocyte_COUPTF2'\n",
       "144. 'Brain_only_COUPTF2'\n",
       "145. 'CTX_only_COUPTF2'\n",
       "146. 'CTX_BG_only_COUPTF2'\n",
       "147. 'BG_only_COUPTF2'\n",
       "148. 'THAL_only_COUPTF2'\n",
       "149. 'Left_only_COUPTF2'\n",
       "150. 'Right_only_COUPTF2'\n",
       "151. 'L_BR_PF_only_COUPTF2'\n",
       "152. 'L_BR_F_only_COUPTF2'\n",
       "153. 'L_BR_P_only_COUPTF2'\n",
       "154. 'L_BR_O_only_COUPTF2'\n",
       "155. 'L_BR_Ins_only_COUPTF2'\n",
       "156. 'R_BR_PF_only_COUPTF2'\n",
       "157. 'R_BR_F_only_COUPTF2'\n",
       "158. 'R_BR_P_only_COUPTF2'\n",
       "159. 'R_BR_O_only_COUPTF2'\n",
       "160. 'R_BR_Ins_only_COUPTF2'\n",
       "161. 'IN_only_COUPTF2'\n",
       "162. 'COUP_only_COUPTF2'\n",
       "163. 'TBR_only_COUPTF2'\n",
       "164. 'TBR_BR_COUPTF2'\n",
       "165. 'DLX_BR_COUPTF2'\n",
       "166. 'COUP_BR_COUPTF2'\n",
       "167. 'MSN_BR_COUPTF2'\n",
       "168. 'nonMSN_BR_COUPTF2'\n",
       "169. 'COUPTF2_enriched_COUPTF2'\n",
       "170. 'sample_number_presented_TBR1'\n",
       "171. 'CTX_TBR1'\n",
       "172. 'CB_TBR1'\n",
       "173. 'Heart_TBR1'\n",
       "174. 'Kidney_TBR1'\n",
       "175. 'BG_TBR1'\n",
       "176. 'THAL_TBR1'\n",
       "177. 'Liver_TBR1'\n",
       "178. 'L_TBR1'\n",
       "179. 'R_TBR1'\n",
       "180. 'PF_TBR1'\n",
       "181. 'F_TBR1'\n",
       "182. 'P_TBR1'\n",
       "183. 'O_TBR1'\n",
       "184. 'T_TBR1'\n",
       "185. 'Ins_TBR1'\n",
       "186. 'Cau_TBR1'\n",
       "187. 'Put_TBR1'\n",
       "188. 'GP_TBR1'\n",
       "189. 'Bulk_TBR1'\n",
       "190. 'DLX1_TBR1'\n",
       "191. 'TBR1_TBR1'\n",
       "192. 'COUPTF2_TBR1'\n",
       "193. 'NeuN_TBR1'\n",
       "194. 'LHX2_TBR1'\n",
       "195. 'Olig2_TBR1'\n",
       "196. 'PU1_TBR1'\n",
       "197. 'DARPP32M.NeuNP_TBR1'\n",
       "198. 'DARPP32P.NeuNP_TBR1'\n",
       "199. 'Cardiomyocyte_TBR1'\n",
       "200. 'Brain_only_TBR1'\n",
       "201. 'CTX_only_TBR1'\n",
       "202. 'CTX_BG_only_TBR1'\n",
       "203. 'BG_only_TBR1'\n",
       "204. 'THAL_only_TBR1'\n",
       "205. 'Left_only_TBR1'\n",
       "206. 'Right_only_TBR1'\n",
       "207. 'L_BR_PF_only_TBR1'\n",
       "208. 'L_BR_F_only_TBR1'\n",
       "209. 'L_BR_P_only_TBR1'\n",
       "210. 'L_BR_O_only_TBR1'\n",
       "211. 'L_BR_Ins_only_TBR1'\n",
       "212. 'R_BR_PF_only_TBR1'\n",
       "213. 'R_BR_F_only_TBR1'\n",
       "214. 'R_BR_P_only_TBR1'\n",
       "215. 'R_BR_O_only_TBR1'\n",
       "216. 'R_BR_Ins_only_TBR1'\n",
       "217. 'IN_only_TBR1'\n",
       "218. 'COUP_only_TBR1'\n",
       "219. 'TBR_only_TBR1'\n",
       "220. 'TBR_BR_TBR1'\n",
       "221. 'DLX_BR_TBR1'\n",
       "222. 'COUP_BR_TBR1'\n",
       "223. 'MSN_BR_TBR1'\n",
       "224. 'nonMSN_BR_TBR1'\n",
       "225. 'sample_number_presented_DLX1'\n",
       "226. 'CTX_DLX1'\n",
       "227. 'CB_DLX1'\n",
       "228. 'Heart_DLX1'\n",
       "229. 'Kidney_DLX1'\n",
       "230. 'BG_DLX1'\n",
       "231. 'THAL_DLX1'\n",
       "232. 'Liver_DLX1'\n",
       "233. 'L_DLX1'\n",
       "234. 'R_DLX1'\n",
       "235. 'PF_DLX1'\n",
       "236. 'F_DLX1'\n",
       "237. 'P_DLX1'\n",
       "238. 'O_DLX1'\n",
       "239. 'T_DLX1'\n",
       "240. 'Ins_DLX1'\n",
       "241. 'Cau_DLX1'\n",
       "242. 'Put_DLX1'\n",
       "243. 'GP_DLX1'\n",
       "244. 'Bulk_DLX1'\n",
       "245. 'DLX1_DLX1'\n",
       "246. 'TBR1_DLX1'\n",
       "247. 'COUPTF2_DLX1'\n",
       "248. 'NeuN_DLX1'\n",
       "249. 'LHX2_DLX1'\n",
       "250. 'Olig2_DLX1'\n",
       "251. 'PU1_DLX1'\n",
       "252. 'DARPP32M.NeuNP_DLX1'\n",
       "253. 'DARPP32P.NeuNP_DLX1'\n",
       "254. 'Cardiomyocyte_DLX1'\n",
       "255. 'Brain_only_DLX1'\n",
       "256. 'CTX_only_DLX1'\n",
       "257. 'CTX_BG_only_DLX1'\n",
       "258. 'BG_only_DLX1'\n",
       "259. 'THAL_only_DLX1'\n",
       "260. 'Left_only_DLX1'\n",
       "261. 'Right_only_DLX1'\n",
       "262. 'L_BR_PF_only_DLX1'\n",
       "263. 'L_BR_F_only_DLX1'\n",
       "264. 'L_BR_P_only_DLX1'\n",
       "265. 'L_BR_O_only_DLX1'\n",
       "266. 'L_BR_Ins_only_DLX1'\n",
       "267. 'R_BR_PF_only_DLX1'\n",
       "268. 'R_BR_F_only_DLX1'\n",
       "269. 'R_BR_P_only_DLX1'\n",
       "270. 'R_BR_O_only_DLX1'\n",
       "271. 'R_BR_Ins_only_DLX1'\n",
       "272. 'IN_only_DLX1'\n",
       "273. 'COUP_only_DLX1'\n",
       "274. 'TBR_only_DLX1'\n",
       "275. 'TBR_BR_DLX1'\n",
       "276. 'DLX_BR_DLX1'\n",
       "277. 'COUP_BR_DLX1'\n",
       "278. 'MSN_BR_DLX1'\n",
       "279. 'nonMSN_BR_DLX1'\n",
       "280. 'sample_number_presented_CBH'\n",
       "281. 'CTX_CBH'\n",
       "282. 'CB_CBH'\n",
       "283. 'Heart_CBH'\n",
       "284. 'Kidney_CBH'\n",
       "285. 'BG_CBH'\n",
       "286. 'THAL_CBH'\n",
       "287. 'HIP_CBH'\n",
       "288. 'Liver_CBH'\n",
       "289. 'PF_CBH'\n",
       "290. 'F_CBH'\n",
       "291. 'P_CBH'\n",
       "292. 'O_CBH'\n",
       "293. 'T_CBH'\n",
       "294. 'Ins_CBH'\n",
       "295. 'Cau_CBH'\n",
       "296. 'Put_CBH'\n",
       "297. 'GP_CBH'\n",
       "298. 'CA1_CBH'\n",
       "299. 'CA3_CBH'\n",
       "300. 'DG_CBH'\n",
       "301. 'DLX1_CBH'\n",
       "302. 'TBR1_CBH'\n",
       "303. 'Brain_only_CBH'\n",
       "304. 'CTX_BG_shared_CBH'\n",
       "305. 'BG_HIP_shared_CBH'\n",
       "306. 'CTX_HIP_shared_CBH'\n",
       "307. 'CTX_only_CBH'\n",
       "308. 'BG_only_CBH'\n",
       "309. 'GP_only_CBH'\n",
       "310. 'HIP_only_CBH'\n",
       "311. 'IN_only_CBH'\n",
       "312. 'TBR_only_CBH'\n",
       "313. 'TBR_BR_CBH'\n",
       "314. 'DLX_BR_CBH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"X\"                               \"variant_list\"                   \n",
       "  [3] \"sample_number_presented\"         \"CTX\"                            \n",
       "  [5] \"CB\"                              \"Heart\"                          \n",
       "  [7] \"Kidney\"                          \"BG\"                             \n",
       "  [9] \"THAL\"                            \"Liver\"                          \n",
       " [11] \"L\"                               \"R\"                              \n",
       " [13] \"PF\"                              \"F\"                              \n",
       " [15] \"P\"                               \"O\"                              \n",
       " [17] \"T\"                               \"Ins\"                            \n",
       " [19] \"Cau\"                             \"Put\"                            \n",
       " [21] \"GP\"                              \"Bulk\"                           \n",
       " [23] \"DLX1\"                            \"TBR1\"                           \n",
       " [25] \"COUPTF2\"                         \"NeuN\"                           \n",
       " [27] \"LHX2\"                            \"Olig2\"                          \n",
       " [29] \"PU1\"                             \"DARPP32M.NeuNP\"                 \n",
       " [31] \"DARPP32P.NeuNP\"                  \"Cardiomyocyte\"                  \n",
       " [33] \"Brain_only\"                      \"CTX_only\"                       \n",
       " [35] \"CTX_BG_only\"                     \"BG_only\"                        \n",
       " [37] \"THAL_only\"                       \"Left_only\"                      \n",
       " [39] \"Right_only\"                      \"L_BR_PF_only\"                   \n",
       " [41] \"L_BR_F_only\"                     \"L_BR_P_only\"                    \n",
       " [43] \"L_BR_O_only\"                     \"L_BR_Ins_only\"                  \n",
       " [45] \"R_BR_PF_only\"                    \"R_BR_F_only\"                    \n",
       " [47] \"R_BR_P_only\"                     \"R_BR_O_only\"                    \n",
       " [49] \"R_BR_Ins_only\"                   \"IN_only\"                        \n",
       " [51] \"COUP_only\"                       \"TBR_only\"                       \n",
       " [53] \"TBR_BR\"                          \"DLX_BR\"                         \n",
       " [55] \"COUP_BR\"                         \"MSN_BR\"                         \n",
       " [57] \"nonMSN_BR\"                       \"sample_number_presented_sort\"   \n",
       " [59] \"CTX_sort\"                        \"CB_sort\"                        \n",
       " [61] \"Heart_sort\"                      \"Kidney_sort\"                    \n",
       " [63] \"BG_sort\"                         \"THAL_sort\"                      \n",
       " [65] \"Liver_sort\"                      \"L_sort\"                         \n",
       " [67] \"R_sort\"                          \"PF_sort\"                        \n",
       " [69] \"F_sort\"                          \"P_sort\"                         \n",
       " [71] \"O_sort\"                          \"T_sort\"                         \n",
       " [73] \"Ins_sort\"                        \"Cau_sort\"                       \n",
       " [75] \"Put_sort\"                        \"GP_sort\"                        \n",
       " [77] \"Bulk_sort\"                       \"DLX1_sort\"                      \n",
       " [79] \"TBR1_sort\"                       \"COUPTF2_sort\"                   \n",
       " [81] \"NeuN_sort\"                       \"LHX2_sort\"                      \n",
       " [83] \"Olig2_sort\"                      \"PU1_sort\"                       \n",
       " [85] \"DARPP32M.NeuNP_sort\"             \"DARPP32P.NeuNP_sort\"            \n",
       " [87] \"Cardiomyocyte_sort\"              \"Brain_only_sort\"                \n",
       " [89] \"CTX_only_sort\"                   \"CTX_BG_only_sort\"               \n",
       " [91] \"BG_only_sort\"                    \"THAL_only_sort\"                 \n",
       " [93] \"Left_only_sort\"                  \"Right_only_sort\"                \n",
       " [95] \"L_BR_PF_only_sort\"               \"L_BR_F_only_sort\"               \n",
       " [97] \"L_BR_P_only_sort\"                \"L_BR_O_only_sort\"               \n",
       " [99] \"L_BR_Ins_only_sort\"              \"R_BR_PF_only_sort\"              \n",
       "[101] \"R_BR_F_only_sort\"                \"R_BR_P_only_sort\"               \n",
       "[103] \"R_BR_O_only_sort\"                \"R_BR_Ins_only_sort\"             \n",
       "[105] \"IN_only_sort\"                    \"COUP_only_sort\"                 \n",
       "[107] \"TBR_only_sort\"                   \"TBR_BR_sort\"                    \n",
       "[109] \"DLX_BR_sort\"                     \"COUP_BR_sort\"                   \n",
       "[111] \"MSN_BR_sort\"                     \"nonMSN_BR_sort\"                 \n",
       "[113] \"COUPTF2_enriched_sort\"           \"sample_number_presented_COUPTF2\"\n",
       "[115] \"CTX_COUPTF2\"                     \"CB_COUPTF2\"                     \n",
       "[117] \"Heart_COUPTF2\"                   \"Kidney_COUPTF2\"                 \n",
       "[119] \"BG_COUPTF2\"                      \"THAL_COUPTF2\"                   \n",
       "[121] \"Liver_COUPTF2\"                   \"L_COUPTF2\"                      \n",
       "[123] \"R_COUPTF2\"                       \"PF_COUPTF2\"                     \n",
       "[125] \"F_COUPTF2\"                       \"P_COUPTF2\"                      \n",
       "[127] \"O_COUPTF2\"                       \"T_COUPTF2\"                      \n",
       "[129] \"Ins_COUPTF2\"                     \"Cau_COUPTF2\"                    \n",
       "[131] \"Put_COUPTF2\"                     \"GP_COUPTF2\"                     \n",
       "[133] \"Bulk_COUPTF2\"                    \"DLX1_COUPTF2\"                   \n",
       "[135] \"TBR1_COUPTF2\"                    \"COUPTF2_COUPTF2\"                \n",
       "[137] \"NeuN_COUPTF2\"                    \"LHX2_COUPTF2\"                   \n",
       "[139] \"Olig2_COUPTF2\"                   \"PU1_COUPTF2\"                    \n",
       "[141] \"DARPP32M.NeuNP_COUPTF2\"          \"DARPP32P.NeuNP_COUPTF2\"         \n",
       "[143] \"Cardiomyocyte_COUPTF2\"           \"Brain_only_COUPTF2\"             \n",
       "[145] \"CTX_only_COUPTF2\"                \"CTX_BG_only_COUPTF2\"            \n",
       "[147] \"BG_only_COUPTF2\"                 \"THAL_only_COUPTF2\"              \n",
       "[149] \"Left_only_COUPTF2\"               \"Right_only_COUPTF2\"             \n",
       "[151] \"L_BR_PF_only_COUPTF2\"            \"L_BR_F_only_COUPTF2\"            \n",
       "[153] \"L_BR_P_only_COUPTF2\"             \"L_BR_O_only_COUPTF2\"            \n",
       "[155] \"L_BR_Ins_only_COUPTF2\"           \"R_BR_PF_only_COUPTF2\"           \n",
       "[157] \"R_BR_F_only_COUPTF2\"             \"R_BR_P_only_COUPTF2\"            \n",
       "[159] \"R_BR_O_only_COUPTF2\"             \"R_BR_Ins_only_COUPTF2\"          \n",
       "[161] \"IN_only_COUPTF2\"                 \"COUP_only_COUPTF2\"              \n",
       "[163] \"TBR_only_COUPTF2\"                \"TBR_BR_COUPTF2\"                 \n",
       "[165] \"DLX_BR_COUPTF2\"                  \"COUP_BR_COUPTF2\"                \n",
       "[167] \"MSN_BR_COUPTF2\"                  \"nonMSN_BR_COUPTF2\"              \n",
       "[169] \"COUPTF2_enriched_COUPTF2\"        \"sample_number_presented_TBR1\"   \n",
       "[171] \"CTX_TBR1\"                        \"CB_TBR1\"                        \n",
       "[173] \"Heart_TBR1\"                      \"Kidney_TBR1\"                    \n",
       "[175] \"BG_TBR1\"                         \"THAL_TBR1\"                      \n",
       "[177] \"Liver_TBR1\"                      \"L_TBR1\"                         \n",
       "[179] \"R_TBR1\"                          \"PF_TBR1\"                        \n",
       "[181] \"F_TBR1\"                          \"P_TBR1\"                         \n",
       "[183] \"O_TBR1\"                          \"T_TBR1\"                         \n",
       "[185] \"Ins_TBR1\"                        \"Cau_TBR1\"                       \n",
       "[187] \"Put_TBR1\"                        \"GP_TBR1\"                        \n",
       "[189] \"Bulk_TBR1\"                       \"DLX1_TBR1\"                      \n",
       "[191] \"TBR1_TBR1\"                       \"COUPTF2_TBR1\"                   \n",
       "[193] \"NeuN_TBR1\"                       \"LHX2_TBR1\"                      \n",
       "[195] \"Olig2_TBR1\"                      \"PU1_TBR1\"                       \n",
       "[197] \"DARPP32M.NeuNP_TBR1\"             \"DARPP32P.NeuNP_TBR1\"            \n",
       "[199] \"Cardiomyocyte_TBR1\"              \"Brain_only_TBR1\"                \n",
       "[201] \"CTX_only_TBR1\"                   \"CTX_BG_only_TBR1\"               \n",
       "[203] \"BG_only_TBR1\"                    \"THAL_only_TBR1\"                 \n",
       "[205] \"Left_only_TBR1\"                  \"Right_only_TBR1\"                \n",
       "[207] \"L_BR_PF_only_TBR1\"               \"L_BR_F_only_TBR1\"               \n",
       "[209] \"L_BR_P_only_TBR1\"                \"L_BR_O_only_TBR1\"               \n",
       "[211] \"L_BR_Ins_only_TBR1\"              \"R_BR_PF_only_TBR1\"              \n",
       "[213] \"R_BR_F_only_TBR1\"                \"R_BR_P_only_TBR1\"               \n",
       "[215] \"R_BR_O_only_TBR1\"                \"R_BR_Ins_only_TBR1\"             \n",
       "[217] \"IN_only_TBR1\"                    \"COUP_only_TBR1\"                 \n",
       "[219] \"TBR_only_TBR1\"                   \"TBR_BR_TBR1\"                    \n",
       "[221] \"DLX_BR_TBR1\"                     \"COUP_BR_TBR1\"                   \n",
       "[223] \"MSN_BR_TBR1\"                     \"nonMSN_BR_TBR1\"                 \n",
       "[225] \"sample_number_presented_DLX1\"    \"CTX_DLX1\"                       \n",
       "[227] \"CB_DLX1\"                         \"Heart_DLX1\"                     \n",
       "[229] \"Kidney_DLX1\"                     \"BG_DLX1\"                        \n",
       "[231] \"THAL_DLX1\"                       \"Liver_DLX1\"                     \n",
       "[233] \"L_DLX1\"                          \"R_DLX1\"                         \n",
       "[235] \"PF_DLX1\"                         \"F_DLX1\"                         \n",
       "[237] \"P_DLX1\"                          \"O_DLX1\"                         \n",
       "[239] \"T_DLX1\"                          \"Ins_DLX1\"                       \n",
       "[241] \"Cau_DLX1\"                        \"Put_DLX1\"                       \n",
       "[243] \"GP_DLX1\"                         \"Bulk_DLX1\"                      \n",
       "[245] \"DLX1_DLX1\"                       \"TBR1_DLX1\"                      \n",
       "[247] \"COUPTF2_DLX1\"                    \"NeuN_DLX1\"                      \n",
       "[249] \"LHX2_DLX1\"                       \"Olig2_DLX1\"                     \n",
       "[251] \"PU1_DLX1\"                        \"DARPP32M.NeuNP_DLX1\"            \n",
       "[253] \"DARPP32P.NeuNP_DLX1\"             \"Cardiomyocyte_DLX1\"             \n",
       "[255] \"Brain_only_DLX1\"                 \"CTX_only_DLX1\"                  \n",
       "[257] \"CTX_BG_only_DLX1\"                \"BG_only_DLX1\"                   \n",
       "[259] \"THAL_only_DLX1\"                  \"Left_only_DLX1\"                 \n",
       "[261] \"Right_only_DLX1\"                 \"L_BR_PF_only_DLX1\"              \n",
       "[263] \"L_BR_F_only_DLX1\"                \"L_BR_P_only_DLX1\"               \n",
       "[265] \"L_BR_O_only_DLX1\"                \"L_BR_Ins_only_DLX1\"             \n",
       "[267] \"R_BR_PF_only_DLX1\"               \"R_BR_F_only_DLX1\"               \n",
       "[269] \"R_BR_P_only_DLX1\"                \"R_BR_O_only_DLX1\"               \n",
       "[271] \"R_BR_Ins_only_DLX1\"              \"IN_only_DLX1\"                   \n",
       "[273] \"COUP_only_DLX1\"                  \"TBR_only_DLX1\"                  \n",
       "[275] \"TBR_BR_DLX1\"                     \"DLX_BR_DLX1\"                    \n",
       "[277] \"COUP_BR_DLX1\"                    \"MSN_BR_DLX1\"                    \n",
       "[279] \"nonMSN_BR_DLX1\"                  \"sample_number_presented_CBH\"    \n",
       "[281] \"CTX_CBH\"                         \"CB_CBH\"                         \n",
       "[283] \"Heart_CBH\"                       \"Kidney_CBH\"                     \n",
       "[285] \"BG_CBH\"                          \"THAL_CBH\"                       \n",
       "[287] \"HIP_CBH\"                         \"Liver_CBH\"                      \n",
       "[289] \"PF_CBH\"                          \"F_CBH\"                          \n",
       "[291] \"P_CBH\"                           \"O_CBH\"                          \n",
       "[293] \"T_CBH\"                           \"Ins_CBH\"                        \n",
       "[295] \"Cau_CBH\"                         \"Put_CBH\"                        \n",
       "[297] \"GP_CBH\"                          \"CA1_CBH\"                        \n",
       "[299] \"CA3_CBH\"                         \"DG_CBH\"                         \n",
       "[301] \"DLX1_CBH\"                        \"TBR1_CBH\"                       \n",
       "[303] \"Brain_only_CBH\"                  \"CTX_BG_shared_CBH\"              \n",
       "[305] \"BG_HIP_shared_CBH\"               \"CTX_HIP_shared_CBH\"             \n",
       "[307] \"CTX_only_CBH\"                    \"BG_only_CBH\"                    \n",
       "[309] \"GP_only_CBH\"                     \"HIP_only_CBH\"                   \n",
       "[311] \"IN_only_CBH\"                     \"TBR_only_CBH\"                   \n",
       "[313] \"TBR_BR_CBH\"                      \"DLX_BR_CBH\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(filterdata)\n",
    "head(annotated)\n",
    "names(filterdata)\n",
    "names(annotated)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Confirmed total variants (WGS + Amp):  287\"\n"
     ]
    }
   ],
   "source": [
    "#sort df for only WGS + Ampliseq confirmed (1) -> confirmed variants (excluding JGG controls)\n",
    "totalConfVarsInclCtrl = length(unique( subset(filterdata, Mosaic_shared_btw_WGS_Amp == 1) ))\n",
    "confirmedVars = subset(filterdata, Mosaic_shared_btw_WGS_Amp == 1 & Organ != \"JGG\")\n",
    "\n",
    "numTotalVars = length(unique((confirmedVars$CHROM.POS.REF.ALT)))\n",
    "pretext = \"Confirmed total variants (WGS + Amp): \"\n",
    "print(paste(pretext, length(unique((confirmedVars$CHROM.POS.REF.ALT))))) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Correlation matrix of Variant x Variant using ComplexHeatmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#ComplexHeatmap plot (working)\n",
    "#PEARSON\n",
    "\n",
    "varsConfirmedIn2Sample = subset(annotated, sample_number_presented >= 2)\n",
    "\n",
    "confirmed2SampleVars = subset(confirmedVars, CHROM.POS.REF.ALT %in% varsConfirmedIn2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleVars$MAF), confirmed2SampleVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "LR_colors = colorRamp2(c(-1, 0, 1), c(\"#d68426\", \"#ffffff\", \"#2a1abf\"))\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[, colSums(varsConfirmedIn2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"287_correlation_7614_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BULK = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Bulk\")),\n",
    "HEART = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Heart\")),\n",
    "LIVER = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Liver\")),\n",
    "KIDNEY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Kidney\")),\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CB = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cerebellum\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"THAL\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"R\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\",\"Left Only\")),\n",
    "L_BR_PF_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR PF Only\")),\n",
    "L_BR_F_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR F Only\")),\n",
    "L_BR_O_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR O Only\")),\n",
    "R_BR_PF_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"R BR PF Only\")),\n",
    "R_BR_F_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"R BR F Only\")),\n",
    "R_BR_O_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR O Only\")))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "color_list = list(LR_ONLY = LR_colors,\n",
    "                COUP_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"black\")),\n",
    "                IN_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"darkgreen\")),\n",
    "                TBR_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"purple\")), \n",
    "                TBR_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"#e58f9d\")),\n",
    "                DLX_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"maroon\")))\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BULK = varsConfirmedIn2Sample$Bulk,\n",
    "                                        HEART = varsConfirmedIn2Sample$Heart,\n",
    "                                        LIVER = varsConfirmedIn2Sample$Liver,\n",
    "                                        KIDNEY = varsConfirmedIn2Sample$Kidney,\n",
    "                                        BRAIN_ONLY = varsConfirmedIn2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedIn2Sample$CB,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX,\n",
    "                                        BG = varsConfirmedIn2Sample$BG,\n",
    "                                        THAL = varsConfirmedIn2Sample$THAL,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP,\n",
    "                                        PF = varsConfirmedIn2Sample$PF,\n",
    "                                        F = varsConfirmedIn2Sample$F,\n",
    "                                        P = varsConfirmedIn2Sample$P,\n",
    "                                        O = varsConfirmedIn2Sample$O,\n",
    "                                        T = varsConfirmedIn2Sample$T,\n",
    "                                        INS = varsConfirmedIn2Sample$Ins,\n",
    "                                        CAU = varsConfirmedIn2Sample$Cau,\n",
    "                                        PUT = varsConfirmedIn2Sample$Put,\n",
    "                                        GP = varsConfirmedIn2Sample$GP,\n",
    "                                        CTX_ONLY = varsConfirmedIn2Sample$CTX_only,\n",
    "                                        CTX_BG_ONLY = varsConfirmedIn2Sample$CTX_BG_only,\n",
    "                                        BG_ONLY = varsConfirmedIn2Sample$BG_only,\n",
    "                                        THAL_ONLY = varsConfirmedIn2Sample$THAL_only,\n",
    "                                        L = varsConfirmedIn2Sample$L,\n",
    "                                        R = varsConfirmedIn2Sample$R,\n",
    "                                        LR_ONLY = varsConfirmedIn2Sample$Left_only - varsConfirmedIn2Sample$Right_only, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedIn2Sample$L_BR_PF_only,\n",
    "                                        L_BR_F_ONLY = varsConfirmedIn2Sample$L_BR_F_only,\n",
    "                                        L_BR_P_ONLY = varsConfirmedIn2Sample$L_BR_P_only,\n",
    "                                        L_BR_O_ONLY = varsConfirmedIn2Sample$L_BR_O_only,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedIn2Sample$L_BR_Ins_only,\n",
    "                                        R_BR_PF_ONLY = varsConfirmedIn2Sample$R_BR_PF_only,\n",
    "                                        R_BR_F_ONLY = varsConfirmedIn2Sample$R_BR_F_only,\n",
    "                                        R_BR_P_ONLY = varsConfirmedIn2Sample$R_BR_P_only,\n",
    "                                        R_BR_O_ONLY = varsConfirmedIn2Sample$R_BR_O_only,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedIn2Sample$R_BR_Ins_only,\n",
    "                                        DARPP32M.NeuNP = varsConfirmedIn2Sample$DARPP32M.NeuNP,\n",
    "                                        DARPP32P.NeuNP = varsConfirmedIn2Sample$DARPP32P.NeuNP,\n",
    "                                        LHX2 = varsConfirmedIn2Sample$LHX2,\n",
    "                                        OLIG2 = varsConfirmedIn2Sample$Olig2,\n",
    "                                        PU1 = varsConfirmedIn2Sample$PU1,\n",
    "                                        NeuN = varsConfirmedIn2Sample$NeuN,\n",
    "                                        IN_ONLY = varsConfirmedIn2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedIn2Sample$COUPTF2,\n",
    "                                        COUP_BR = varsConfirmedIn2Sample$COUP_BR,\n",
    "                                        COUP_ONLY = varsConfirmedIn2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedIn2Sample$TBR1,\n",
    "                                        TBR_ONLY = varsConfirmedIn2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedIn2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR,\n",
    "                                        MSN_BR = varsConfirmedIn2Sample$MSN_BR,\n",
    "                                        nonMSN_BR = varsConfirmedIn2Sample$nonMSN_BR,\n",
    "                                        col = color_list,\n",
    "                                        annotation_legend_param = legendDets)\n",
    "\n",
    "leftMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_7614_Pearson_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "leftMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#SPEARMAN Heatmap\n",
    "pdf(\"287_correlation_7614_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BULK = varsConfirmedIn2Sample$Bulk,\n",
    "                                        HEART = varsConfirmedIn2Sample$Heart,\n",
    "                                        LIVER = varsConfirmedIn2Sample$Liver,\n",
    "                                        KIDNEY = varsConfirmedIn2Sample$Kidney,\n",
    "                                        BRAIN_ONLY = varsConfirmedIn2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedIn2Sample$CB,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX,\n",
    "                                        BG = varsConfirmedIn2Sample$BG,\n",
    "                                        THAL = varsConfirmedIn2Sample$THAL,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP,\n",
    "                                        PF = varsConfirmedIn2Sample$PF,\n",
    "                                        F = varsConfirmedIn2Sample$F,\n",
    "                                        P = varsConfirmedIn2Sample$P,\n",
    "                                        O = varsConfirmedIn2Sample$O,\n",
    "                                        T = varsConfirmedIn2Sample$T,\n",
    "                                        INS = varsConfirmedIn2Sample$Ins,\n",
    "                                        CAU = varsConfirmedIn2Sample$Cau,\n",
    "                                        PUT = varsConfirmedIn2Sample$Put,\n",
    "                                        GP = varsConfirmedIn2Sample$GP,\n",
    "                                        CTX_ONLY = varsConfirmedIn2Sample$CTX_only,\n",
    "                                        CTX_BG_ONLY = varsConfirmedIn2Sample$CTX_BG_only,\n",
    "                                        BG_ONLY = varsConfirmedIn2Sample$BG_only,\n",
    "                                        THAL_ONLY = varsConfirmedIn2Sample$THAL_only,\n",
    "                                        L = varsConfirmedIn2Sample$L,\n",
    "                                        R = varsConfirmedIn2Sample$R,\n",
    "                                        LR_ONLY = varsConfirmedIn2Sample$Left_only - varsConfirmedIn2Sample$Right_only, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedIn2Sample$L_BR_PF_only,\n",
    "                                        L_BR_F_ONLY = varsConfirmedIn2Sample$L_BR_F_only,\n",
    "                                        L_BR_P_ONLY = varsConfirmedIn2Sample$L_BR_P_only,\n",
    "                                        L_BR_O_ONLY = varsConfirmedIn2Sample$L_BR_O_only,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedIn2Sample$L_BR_Ins_only,\n",
    "                                        R_BR_PF_ONLY = varsConfirmedIn2Sample$R_BR_PF_only,\n",
    "                                        R_BR_F_ONLY = varsConfirmedIn2Sample$R_BR_F_only,\n",
    "                                        R_BR_P_ONLY = varsConfirmedIn2Sample$R_BR_P_only,\n",
    "                                        R_BR_O_ONLY = varsConfirmedIn2Sample$R_BR_O_only,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedIn2Sample$R_BR_Ins_only,\n",
    "                                        DARPP32M.NeuNP = varsConfirmedIn2Sample$DARPP32M.NeuNP,\n",
    "                                        DARPP32P.NeuNP = varsConfirmedIn2Sample$DARPP32P.NeuNP,\n",
    "                                        LHX2 = varsConfirmedIn2Sample$LHX2,\n",
    "                                        OLIG2 = varsConfirmedIn2Sample$Olig2,\n",
    "                                        PU1 = varsConfirmedIn2Sample$PU1,\n",
    "                                        NeuN = varsConfirmedIn2Sample$NeuN,\n",
    "                                        IN_ONLY = varsConfirmedIn2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedIn2Sample$COUPTF2,\n",
    "                                        COUP_BR = varsConfirmedIn2Sample$COUP_BR,\n",
    "                                        COUP_ONLY = varsConfirmedIn2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedIn2Sample$TBR1,\n",
    "                                        TBR_ONLY = varsConfirmedIn2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedIn2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR,\n",
    "                                        MSN_BR = varsConfirmedIn2Sample$MSN_BR,\n",
    "                                        nonMSN_BR = varsConfirmedIn2Sample$nonMSN_BR,\n",
    "                                        col = color_list,\n",
    "                                        annotation_legend_param = legendDets)\n",
    "\n",
    "\n",
    "leftMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#BULK\n",
    "\n",
    "bulkVars = subset(confirmedVars, Cell_Type == \"Bulk\")\n",
    "varsConfirmedInBulk2Sample = subset(varsConfirmedIn2Sample, Bulk == 1)\n",
    "confirmed2SampleBulkVars = subset(bulkVars, CHROM.POS.REF.ALT %in% varsConfirmedInBulk2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleBulkVars$MAF), confirmed2SampleBulkVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInBulk2Sample = varsConfirmedInBulk2Sample[, colSums(varsConfirmedInBulk2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"287_correlation_BULK_Pearson_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BULK = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Bulk\")),\n",
    "HEART = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Heart\")),\n",
    "LIVER = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Liver\")),\n",
    "KIDNEY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Kidney\")),\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CB = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cerebellum\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "L_BR_PF_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR PF Only\")),\n",
    "L_BR_F_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR F Only\")),\n",
    "L_BR_O_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR O Only\")),\n",
    "R_BR_PF_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"R BR PF Only\")),\n",
    "R_BR_F_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"R BR F Only\")),\n",
    "R_BR_O_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"L BR O Only\")))\n",
    "\n",
    "varsConfirmedInBulk2Sample = varsConfirmedInBulk2Sample[order(varsConfirmedInBulk2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BULK = varsConfirmedIn2Sample$Bulk,\n",
    "                                        HEART = varsConfirmedIn2Sample$Heart,\n",
    "                                        LIVER = varsConfirmedIn2Sample$Liver,\n",
    "                                        KIDNEY = varsConfirmedIn2Sample$Kidney,\n",
    "                                        BRAIN_ONLY = varsConfirmedIn2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedIn2Sample$CB,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX,\n",
    "                                        BG = varsConfirmedIn2Sample$BG,\n",
    "                                        THAL = varsConfirmedIn2Sample$THAL,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP,\n",
    "                                        PF = varsConfirmedIn2Sample$PF,\n",
    "                                        F = varsConfirmedIn2Sample$F,\n",
    "                                        P = varsConfirmedIn2Sample$P,\n",
    "                                        O = varsConfirmedIn2Sample$O,\n",
    "                                        T = varsConfirmedIn2Sample$T,\n",
    "                                        INS = varsConfirmedIn2Sample$Ins,\n",
    "                                        CAU = varsConfirmedIn2Sample$Cau,\n",
    "                                        PUT = varsConfirmedIn2Sample$Put,\n",
    "                                        GP = varsConfirmedIn2Sample$GP,\n",
    "                                        CTX_ONLY = varsConfirmedIn2Sample$CTX_only,\n",
    "                                        CTX_BG_ONLY = varsConfirmedIn2Sample$CTX_BG_only,\n",
    "                                        BG_ONLY = varsConfirmedIn2Sample$BG_only,\n",
    "                                        THAL_ONLY = varsConfirmedIn2Sample$THAL_only,\n",
    "                                        L = varsConfirmedIn2Sample$L,\n",
    "                                        R = varsConfirmedIn2Sample$R,\n",
    "                                        LR_ONLY = varsConfirmedIn2Sample$Left_only - varsConfirmedIn2Sample$Right_only, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedIn2Sample$L_BR_PF_only,\n",
    "                                        L_BR_F_ONLY = varsConfirmedIn2Sample$L_BR_F_only,\n",
    "                                        L_BR_P_ONLY = varsConfirmedIn2Sample$L_BR_P_only,\n",
    "                                        L_BR_O_ONLY = varsConfirmedIn2Sample$L_BR_O_only,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedIn2Sample$L_BR_Ins_only,\n",
    "                                        R_BR_PF_ONLY = varsConfirmedIn2Sample$R_BR_PF_only,\n",
    "                                        R_BR_F_ONLY = varsConfirmedIn2Sample$R_BR_F_only,\n",
    "                                        R_BR_P_ONLY = varsConfirmedIn2Sample$R_BR_P_only,\n",
    "                                        R_BR_O_ONLY = varsConfirmedIn2Sample$R_BR_O_only,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedIn2Sample$R_BR_Ins_only,\n",
    "                                        DARPP32M.NeuNP = varsConfirmedIn2Sample$DARPP32M.NeuNP,\n",
    "                                        DARPP32P.NeuNP = varsConfirmedIn2Sample$DARPP32P.NeuNP,\n",
    "                                        LHX2 = varsConfirmedIn2Sample$LHX2,\n",
    "                                        OLIG2 = varsConfirmedIn2Sample$Olig2,\n",
    "                                        PU1 = varsConfirmedIn2Sample$PU1,\n",
    "                                        NeuN = varsConfirmedIn2Sample$NeuN,\n",
    "                                        IN_ONLY = varsConfirmedIn2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedIn2Sample$COUPTF2,\n",
    "                                        COUP_BR = varsConfirmedIn2Sample$COUP_BR,\n",
    "                                        COUP_ONLY = varsConfirmedIn2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedIn2Sample$TBR1,\n",
    "                                        TBR_ONLY = varsConfirmedIn2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedIn2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR,\n",
    "                                        MSN_BR = varsConfirmedIn2Sample$MSN_BR,\n",
    "                                        nonMSN_BR = varsConfirmedIn2Sample$nonMSN_BR,\n",
    "                                        col = color_list,\n",
    "                                        annotation_legend_param = legendDets)\n",
    "\n",
    "bulkMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "bulkMapUpdate\n",
    "dev.off()\n",
    "\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "pdf(\"287_correlation_BULK_Spearman_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInBulk2Sample = varsConfirmedInBulk2Sample[order(varsConfirmedInBulk2Sample$X), ]\n",
    "\n",
    "bulkMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "bulkMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DLX1, TBR1, COUPTF2 Variant x Variant Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_DLX1'</li><li>'CTX_DLX1'</li><li>'BG_DLX1'</li><li>'THAL_DLX1'</li><li>'L_DLX1'</li><li>'R_DLX1'</li><li>'PF_DLX1'</li><li>'F_DLX1'</li><li>'P_DLX1'</li><li>'O_DLX1'</li><li>'T_DLX1'</li><li>'Ins_DLX1'</li><li>'Put_DLX1'</li><li>'GP_DLX1'</li><li>'COUPTF2_DLX1'</li><li>'Brain_only_DLX1'</li><li>'CTX_only_DLX1'</li><li>'CTX_BG_only_DLX1'</li><li>'BG_only_DLX1'</li><li>'Left_only_DLX1'</li><li>'Right_only_DLX1'</li><li>'L_BR_PF_only_DLX1'</li><li>'L_BR_F_only_DLX1'</li><li>'L_BR_O_only_DLX1'</li><li>'L_BR_Ins_only_DLX1'</li><li>'R_BR_F_only_DLX1'</li><li>'R_BR_O_only_DLX1'</li><li>'R_BR_Ins_only_DLX1'</li><li>'IN_only_DLX1'</li><li>'COUP_only_DLX1'</li><li>'COUP_BR_DLX1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_DLX1'\n",
       "\\item 'CTX\\_DLX1'\n",
       "\\item 'BG\\_DLX1'\n",
       "\\item 'THAL\\_DLX1'\n",
       "\\item 'L\\_DLX1'\n",
       "\\item 'R\\_DLX1'\n",
       "\\item 'PF\\_DLX1'\n",
       "\\item 'F\\_DLX1'\n",
       "\\item 'P\\_DLX1'\n",
       "\\item 'O\\_DLX1'\n",
       "\\item 'T\\_DLX1'\n",
       "\\item 'Ins\\_DLX1'\n",
       "\\item 'Put\\_DLX1'\n",
       "\\item 'GP\\_DLX1'\n",
       "\\item 'COUPTF2\\_DLX1'\n",
       "\\item 'Brain\\_only\\_DLX1'\n",
       "\\item 'CTX\\_only\\_DLX1'\n",
       "\\item 'CTX\\_BG\\_only\\_DLX1'\n",
       "\\item 'BG\\_only\\_DLX1'\n",
       "\\item 'Left\\_only\\_DLX1'\n",
       "\\item 'Right\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_F\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_O\\_only\\_DLX1'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_F\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_O\\_only\\_DLX1'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_DLX1'\n",
       "\\item 'IN\\_only\\_DLX1'\n",
       "\\item 'COUP\\_only\\_DLX1'\n",
       "\\item 'COUP\\_BR\\_DLX1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_DLX1'\n",
       "2. 'CTX_DLX1'\n",
       "3. 'BG_DLX1'\n",
       "4. 'THAL_DLX1'\n",
       "5. 'L_DLX1'\n",
       "6. 'R_DLX1'\n",
       "7. 'PF_DLX1'\n",
       "8. 'F_DLX1'\n",
       "9. 'P_DLX1'\n",
       "10. 'O_DLX1'\n",
       "11. 'T_DLX1'\n",
       "12. 'Ins_DLX1'\n",
       "13. 'Put_DLX1'\n",
       "14. 'GP_DLX1'\n",
       "15. 'COUPTF2_DLX1'\n",
       "16. 'Brain_only_DLX1'\n",
       "17. 'CTX_only_DLX1'\n",
       "18. 'CTX_BG_only_DLX1'\n",
       "19. 'BG_only_DLX1'\n",
       "20. 'Left_only_DLX1'\n",
       "21. 'Right_only_DLX1'\n",
       "22. 'L_BR_PF_only_DLX1'\n",
       "23. 'L_BR_F_only_DLX1'\n",
       "24. 'L_BR_O_only_DLX1'\n",
       "25. 'L_BR_Ins_only_DLX1'\n",
       "26. 'R_BR_F_only_DLX1'\n",
       "27. 'R_BR_O_only_DLX1'\n",
       "28. 'R_BR_Ins_only_DLX1'\n",
       "29. 'IN_only_DLX1'\n",
       "30. 'COUP_only_DLX1'\n",
       "31. 'COUP_BR_DLX1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_DLX1\" \"CTX_DLX1\"                    \n",
       " [3] \"BG_DLX1\"                      \"THAL_DLX1\"                   \n",
       " [5] \"L_DLX1\"                       \"R_DLX1\"                      \n",
       " [7] \"PF_DLX1\"                      \"F_DLX1\"                      \n",
       " [9] \"P_DLX1\"                       \"O_DLX1\"                      \n",
       "[11] \"T_DLX1\"                       \"Ins_DLX1\"                    \n",
       "[13] \"Put_DLX1\"                     \"GP_DLX1\"                     \n",
       "[15] \"COUPTF2_DLX1\"                 \"Brain_only_DLX1\"             \n",
       "[17] \"CTX_only_DLX1\"                \"CTX_BG_only_DLX1\"            \n",
       "[19] \"BG_only_DLX1\"                 \"Left_only_DLX1\"              \n",
       "[21] \"Right_only_DLX1\"              \"L_BR_PF_only_DLX1\"           \n",
       "[23] \"L_BR_F_only_DLX1\"             \"L_BR_O_only_DLX1\"            \n",
       "[25] \"L_BR_Ins_only_DLX1\"           \"R_BR_F_only_DLX1\"            \n",
       "[27] \"R_BR_O_only_DLX1\"             \"R_BR_Ins_only_DLX1\"          \n",
       "[29] \"IN_only_DLX1\"                 \"COUP_only_DLX1\"              \n",
       "[31] \"COUP_BR_DLX1\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#DLX1\n",
    "dlxVars = subset(confirmedVars, Sort_main == 1 & Cell_Type == \"DLX1\")\n",
    "varsConfirmedInDLX1_2Sample = subset(annotated, sample_number_presented_DLX1 >= 2) # using __DLX1 columns\n",
    "confirmed2SampleDLXVars = subset(dlxVars, CHROM.POS.REF.ALT %in% varsConfirmedInDLX1_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleDLXVars$MAF), confirmed2SampleDLXVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[, colSums(varsConfirmedInDLX1_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"287_correlation_DLX_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "names(varsConfirmedInDLX1_2Sample %>% select(ends_with(\"_DLX1\")))\n",
    "\n",
    "legendDets = list(\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "DARPP32M.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32M.NeuNP\")),\n",
    "DARPP32P.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32P.NeuNP\")),\n",
    "LHX2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"LHX2\")),\n",
    "OLIG2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"OLIG2\")),\n",
    "PU1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PU1\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUPTF2_ENRICHED = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Enriched\")),\n",
    "COUP_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "COUP_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Brain only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BRAIN_ONLY = varsConfirmedInDLX1_2Sample$Brain_only_DLX1,\n",
    "                                        CTX = varsConfirmedInDLX1_2Sample$CTX_DLX1,\n",
    "                                        BG = varsConfirmedInDLX1_2Sample$BG_DLX1,\n",
    "                                        THAL = varsConfirmedInDLX1_2Sample$THAL_DLX1,\n",
    "                                        PF = varsConfirmedInDLX1_2Sample$PF_DLX1,\n",
    "                                        F = varsConfirmedInDLX1_2Sample$F_DLX1,\n",
    "                                        P = varsConfirmedInDLX1_2Sample$P_DLX1,\n",
    "                                        O = varsConfirmedInDLX1_2Sample$O_DLX1,\n",
    "                                        T = varsConfirmedInDLX1_2Sample$T_DLX1,\n",
    "                                        INS = varsConfirmedInDLX1_2Sample$Ins_DLX1,\n",
    "                                        PUT = varsConfirmedInDLX1_2Sample$Put_DLX1,\n",
    "                                        GP = varsConfirmedInDLX1_2Sample$GP_DLX1,\n",
    "                                        CTX_ONLY = varsConfirmedInDLX1_2Sample$CTX_only_DLX1,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInDLX1_2Sample$CTX_BG_only_DLX1,\n",
    "                                        BG_ONLY = varsConfirmedInDLX1_2Sample$BG_only_DLX1,\n",
    "                                        #L = varsConfirmedInDLX1_2Sample$L_sort,\n",
    "                                        #R = varsConfirmedInDLX1_2Sample$R_sort,\n",
    "                                        LR_ONLY = varsConfirmedInDLX1_2Sample$Left_only_DLX1 - varsConfirmedInDLX1_2Sample$Right_only_DLX1, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInDLX1_2Sample$L_BR_PF_only_DLX1,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInDLX1_2Sample$L_BR_F_only_DLX1,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInDLX1_2Sample$L_BR_O_only_DLX1,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedInDLX1_2Sample$L_BR_Ins_only_DLX1,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInDLX1_2Sample$R_BR_F_only_DLX1,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInDLX1_2Sample$R_BR_O_only_DLX1,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedInDLX1_2Sample$R_BR_Ins_only_DLX1,\n",
    "                                        IN_ONLY = varsConfirmedInDLX1_2Sample$IN_only_DLX1,\n",
    "                                        COUPTF2 = varsConfirmedInDLX1_2Sample$COUPTF2_DLX1,\n",
    "                                        COUP_BR = varsConfirmedInDLX1_2Sample$COUP_BR_DLX1,\n",
    "                                        COUP_ONLY = varsConfirmedInDLX1_2Sample$COUP_only_DLX1,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "dlxMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_DLX_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "\n",
    "dlxMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_DLX_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "\n",
    "dlxMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_TBR1'</li><li>'CTX_TBR1'</li><li>'L_TBR1'</li><li>'R_TBR1'</li><li>'PF_TBR1'</li><li>'F_TBR1'</li><li>'P_TBR1'</li><li>'O_TBR1'</li><li>'T_TBR1'</li><li>'Ins_TBR1'</li><li>'TBR1_TBR1'</li><li>'Brain_only_TBR1'</li><li>'CTX_only_TBR1'</li><li>'CTX_BG_only_TBR1'</li><li>'Left_only_TBR1'</li><li>'Right_only_TBR1'</li><li>'L_BR_PF_only_TBR1'</li><li>'L_BR_O_only_TBR1'</li><li>'R_BR_F_only_TBR1'</li><li>'R_BR_O_only_TBR1'</li><li>'TBR_only_TBR1'</li><li>'TBR_BR_TBR1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_TBR1'\n",
       "\\item 'CTX\\_TBR1'\n",
       "\\item 'L\\_TBR1'\n",
       "\\item 'R\\_TBR1'\n",
       "\\item 'PF\\_TBR1'\n",
       "\\item 'F\\_TBR1'\n",
       "\\item 'P\\_TBR1'\n",
       "\\item 'O\\_TBR1'\n",
       "\\item 'T\\_TBR1'\n",
       "\\item 'Ins\\_TBR1'\n",
       "\\item 'TBR1\\_TBR1'\n",
       "\\item 'Brain\\_only\\_TBR1'\n",
       "\\item 'CTX\\_only\\_TBR1'\n",
       "\\item 'CTX\\_BG\\_only\\_TBR1'\n",
       "\\item 'Left\\_only\\_TBR1'\n",
       "\\item 'Right\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_TBR1'\n",
       "\\item 'L\\_BR\\_O\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_F\\_only\\_TBR1'\n",
       "\\item 'R\\_BR\\_O\\_only\\_TBR1'\n",
       "\\item 'TBR\\_only\\_TBR1'\n",
       "\\item 'TBR\\_BR\\_TBR1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_TBR1'\n",
       "2. 'CTX_TBR1'\n",
       "3. 'L_TBR1'\n",
       "4. 'R_TBR1'\n",
       "5. 'PF_TBR1'\n",
       "6. 'F_TBR1'\n",
       "7. 'P_TBR1'\n",
       "8. 'O_TBR1'\n",
       "9. 'T_TBR1'\n",
       "10. 'Ins_TBR1'\n",
       "11. 'TBR1_TBR1'\n",
       "12. 'Brain_only_TBR1'\n",
       "13. 'CTX_only_TBR1'\n",
       "14. 'CTX_BG_only_TBR1'\n",
       "15. 'Left_only_TBR1'\n",
       "16. 'Right_only_TBR1'\n",
       "17. 'L_BR_PF_only_TBR1'\n",
       "18. 'L_BR_O_only_TBR1'\n",
       "19. 'R_BR_F_only_TBR1'\n",
       "20. 'R_BR_O_only_TBR1'\n",
       "21. 'TBR_only_TBR1'\n",
       "22. 'TBR_BR_TBR1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_TBR1\" \"CTX_TBR1\"                    \n",
       " [3] \"L_TBR1\"                       \"R_TBR1\"                      \n",
       " [5] \"PF_TBR1\"                      \"F_TBR1\"                      \n",
       " [7] \"P_TBR1\"                       \"O_TBR1\"                      \n",
       " [9] \"T_TBR1\"                       \"Ins_TBR1\"                    \n",
       "[11] \"TBR1_TBR1\"                    \"Brain_only_TBR1\"             \n",
       "[13] \"CTX_only_TBR1\"                \"CTX_BG_only_TBR1\"            \n",
       "[15] \"Left_only_TBR1\"               \"Right_only_TBR1\"             \n",
       "[17] \"L_BR_PF_only_TBR1\"            \"L_BR_O_only_TBR1\"            \n",
       "[19] \"R_BR_F_only_TBR1\"             \"R_BR_O_only_TBR1\"            \n",
       "[21] \"TBR_only_TBR1\"                \"TBR_BR_TBR1\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#TBR1\n",
    "tbrVars = subset(confirmedVars, Sort_main == 1 & Cell_Type == \"TBR1\")\n",
    "varsConfirmedInTBR1_2Sample = subset(annotated, sample_number_presented_TBR1 >= 2) # using __TBR1 columns\n",
    "confirmed2SampleTBRVars = subset(tbrVars, CHROM.POS.REF.ALT %in% varsConfirmedInTBR1_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleTBRVars$MAF), confirmed2SampleTBRVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[, colSums(varsConfirmedInTBR1_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "names(varsConfirmedInTBR1_2Sample %>% select(ends_with(\"_TBR1\")))\n",
    "pdf(\"287_correlation_TBR_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "DARPP32M.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32M.NeuNP\")),\n",
    "DARPP32P.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32P.NeuNP\")),\n",
    "LHX2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"LHX2\")),\n",
    "OLIG2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"OLIG2\")),\n",
    "PU1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PU1\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUPTF2_ENRICHED = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Enriched\")),\n",
    "COUP_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "COUP_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Brain only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BRAIN_ONLY = varsConfirmedInTBR1_2Sample$Brain_only_TBR1,\n",
    "                                        CTX = varsConfirmedInTBR1_2Sample$CTX_TBR1,\n",
    "                                        PF = varsConfirmedInTBR1_2Sample$PF_TBR1,\n",
    "                                        F = varsConfirmedInTBR1_2Sample$F_TBR1,\n",
    "                                        P = varsConfirmedInTBR1_2Sample$P_TBR1,\n",
    "                                        O = varsConfirmedInTBR1_2Sample$O_TBR1,\n",
    "                                        T = varsConfirmedInTBR1_2Sample$T_TBR1,\n",
    "                                        INS = varsConfirmedInTBR1_2Sample$Ins_TBR1,\n",
    "                                        CTX_ONLY = varsConfirmedInTBR1_2Sample$CTX_only_TBR1,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInTBR1_2Sample$CTX_BG_only_TBR1,\n",
    "                                        L = varsConfirmedInTBR1_2Sample$L_TBR1,\n",
    "                                        R = varsConfirmedInTBR1_2Sample$R_TBR1,\n",
    "                                        LR_ONLY = varsConfirmedInTBR1_2Sample$Left_only_TBR1 - varsConfirmedInTBR1_2Sample$Right_only_TBR1, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInTBR1_2Sample$L_BR_PF_only_TBR1,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInTBR1_2Sample$L_BR_F_only_TBR1,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInTBR1_2Sample$L_BR_O_only_TBR1,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInTBR1_2Sample$R_BR_F_only_TBR1,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInTBR1_2Sample$R_BR_O_only_TBR1,\n",
    "                                        TBR1 = varsConfirmedInTBR1_2Sample$TBR1_TBR1,\n",
    "                                        TBR_ONLY = varsConfirmedInTBR1_2Sample$TBR_only_TBR1,\n",
    "                                        TBR_BR = varsConfirmedInTBR1_2Sample$TBR_BR_TBR1,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "tbrMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_TBR_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "\n",
    "tbrMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_TBR_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "\n",
    "tbrMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_COUPTF2'</li><li>'CTX_COUPTF2'</li><li>'BG_COUPTF2'</li><li>'THAL_COUPTF2'</li><li>'L_COUPTF2'</li><li>'R_COUPTF2'</li><li>'PF_COUPTF2'</li><li>'F_COUPTF2'</li><li>'P_COUPTF2'</li><li>'O_COUPTF2'</li><li>'T_COUPTF2'</li><li>'Ins_COUPTF2'</li><li>'Put_COUPTF2'</li><li>'GP_COUPTF2'</li><li>'COUPTF2_COUPTF2'</li><li>'Brain_only_COUPTF2'</li><li>'CTX_only_COUPTF2'</li><li>'CTX_BG_only_COUPTF2'</li><li>'BG_only_COUPTF2'</li><li>'Left_only_COUPTF2'</li><li>'Right_only_COUPTF2'</li><li>'L_BR_PF_only_COUPTF2'</li><li>'L_BR_F_only_COUPTF2'</li><li>'L_BR_O_only_COUPTF2'</li><li>'L_BR_Ins_only_COUPTF2'</li><li>'R_BR_F_only_COUPTF2'</li><li>'R_BR_O_only_COUPTF2'</li><li>'R_BR_Ins_only_COUPTF2'</li><li>'IN_only_COUPTF2'</li><li>'COUP_only_COUPTF2'</li><li>'COUP_BR_COUPTF2'</li><li>'COUPTF2_enriched_COUPTF2'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_COUPTF2'\n",
       "\\item 'CTX\\_COUPTF2'\n",
       "\\item 'BG\\_COUPTF2'\n",
       "\\item 'THAL\\_COUPTF2'\n",
       "\\item 'L\\_COUPTF2'\n",
       "\\item 'R\\_COUPTF2'\n",
       "\\item 'PF\\_COUPTF2'\n",
       "\\item 'F\\_COUPTF2'\n",
       "\\item 'P\\_COUPTF2'\n",
       "\\item 'O\\_COUPTF2'\n",
       "\\item 'T\\_COUPTF2'\n",
       "\\item 'Ins\\_COUPTF2'\n",
       "\\item 'Put\\_COUPTF2'\n",
       "\\item 'GP\\_COUPTF2'\n",
       "\\item 'COUPTF2\\_COUPTF2'\n",
       "\\item 'Brain\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_BG\\_only\\_COUPTF2'\n",
       "\\item 'BG\\_only\\_COUPTF2'\n",
       "\\item 'Left\\_only\\_COUPTF2'\n",
       "\\item 'Right\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_PF\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_F\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_O\\_only\\_COUPTF2'\n",
       "\\item 'L\\_BR\\_Ins\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_F\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_O\\_only\\_COUPTF2'\n",
       "\\item 'R\\_BR\\_Ins\\_only\\_COUPTF2'\n",
       "\\item 'IN\\_only\\_COUPTF2'\n",
       "\\item 'COUP\\_only\\_COUPTF2'\n",
       "\\item 'COUP\\_BR\\_COUPTF2'\n",
       "\\item 'COUPTF2\\_enriched\\_COUPTF2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_COUPTF2'\n",
       "2. 'CTX_COUPTF2'\n",
       "3. 'BG_COUPTF2'\n",
       "4. 'THAL_COUPTF2'\n",
       "5. 'L_COUPTF2'\n",
       "6. 'R_COUPTF2'\n",
       "7. 'PF_COUPTF2'\n",
       "8. 'F_COUPTF2'\n",
       "9. 'P_COUPTF2'\n",
       "10. 'O_COUPTF2'\n",
       "11. 'T_COUPTF2'\n",
       "12. 'Ins_COUPTF2'\n",
       "13. 'Put_COUPTF2'\n",
       "14. 'GP_COUPTF2'\n",
       "15. 'COUPTF2_COUPTF2'\n",
       "16. 'Brain_only_COUPTF2'\n",
       "17. 'CTX_only_COUPTF2'\n",
       "18. 'CTX_BG_only_COUPTF2'\n",
       "19. 'BG_only_COUPTF2'\n",
       "20. 'Left_only_COUPTF2'\n",
       "21. 'Right_only_COUPTF2'\n",
       "22. 'L_BR_PF_only_COUPTF2'\n",
       "23. 'L_BR_F_only_COUPTF2'\n",
       "24. 'L_BR_O_only_COUPTF2'\n",
       "25. 'L_BR_Ins_only_COUPTF2'\n",
       "26. 'R_BR_F_only_COUPTF2'\n",
       "27. 'R_BR_O_only_COUPTF2'\n",
       "28. 'R_BR_Ins_only_COUPTF2'\n",
       "29. 'IN_only_COUPTF2'\n",
       "30. 'COUP_only_COUPTF2'\n",
       "31. 'COUP_BR_COUPTF2'\n",
       "32. 'COUPTF2_enriched_COUPTF2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_COUPTF2\" \"CTX_COUPTF2\"                    \n",
       " [3] \"BG_COUPTF2\"                      \"THAL_COUPTF2\"                   \n",
       " [5] \"L_COUPTF2\"                       \"R_COUPTF2\"                      \n",
       " [7] \"PF_COUPTF2\"                      \"F_COUPTF2\"                      \n",
       " [9] \"P_COUPTF2\"                       \"O_COUPTF2\"                      \n",
       "[11] \"T_COUPTF2\"                       \"Ins_COUPTF2\"                    \n",
       "[13] \"Put_COUPTF2\"                     \"GP_COUPTF2\"                     \n",
       "[15] \"COUPTF2_COUPTF2\"                 \"Brain_only_COUPTF2\"             \n",
       "[17] \"CTX_only_COUPTF2\"                \"CTX_BG_only_COUPTF2\"            \n",
       "[19] \"BG_only_COUPTF2\"                 \"Left_only_COUPTF2\"              \n",
       "[21] \"Right_only_COUPTF2\"              \"L_BR_PF_only_COUPTF2\"           \n",
       "[23] \"L_BR_F_only_COUPTF2\"             \"L_BR_O_only_COUPTF2\"            \n",
       "[25] \"L_BR_Ins_only_COUPTF2\"           \"R_BR_F_only_COUPTF2\"            \n",
       "[27] \"R_BR_O_only_COUPTF2\"             \"R_BR_Ins_only_COUPTF2\"          \n",
       "[29] \"IN_only_COUPTF2\"                 \"COUP_only_COUPTF2\"              \n",
       "[31] \"COUP_BR_COUPTF2\"                 \"COUPTF2_enriched_COUPTF2\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#COUPTF2\n",
    "coupVars = subset(confirmedVars, Sort_main == 1 & Cell_Type == \"COUPTF2\")\n",
    "varsConfirmedInCOUPTF2_2Sample = subset(annotated, sample_number_presented_COUPTF2 >= 2) # using __COUPTF2 columns\n",
    "confirmed2SampleCOUPVars = subset(coupVars, CHROM.POS.REF.ALT %in% varsConfirmedInCOUPTF2_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleCOUPVars$MAF), confirmed2SampleCOUPVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[, colSums(varsConfirmedInCOUPTF2_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "names(varsConfirmedInCOUPTF2_2Sample %>% select(ends_with(\"_COUPTF2\")))\n",
    "pdf(\"287_correlation_COUP_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "DARPP32M.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32M.NeuNP\")),\n",
    "DARPP32P.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32P.NeuNP\")),\n",
    "LHX2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"LHX2\")),\n",
    "OLIG2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"OLIG2\")),\n",
    "PU1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PU1\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUPTF2_ENRICHED = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Enriched\")),\n",
    "COUP_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "COUP_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Brain only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BRAIN_ONLY = varsConfirmedInCOUPTF2_2Sample$Brain_only_COUPTF2,\n",
    "                                        CTX = varsConfirmedInCOUPTF2_2Sample$CTX_COUPTF2,\n",
    "                                        BG = varsConfirmedInCOUPTF2_2Sample$BG_COUPTF2,\n",
    "                                        THAL = varsConfirmedInCOUPTF2_2Sample$THAL_COUPTF2,\n",
    "                                        PF = varsConfirmedInCOUPTF2_2Sample$PF_COUPTF2,\n",
    "                                        F = varsConfirmedInCOUPTF2_2Sample$F_COUPTF2,\n",
    "                                        P = varsConfirmedInCOUPTF2_2Sample$P_COUPTF2,\n",
    "                                        O = varsConfirmedInCOUPTF2_2Sample$O_COUPTF2,\n",
    "                                        T = varsConfirmedInCOUPTF2_2Sample$T_COUPTF2,\n",
    "                                        INS = varsConfirmedInCOUPTF2_2Sample$Ins_COUPTF2,\n",
    "                                        PUT = varsConfirmedInCOUPTF2_2Sample$Put_COUPTF2,\n",
    "                                        GP = varsConfirmedInCOUPTF2_2Sample$GP_COUPTF2,\n",
    "                                        CTX_ONLY = varsConfirmedInCOUPTF2_2Sample$CTX_only_COUPTF2,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInCOUPTF2_2Sample$CTX_BG_only_COUPTF2,\n",
    "                                        BG_ONLY = varsConfirmedInCOUPTF2_2Sample$BG_only_COUPTF2,\n",
    "                                        L = varsConfirmedInCOUPTF2_2Sample$L_COUPTF2,\n",
    "                                        R = varsConfirmedInCOUPTF2_2Sample$R_COUPTF2,\n",
    "                                        LR_ONLY = varsConfirmedInCOUPTF2_2Sample$Left_only_COUPTF2 - varsConfirmedInCOUPTF2_2Sample$Right_only_COUPTF2, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_PF_only_COUPTF2,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_F_only_COUPTF2,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_O_only_COUPTF2,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_Ins_only_COUPTF2,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_F_only_COUPTF2,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_O_only_COUPTF2,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_Ins_only_COUPTF2,\n",
    "                                        IN_ONLY = varsConfirmedInCOUPTF2_2Sample$IN_only_COUPTF2,\n",
    "                                        COUPTF2 = varsConfirmedInCOUPTF2_2Sample$COUPTF2_COUPTF2,\n",
    "                                        COUP_ONLY = varsConfirmedInCOUPTF2_2Sample$COUP_only_COUPTF2,\n",
    "                                        COUP_BR = varsConfirmedInCOUPTF2_2Sample$COUP_BR_COUPTF2,\n",
    "                                        COUPTF2_ENRICHED = varsConfirmedInCOUPTF2_2Sample$COUPTF2_enriched_COUPTF2,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "coupMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_COUP_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "coupMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_COUP_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "coupMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_CBH'</li><li>'CTX_CBH'</li><li>'BG_CBH'</li><li>'HIP_CBH'</li><li>'PF_CBH'</li><li>'F_CBH'</li><li>'P_CBH'</li><li>'O_CBH'</li><li>'T_CBH'</li><li>'Ins_CBH'</li><li>'Cau_CBH'</li><li>'Put_CBH'</li><li>'GP_CBH'</li><li>'CA1_CBH'</li><li>'CA3_CBH'</li><li>'DG_CBH'</li><li>'DLX1_CBH'</li><li>'TBR1_CBH'</li><li>'Brain_only_CBH'</li><li>'CTX_BG_shared_CBH'</li><li>'BG_HIP_shared_CBH'</li><li>'CTX_HIP_shared_CBH'</li><li>'CTX_only_CBH'</li><li>'BG_only_CBH'</li><li>'GP_only_CBH'</li><li>'HIP_only_CBH'</li><li>'IN_only_CBH'</li><li>'TBR_only_CBH'</li><li>'TBR_BR_CBH'</li><li>'DLX_BR_CBH'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_CBH'\n",
       "\\item 'CTX\\_CBH'\n",
       "\\item 'BG\\_CBH'\n",
       "\\item 'HIP\\_CBH'\n",
       "\\item 'PF\\_CBH'\n",
       "\\item 'F\\_CBH'\n",
       "\\item 'P\\_CBH'\n",
       "\\item 'O\\_CBH'\n",
       "\\item 'T\\_CBH'\n",
       "\\item 'Ins\\_CBH'\n",
       "\\item 'Cau\\_CBH'\n",
       "\\item 'Put\\_CBH'\n",
       "\\item 'GP\\_CBH'\n",
       "\\item 'CA1\\_CBH'\n",
       "\\item 'CA3\\_CBH'\n",
       "\\item 'DG\\_CBH'\n",
       "\\item 'DLX1\\_CBH'\n",
       "\\item 'TBR1\\_CBH'\n",
       "\\item 'Brain\\_only\\_CBH'\n",
       "\\item 'CTX\\_BG\\_shared\\_CBH'\n",
       "\\item 'BG\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_only\\_CBH'\n",
       "\\item 'BG\\_only\\_CBH'\n",
       "\\item 'GP\\_only\\_CBH'\n",
       "\\item 'HIP\\_only\\_CBH'\n",
       "\\item 'IN\\_only\\_CBH'\n",
       "\\item 'TBR\\_only\\_CBH'\n",
       "\\item 'TBR\\_BR\\_CBH'\n",
       "\\item 'DLX\\_BR\\_CBH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_CBH'\n",
       "2. 'CTX_CBH'\n",
       "3. 'BG_CBH'\n",
       "4. 'HIP_CBH'\n",
       "5. 'PF_CBH'\n",
       "6. 'F_CBH'\n",
       "7. 'P_CBH'\n",
       "8. 'O_CBH'\n",
       "9. 'T_CBH'\n",
       "10. 'Ins_CBH'\n",
       "11. 'Cau_CBH'\n",
       "12. 'Put_CBH'\n",
       "13. 'GP_CBH'\n",
       "14. 'CA1_CBH'\n",
       "15. 'CA3_CBH'\n",
       "16. 'DG_CBH'\n",
       "17. 'DLX1_CBH'\n",
       "18. 'TBR1_CBH'\n",
       "19. 'Brain_only_CBH'\n",
       "20. 'CTX_BG_shared_CBH'\n",
       "21. 'BG_HIP_shared_CBH'\n",
       "22. 'CTX_HIP_shared_CBH'\n",
       "23. 'CTX_only_CBH'\n",
       "24. 'BG_only_CBH'\n",
       "25. 'GP_only_CBH'\n",
       "26. 'HIP_only_CBH'\n",
       "27. 'IN_only_CBH'\n",
       "28. 'TBR_only_CBH'\n",
       "29. 'TBR_BR_CBH'\n",
       "30. 'DLX_BR_CBH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_CBH\" \"CTX_CBH\"                    \n",
       " [3] \"BG_CBH\"                      \"HIP_CBH\"                    \n",
       " [5] \"PF_CBH\"                      \"F_CBH\"                      \n",
       " [7] \"P_CBH\"                       \"O_CBH\"                      \n",
       " [9] \"T_CBH\"                       \"Ins_CBH\"                    \n",
       "[11] \"Cau_CBH\"                     \"Put_CBH\"                    \n",
       "[13] \"GP_CBH\"                      \"CA1_CBH\"                    \n",
       "[15] \"CA3_CBH\"                     \"DG_CBH\"                     \n",
       "[17] \"DLX1_CBH\"                    \"TBR1_CBH\"                   \n",
       "[19] \"Brain_only_CBH\"              \"CTX_BG_shared_CBH\"          \n",
       "[21] \"BG_HIP_shared_CBH\"           \"CTX_HIP_shared_CBH\"         \n",
       "[23] \"CTX_only_CBH\"                \"BG_only_CBH\"                \n",
       "[25] \"GP_only_CBH\"                 \"HIP_only_CBH\"               \n",
       "[27] \"IN_only_CBH\"                 \"TBR_only_CBH\"               \n",
       "[29] \"TBR_BR_CBH\"                  \"DLX_BR_CBH\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#CBH\n",
    "# CTX  |  BG  |  HIP |\n",
    "# TBR1 | DLX1 | TBR1 |\n",
    "\n",
    "cbhVars = subset(confirmedVars, CTX_BG_HIP_relationship == 1)\n",
    "varsConfirmedIn2Sample = subset(annotated, sample_number_presented_CBH >= 2) # using CBH columns\n",
    "#varsConfirmedIn2Sample = subset(varsConfirmedIn2Sample, (CTX_CBH == 1 & TBR_only_CBH == 1) | (BG_CBH == 1 & DLX1_CBH == 1) | (HIP_CBH == 1 & TBR_only_CBH == 1) ) # 5/23/23 - __CBH already accounts for cell type\n",
    "varsConfirmedIn2Sample = subset(varsConfirmedIn2Sample, (CTX_CBH == 1) | (BG_CBH == 1) | (HIP_CBH == 1) )\n",
    "\n",
    "confirmedCBH_vars = subset(cbhVars, CHROM.POS.REF.ALT %in% varsConfirmedIn2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmedCBH_vars$MAF), confirmedCBH_vars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[, colSums(varsConfirmedIn2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"287_correlation_CBH_Pearson.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "DARPP32M.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32M.NeuNP\")),\n",
    "DARPP32P.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32P.NeuNP\")),\n",
    "LHX2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"LHX2\")),\n",
    "OLIG2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"OLIG2\")),\n",
    "PU1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PU1\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUPTF2_ENRICHED = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Enriched\")),\n",
    "COUP_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "COUP_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Brain only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "names(varsConfirmedIn2Sample %>% select(ends_with(\"_CBH\")))\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.5, \"cm\"), #previously 0.2\n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 6), #previously 4 fontsize\n",
    "                                        #BRAIN_ONLY = varsConfirmedIn2Sample$Brain_only_CBH,\n",
    "                                        #INS = varsConfirmedIn2Sample$Ins_CBH,\n",
    "                                        #CAU = varsConfirmedIn2Sample$Cau_CBH,\n",
    "                                        #PUT = varsConfirmedIn2Sample$Put_CBH,\n",
    "                                        #GP = varsConfirmedIn2Sample$GP_CBH,\n",
    "                                        #GP_ONLY = varsConfirmedIn2Sample$GP_only_CBH,\n",
    "                                        #PF = varsConfirmedIn2Sample$PF_CBH,\n",
    "                                        #F = varsConfirmedIn2Sample$F_CBH,\n",
    "                                        #P = varsConfirmedIn2Sample$P_CBH,\n",
    "                                        #O = varsConfirmedIn2Sample$O_CBH,\n",
    "                                        #T = varsConfirmedIn2Sample$T_CBH,\n",
    "                                        #CA1 = varsConfirmedIn2Sample$CA1_CBH,\n",
    "                                        #CA3 = varsConfirmedIn2Sample$CA3_CBH,\n",
    "                                        #DG = varsConfirmedIn2Sample$DG_CBH,\n",
    "                                        IN_ONLY = varsConfirmedIn2Sample$IN_only_CBH,\n",
    "                                        #DLX1 = varsConfirmedIn2Sample$DLX1_CBH,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR_CBH,\n",
    "                                        #TBR1 = varsConfirmedIn2Sample$TBR1_CBH,\n",
    "                                        #TBR_ONLY = varsConfirmedIn2Sample$TBR_only_CBH,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR_CBH,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX_CBH,\n",
    "                                        BG = varsConfirmedIn2Sample$BG_CBH,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP_CBH,\n",
    "                                        CTX_ONLY = varsConfirmedIn2Sample$CTX_only_CBH,\n",
    "                                        BG_ONLY = varsConfirmedIn2Sample$BG_only_CBH,\n",
    "                                        HIP_ONLY = varsConfirmedIn2Sample$HIP_only_CBH,\n",
    "                                        CTX_BG = varsConfirmedIn2Sample$CTX_BG_shared_CBH,\n",
    "                                        BG_HIP = varsConfirmedIn2Sample$BG_HIP_shared_CBH,\n",
    "                                        CTX_HIP = varsConfirmedIn2Sample$CTX_HIP_shared_CBH,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "cbhMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_CBH_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "cbhMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"287_correlation_CBH_Spearman.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "cbhMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
