# Human_Interneuron
This repository contains codes and pipelines for the analysis for human interneuron migration study and clonal distribution project.

-----------------------------------

### 1. Pipelines for the process of whole-genome sequencing data

#### 1.1 Pipelines for WGS data process and quality control

#### 1.2 Codes for the population origin analysis

#### 1.3 Pipelines for mosaic SNV/indel calling and variant annotations

-----------------------------------

### 2. Pipelines for the process of Targeted Amplicon Sequencing (TAS)

#### 2.1 Pipelines for TAS data alignment and processing

#### 2.2 Pipelines for AF quantification and variant annotations

-----------------------------------

### 3. Identification the cell types by using whole-genome bisulfite sequencing analysis

#### 3.1 [Pipelines](https://github.com/shishenyxx/Human_Interneuron/tree/main/Methylome) for methylome analysis using BisMark

#### 3.2 [Statical analysis](https://github.com/shishenyxx/Human_Interneuron/blob/main/Methylome/plots/Human_Interneuron_Methylome_Plots.ipynb) combining published standards to identify the cell types

-----------------------------------

### 4. Computational and statistical analyses for interneuron clonal distribution patterns

#### 4.1 Pipelines for mosaic variant determination, annotations, and plotting

#### 4.2 Pipelines for statistically analysis, and the related plotting

-----------------------------------

### 5. Contact:

:email: Changuk Chung: [chchung@health.ucsd.edu](mailto:chchung@health.ucsd.edu)

:email: Xiaoxu Yang: [xiy010@health.ucsd.edu](mailto:xiy010@health.ucsd.edu), [yangxiaoxu-shishen@hotmail.com](mailto:yangxiaoxu-shishen@hotmail.com)

:email: Joseph Gleeson: [jogleeson@health.ucsd.edu](mailto:jogleeson@health.ucsd.edu), or the Gleeson lab [gleesonlab@health.ucsd.edu](gleesonlab@health.ucsd.edu)

-----------------------------------

### 6. Cite the data and codes:
