{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d379d859",
   "metadata": {},
   "source": [
    "# Patient ID 7669"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "67974f1f-da3c-4295-82a9-548083681668",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'gplots'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'reshape2'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:tidyr':\n",
      "\n",
      "    smiths\n",
      "\n",
      "\n",
      "corrplot 0.92 loaded\n",
      "\n",
      "Loading required package: plotly\n",
      "\n",
      "\n",
      "Attaching package: 'plotly'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:ggplot2':\n",
      "\n",
      "    last_plot\n",
      "\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "The following object is masked from 'package:graphics':\n",
      "\n",
      "    layout\n",
      "\n",
      "\n",
      "Loading required package: viridis\n",
      "\n",
      "Loading required package: viridisLite\n",
      "\n",
      "\n",
      "======================\n",
      "Welcome to heatmaply version 1.4.2\n",
      "\n",
      "Type citation('heatmaply') for how to cite the package.\n",
      "Type ?heatmaply for the main documentation.\n",
      "\n",
      "The github page is: https://github.com/talgalili/heatmaply/\n",
      "Please submit your suggestions and bug-reports at: https://github.com/talgalili/heatmaply/issues\n",
      "You may ask questions at stackoverflow, use the r and heatmaply tags: \n",
      "\t https://stackoverflow.com/questions/tagged/heatmaply\n",
      "======================\n",
      "\n",
      "\n",
      "Loading required package: usethis\n",
      "\n",
      "========================================\n",
      "ComplexHeatmap version 2.15.3\n",
      "Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/\n",
      "Github page: https://github.com/jokergoo/ComplexHeatmap\n",
      "Documentation: http://jokergoo.github.io/ComplexHeatmap-reference\n",
      "\n",
      "If you use it in published research, please cite either one:\n",
      "- Gu, Z. Complex Heatmap Visualization. iMeta 2022.\n",
      "- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional \n",
      "    genomic data. Bioinformatics 2016.\n",
      "\n",
      "\n",
      "The new InteractiveComplexHeatmap package can directly export static \n",
      "complex heatmaps into an interactive Shiny app with zero effort. Have a try!\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(ComplexHeatmap))\n",
      "========================================\n",
      "! pheatmap() has been masked by ComplexHeatmap::pheatmap(). Most of the arguments\n",
      "   in the original pheatmap() are identically supported in the new function. You \n",
      "   can still use the original function by explicitly calling pheatmap::pheatmap().\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'ComplexHeatmap'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:plotly':\n",
      "\n",
      "    add_heatmap\n",
      "\n",
      "\n",
      "The following object is masked from 'package:pheatmap':\n",
      "\n",
      "    pheatmap\n",
      "\n",
      "\n",
      "========================================\n",
      "circlize version 0.4.15\n",
      "CRAN page: https://cran.r-project.org/package=circlize\n",
      "Github page: https://github.com/jokergoo/circlize\n",
      "Documentation: https://jokergoo.github.io/circlize_book/book/\n",
      "\n",
      "If you use it in published research, please cite:\n",
      "Gu, Z. circlize implements and enhances circular visualization\n",
      "  in R. Bioinformatics 2014.\n",
      "\n",
      "This message can be suppressed by:\n",
      "  suppressPackageStartupMessages(library(circlize))\n",
      "========================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# original path from server \"/projects/ps-gleesonlab8/User/chchung/Interneuron/20221223_7669_Ampliseq_QC/20230101_QC/final_summary_anno3.txt\"\n",
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggcorrplot)\n",
    "library(gplots)\n",
    "library(reshape2)\n",
    "library(corrplot)\n",
    "library(grid)\n",
    "library(pheatmap)\n",
    "library(ggdendro)\n",
    "library(heatmaply)\n",
    "library(devtools) \n",
    "library(ComplexHeatmap) #ComplexHeatmap cite \"Gu, Z. (2022) Complex Heatmap Visualization, iMeta. DOI: 10.1002/imt2.43.\"\n",
    "library(circlize)\n",
    "library(dendsort)\n",
    "\n",
    "filterdata <- read.table(\"/Users/rahulnedunuri/Documents/GleesonLab/Mosaic Variant Projects/final_summary_anno3.txt\" ,header=T,sep=\"\\t\")\n",
    "annotated <- read.csv(\"/Users/rahulnedunuri/Documents/GleesonLab/Mosaic Variant Projects/780_variant_annotation_COUPTF2_TBR1_DLX1_CBH.csv\" ,header=T,sep=\",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b49f9d5c-697e-4953-8802-5fac3d1be060",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ID'</li><li>'CHROM'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'CHROM.POS.REF.ALT'</li><li>'ANNO'</li><li>'GENE'</li><li>'GNOMAD_FREQ'</li><li>'REPEAT_MASKER'</li><li>'SEGDUP'</li><li>'HOMOPOLYMER'</li><li>'REF_SEQ'</li><li>'DINUCLEOTIDE'</li><li>'NEAR_INDEL'</li><li>'UCSC_RPMSK'</li><li>'REF_COUNT'</li><li>'ALT_COUNT'</li><li>'MAF'</li><li>'LOWER_CI'</li><li>'UPPER_CI'</li><li>'CI_IS_GREATER'</li><li>'NORMAL_REF_COUNT'</li><li>'NORMAL_ALT_COUNT'</li><li>'NORMAL_MAF'</li><li>'NORMAL_LOWER_CI'</li><li>'NORMAL_UPPER_CI'</li><li>'NORMAL_CI_IS_GREATER'</li><li>'Mosaic'</li><li>'Mosaic_shared_btw_WGS_Amp'</li><li>'Hemisphere'</li><li>'Region'</li><li>'Punch'</li><li>'Cell_Type'</li><li>'Location'</li><li>'AntPost'</li><li>'DorVen'</li><li>'Organ'</li><li>'Note'</li><li>'Bulk_main'</li><li>'Sort_main'</li><li>'CTX_BG_HIP_relationship'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ID'\n",
       "\\item 'CHROM'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'CHROM.POS.REF.ALT'\n",
       "\\item 'ANNO'\n",
       "\\item 'GENE'\n",
       "\\item 'GNOMAD\\_FREQ'\n",
       "\\item 'REPEAT\\_MASKER'\n",
       "\\item 'SEGDUP'\n",
       "\\item 'HOMOPOLYMER'\n",
       "\\item 'REF\\_SEQ'\n",
       "\\item 'DINUCLEOTIDE'\n",
       "\\item 'NEAR\\_INDEL'\n",
       "\\item 'UCSC\\_RPMSK'\n",
       "\\item 'REF\\_COUNT'\n",
       "\\item 'ALT\\_COUNT'\n",
       "\\item 'MAF'\n",
       "\\item 'LOWER\\_CI'\n",
       "\\item 'UPPER\\_CI'\n",
       "\\item 'CI\\_IS\\_GREATER'\n",
       "\\item 'NORMAL\\_REF\\_COUNT'\n",
       "\\item 'NORMAL\\_ALT\\_COUNT'\n",
       "\\item 'NORMAL\\_MAF'\n",
       "\\item 'NORMAL\\_LOWER\\_CI'\n",
       "\\item 'NORMAL\\_UPPER\\_CI'\n",
       "\\item 'NORMAL\\_CI\\_IS\\_GREATER'\n",
       "\\item 'Mosaic'\n",
       "\\item 'Mosaic\\_shared\\_btw\\_WGS\\_Amp'\n",
       "\\item 'Hemisphere'\n",
       "\\item 'Region'\n",
       "\\item 'Punch'\n",
       "\\item 'Cell\\_Type'\n",
       "\\item 'Location'\n",
       "\\item 'AntPost'\n",
       "\\item 'DorVen'\n",
       "\\item 'Organ'\n",
       "\\item 'Note'\n",
       "\\item 'Bulk\\_main'\n",
       "\\item 'Sort\\_main'\n",
       "\\item 'CTX\\_BG\\_HIP\\_relationship'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ID'\n",
       "2. 'CHROM'\n",
       "3. 'POS'\n",
       "4. 'REF'\n",
       "5. 'ALT'\n",
       "6. 'CHROM.POS.REF.ALT'\n",
       "7. 'ANNO'\n",
       "8. 'GENE'\n",
       "9. 'GNOMAD_FREQ'\n",
       "10. 'REPEAT_MASKER'\n",
       "11. 'SEGDUP'\n",
       "12. 'HOMOPOLYMER'\n",
       "13. 'REF_SEQ'\n",
       "14. 'DINUCLEOTIDE'\n",
       "15. 'NEAR_INDEL'\n",
       "16. 'UCSC_RPMSK'\n",
       "17. 'REF_COUNT'\n",
       "18. 'ALT_COUNT'\n",
       "19. 'MAF'\n",
       "20. 'LOWER_CI'\n",
       "21. 'UPPER_CI'\n",
       "22. 'CI_IS_GREATER'\n",
       "23. 'NORMAL_REF_COUNT'\n",
       "24. 'NORMAL_ALT_COUNT'\n",
       "25. 'NORMAL_MAF'\n",
       "26. 'NORMAL_LOWER_CI'\n",
       "27. 'NORMAL_UPPER_CI'\n",
       "28. 'NORMAL_CI_IS_GREATER'\n",
       "29. 'Mosaic'\n",
       "30. 'Mosaic_shared_btw_WGS_Amp'\n",
       "31. 'Hemisphere'\n",
       "32. 'Region'\n",
       "33. 'Punch'\n",
       "34. 'Cell_Type'\n",
       "35. 'Location'\n",
       "36. 'AntPost'\n",
       "37. 'DorVen'\n",
       "38. 'Organ'\n",
       "39. 'Note'\n",
       "40. 'Bulk_main'\n",
       "41. 'Sort_main'\n",
       "42. 'CTX_BG_HIP_relationship'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ID\"                        \"CHROM\"                    \n",
       " [3] \"POS\"                       \"REF\"                      \n",
       " [5] \"ALT\"                       \"CHROM.POS.REF.ALT\"        \n",
       " [7] \"ANNO\"                      \"GENE\"                     \n",
       " [9] \"GNOMAD_FREQ\"               \"REPEAT_MASKER\"            \n",
       "[11] \"SEGDUP\"                    \"HOMOPOLYMER\"              \n",
       "[13] \"REF_SEQ\"                   \"DINUCLEOTIDE\"             \n",
       "[15] \"NEAR_INDEL\"                \"UCSC_RPMSK\"               \n",
       "[17] \"REF_COUNT\"                 \"ALT_COUNT\"                \n",
       "[19] \"MAF\"                       \"LOWER_CI\"                 \n",
       "[21] \"UPPER_CI\"                  \"CI_IS_GREATER\"            \n",
       "[23] \"NORMAL_REF_COUNT\"          \"NORMAL_ALT_COUNT\"         \n",
       "[25] \"NORMAL_MAF\"                \"NORMAL_LOWER_CI\"          \n",
       "[27] \"NORMAL_UPPER_CI\"           \"NORMAL_CI_IS_GREATER\"     \n",
       "[29] \"Mosaic\"                    \"Mosaic_shared_btw_WGS_Amp\"\n",
       "[31] \"Hemisphere\"                \"Region\"                   \n",
       "[33] \"Punch\"                     \"Cell_Type\"                \n",
       "[35] \"Location\"                  \"AntPost\"                  \n",
       "[37] \"DorVen\"                    \"Organ\"                    \n",
       "[39] \"Note\"                      \"Bulk_main\"                \n",
       "[41] \"Sort_main\"                 \"CTX_BG_HIP_relationship\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'X'</li><li>'variant_list'</li><li>'sample_number_presented'</li><li>'HEART'</li><li>'ADRENAL'</li><li>'CB'</li><li>'CTX'</li><li>'LIVER'</li><li>'KIDNEY'</li><li>'POA'</li><li>'BG'</li><li>'THAL'</li><li>'HIP'</li><li>'SKIN'</li><li>'OLF'</li><li>'L'</li><li>'R'</li><li>'I'</li><li>'F'</li><li>'P'</li><li>'O'</li><li>'T'</li><li>'Cau'</li><li>'Put'</li><li>'Thal'</li><li>'GP'</li><li>'Hip'</li><li>'EC'</li><li>'Olf'</li><li>'CC'</li><li>'mO'</li><li>'Cb'</li><li>'DLX1'</li><li>'TBR1'</li><li>'COUPTF2'</li><li>'Bulk'</li><li>'NeuN'</li><li>'Brain_only'</li><li>'CTX_only'</li><li>'BG_only'</li><li>'THAL_only'</li><li>'CTX_BG_only'</li><li>'Left_only'</li><li>'Right_only'</li><li>'CTX_BG_shared'</li><li>'CTX_THAL_shared'</li><li>'sample_number_presented_COUPTF2'</li><li>'CTX_COUPTF2'</li><li>'THAL_COUPTF2'</li><li>'HIP_COUPTF2'</li><li>'L_COUPTF2'</li><li>'R_COUPTF2'</li><li>'I_COUPTF2'</li><li>'F_COUPTF2'</li><li>'P_COUPTF2'</li><li>'O_COUPTF2'</li><li>'T_COUPTF2'</li><li>'Thal_COUPTF2'</li><li>'Hip_COUPTF2'</li><li>'CTX_only_COUPTF2'</li><li>'HIP_only_COUPTF2'</li><li>'THAL_only_COUPTF2'</li><li>'CTX_THAL_shared_COUPTF2'</li><li>'CTX_HIP_shared_COUPTF2'</li><li>'HIP_THAL_shared_COUPTF2'</li><li>'Left_only_COUPTF2'</li><li>'Right_only_COUPTF2'</li><li>'sample_number_presented_TBR1'</li><li>'CTX_TBR1'</li><li>'HIP_TBR1'</li><li>'L_TBR1'</li><li>'R_TBR1'</li><li>'I_TBR1'</li><li>'F_TBR1'</li><li>'P_TBR1'</li><li>'O_TBR1'</li><li>'T_TBR1'</li><li>'Hip_TBR1'</li><li>'CTX_only_TBR1'</li><li>'HIP_only_TBR1'</li><li>'Left_only_TBR1'</li><li>'Right_only_TBR1'</li><li>'sample_number_presented_DLX1'</li><li>'CTX_DLX1'</li><li>'BG_DLX1'</li><li>'THAL_DLX1'</li><li>'HIP_DLX1'</li><li>'L_DLX1'</li><li>'R_DLX1'</li><li>'I_DLX1'</li><li>'F_DLX1'</li><li>'P_DLX1'</li><li>'O_DLX1'</li><li>'T_DLX1'</li><li>'Cau_DLX1'</li><li>'GP_DLX1'</li><li>'CTX_only_DLX1'</li><li>'HIP_only_DLX1'</li><li>'THAL_only_DLX1'</li><li>'BG_only_DLX1'</li><li>'CTX_HIP_shared_DLX1'</li><li>'CTX_THAL_shared_DLX1'</li><li>'CTX_BG_shared_DLX1'</li><li>'Left_only_DLX1'</li><li>'Right_only_DLX1'</li><li>'sample_number_presented_CBH'</li><li>'CTX_CBH'</li><li>'BG_CBH'</li><li>'HIP_CBH'</li><li>'I_CBH'</li><li>'F_CBH'</li><li>'P_CBH'</li><li>'O_CBH'</li><li>'mO_CBH'</li><li>'T_CBH'</li><li>'Cau_CBH'</li><li>'Put_CBH'</li><li>'GP_CBH'</li><li>'CTX_only_CBH'</li><li>'HIP_only_CBH'</li><li>'BG_only_CBH'</li><li>'GP_only_CBH'</li><li>'CTX_HIP_shared_CBH'</li><li>'CTX_BG_shared_CBH'</li><li>'BG_HIP_shared_CBH'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'X'\n",
       "\\item 'variant\\_list'\n",
       "\\item 'sample\\_number\\_presented'\n",
       "\\item 'HEART'\n",
       "\\item 'ADRENAL'\n",
       "\\item 'CB'\n",
       "\\item 'CTX'\n",
       "\\item 'LIVER'\n",
       "\\item 'KIDNEY'\n",
       "\\item 'POA'\n",
       "\\item 'BG'\n",
       "\\item 'THAL'\n",
       "\\item 'HIP'\n",
       "\\item 'SKIN'\n",
       "\\item 'OLF'\n",
       "\\item 'L'\n",
       "\\item 'R'\n",
       "\\item 'I'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'O'\n",
       "\\item 'T'\n",
       "\\item 'Cau'\n",
       "\\item 'Put'\n",
       "\\item 'Thal'\n",
       "\\item 'GP'\n",
       "\\item 'Hip'\n",
       "\\item 'EC'\n",
       "\\item 'Olf'\n",
       "\\item 'CC'\n",
       "\\item 'mO'\n",
       "\\item 'Cb'\n",
       "\\item 'DLX1'\n",
       "\\item 'TBR1'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'Bulk'\n",
       "\\item 'NeuN'\n",
       "\\item 'Brain\\_only'\n",
       "\\item 'CTX\\_only'\n",
       "\\item 'BG\\_only'\n",
       "\\item 'THAL\\_only'\n",
       "\\item 'CTX\\_BG\\_only'\n",
       "\\item 'Left\\_only'\n",
       "\\item 'Right\\_only'\n",
       "\\item 'CTX\\_BG\\_shared'\n",
       "\\item 'CTX\\_THAL\\_shared'\n",
       "\\item 'sample\\_number\\_presented\\_COUPTF2'\n",
       "\\item 'CTX\\_COUPTF2'\n",
       "\\item 'THAL\\_COUPTF2'\n",
       "\\item 'HIP\\_COUPTF2'\n",
       "\\item 'L\\_COUPTF2'\n",
       "\\item 'R\\_COUPTF2'\n",
       "\\item 'I\\_COUPTF2'\n",
       "\\item 'F\\_COUPTF2'\n",
       "\\item 'P\\_COUPTF2'\n",
       "\\item 'O\\_COUPTF2'\n",
       "\\item 'T\\_COUPTF2'\n",
       "\\item 'Thal\\_COUPTF2'\n",
       "\\item 'Hip\\_COUPTF2'\n",
       "\\item 'CTX\\_only\\_COUPTF2'\n",
       "\\item 'HIP\\_only\\_COUPTF2'\n",
       "\\item 'THAL\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_THAL\\_shared\\_COUPTF2'\n",
       "\\item 'CTX\\_HIP\\_shared\\_COUPTF2'\n",
       "\\item 'HIP\\_THAL\\_shared\\_COUPTF2'\n",
       "\\item 'Left\\_only\\_COUPTF2'\n",
       "\\item 'Right\\_only\\_COUPTF2'\n",
       "\\item 'sample\\_number\\_presented\\_TBR1'\n",
       "\\item 'CTX\\_TBR1'\n",
       "\\item 'HIP\\_TBR1'\n",
       "\\item 'L\\_TBR1'\n",
       "\\item 'R\\_TBR1'\n",
       "\\item 'I\\_TBR1'\n",
       "\\item 'F\\_TBR1'\n",
       "\\item 'P\\_TBR1'\n",
       "\\item 'O\\_TBR1'\n",
       "\\item 'T\\_TBR1'\n",
       "\\item 'Hip\\_TBR1'\n",
       "\\item 'CTX\\_only\\_TBR1'\n",
       "\\item 'HIP\\_only\\_TBR1'\n",
       "\\item 'Left\\_only\\_TBR1'\n",
       "\\item 'Right\\_only\\_TBR1'\n",
       "\\item 'sample\\_number\\_presented\\_DLX1'\n",
       "\\item 'CTX\\_DLX1'\n",
       "\\item 'BG\\_DLX1'\n",
       "\\item 'THAL\\_DLX1'\n",
       "\\item 'HIP\\_DLX1'\n",
       "\\item 'L\\_DLX1'\n",
       "\\item 'R\\_DLX1'\n",
       "\\item 'I\\_DLX1'\n",
       "\\item 'F\\_DLX1'\n",
       "\\item 'P\\_DLX1'\n",
       "\\item 'O\\_DLX1'\n",
       "\\item 'T\\_DLX1'\n",
       "\\item 'Cau\\_DLX1'\n",
       "\\item 'GP\\_DLX1'\n",
       "\\item 'CTX\\_only\\_DLX1'\n",
       "\\item 'HIP\\_only\\_DLX1'\n",
       "\\item 'THAL\\_only\\_DLX1'\n",
       "\\item 'BG\\_only\\_DLX1'\n",
       "\\item 'CTX\\_HIP\\_shared\\_DLX1'\n",
       "\\item 'CTX\\_THAL\\_shared\\_DLX1'\n",
       "\\item 'CTX\\_BG\\_shared\\_DLX1'\n",
       "\\item 'Left\\_only\\_DLX1'\n",
       "\\item 'Right\\_only\\_DLX1'\n",
       "\\item 'sample\\_number\\_presented\\_CBH'\n",
       "\\item 'CTX\\_CBH'\n",
       "\\item 'BG\\_CBH'\n",
       "\\item 'HIP\\_CBH'\n",
       "\\item 'I\\_CBH'\n",
       "\\item 'F\\_CBH'\n",
       "\\item 'P\\_CBH'\n",
       "\\item 'O\\_CBH'\n",
       "\\item 'mO\\_CBH'\n",
       "\\item 'T\\_CBH'\n",
       "\\item 'Cau\\_CBH'\n",
       "\\item 'Put\\_CBH'\n",
       "\\item 'GP\\_CBH'\n",
       "\\item 'CTX\\_only\\_CBH'\n",
       "\\item 'HIP\\_only\\_CBH'\n",
       "\\item 'BG\\_only\\_CBH'\n",
       "\\item 'GP\\_only\\_CBH'\n",
       "\\item 'CTX\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_BG\\_shared\\_CBH'\n",
       "\\item 'BG\\_HIP\\_shared\\_CBH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'X'\n",
       "2. 'variant_list'\n",
       "3. 'sample_number_presented'\n",
       "4. 'HEART'\n",
       "5. 'ADRENAL'\n",
       "6. 'CB'\n",
       "7. 'CTX'\n",
       "8. 'LIVER'\n",
       "9. 'KIDNEY'\n",
       "10. 'POA'\n",
       "11. 'BG'\n",
       "12. 'THAL'\n",
       "13. 'HIP'\n",
       "14. 'SKIN'\n",
       "15. 'OLF'\n",
       "16. 'L'\n",
       "17. 'R'\n",
       "18. 'I'\n",
       "19. 'F'\n",
       "20. 'P'\n",
       "21. 'O'\n",
       "22. 'T'\n",
       "23. 'Cau'\n",
       "24. 'Put'\n",
       "25. 'Thal'\n",
       "26. 'GP'\n",
       "27. 'Hip'\n",
       "28. 'EC'\n",
       "29. 'Olf'\n",
       "30. 'CC'\n",
       "31. 'mO'\n",
       "32. 'Cb'\n",
       "33. 'DLX1'\n",
       "34. 'TBR1'\n",
       "35. 'COUPTF2'\n",
       "36. 'Bulk'\n",
       "37. 'NeuN'\n",
       "38. 'Brain_only'\n",
       "39. 'CTX_only'\n",
       "40. 'BG_only'\n",
       "41. 'THAL_only'\n",
       "42. 'CTX_BG_only'\n",
       "43. 'Left_only'\n",
       "44. 'Right_only'\n",
       "45. 'CTX_BG_shared'\n",
       "46. 'CTX_THAL_shared'\n",
       "47. 'sample_number_presented_COUPTF2'\n",
       "48. 'CTX_COUPTF2'\n",
       "49. 'THAL_COUPTF2'\n",
       "50. 'HIP_COUPTF2'\n",
       "51. 'L_COUPTF2'\n",
       "52. 'R_COUPTF2'\n",
       "53. 'I_COUPTF2'\n",
       "54. 'F_COUPTF2'\n",
       "55. 'P_COUPTF2'\n",
       "56. 'O_COUPTF2'\n",
       "57. 'T_COUPTF2'\n",
       "58. 'Thal_COUPTF2'\n",
       "59. 'Hip_COUPTF2'\n",
       "60. 'CTX_only_COUPTF2'\n",
       "61. 'HIP_only_COUPTF2'\n",
       "62. 'THAL_only_COUPTF2'\n",
       "63. 'CTX_THAL_shared_COUPTF2'\n",
       "64. 'CTX_HIP_shared_COUPTF2'\n",
       "65. 'HIP_THAL_shared_COUPTF2'\n",
       "66. 'Left_only_COUPTF2'\n",
       "67. 'Right_only_COUPTF2'\n",
       "68. 'sample_number_presented_TBR1'\n",
       "69. 'CTX_TBR1'\n",
       "70. 'HIP_TBR1'\n",
       "71. 'L_TBR1'\n",
       "72. 'R_TBR1'\n",
       "73. 'I_TBR1'\n",
       "74. 'F_TBR1'\n",
       "75. 'P_TBR1'\n",
       "76. 'O_TBR1'\n",
       "77. 'T_TBR1'\n",
       "78. 'Hip_TBR1'\n",
       "79. 'CTX_only_TBR1'\n",
       "80. 'HIP_only_TBR1'\n",
       "81. 'Left_only_TBR1'\n",
       "82. 'Right_only_TBR1'\n",
       "83. 'sample_number_presented_DLX1'\n",
       "84. 'CTX_DLX1'\n",
       "85. 'BG_DLX1'\n",
       "86. 'THAL_DLX1'\n",
       "87. 'HIP_DLX1'\n",
       "88. 'L_DLX1'\n",
       "89. 'R_DLX1'\n",
       "90. 'I_DLX1'\n",
       "91. 'F_DLX1'\n",
       "92. 'P_DLX1'\n",
       "93. 'O_DLX1'\n",
       "94. 'T_DLX1'\n",
       "95. 'Cau_DLX1'\n",
       "96. 'GP_DLX1'\n",
       "97. 'CTX_only_DLX1'\n",
       "98. 'HIP_only_DLX1'\n",
       "99. 'THAL_only_DLX1'\n",
       "100. 'BG_only_DLX1'\n",
       "101. 'CTX_HIP_shared_DLX1'\n",
       "102. 'CTX_THAL_shared_DLX1'\n",
       "103. 'CTX_BG_shared_DLX1'\n",
       "104. 'Left_only_DLX1'\n",
       "105. 'Right_only_DLX1'\n",
       "106. 'sample_number_presented_CBH'\n",
       "107. 'CTX_CBH'\n",
       "108. 'BG_CBH'\n",
       "109. 'HIP_CBH'\n",
       "110. 'I_CBH'\n",
       "111. 'F_CBH'\n",
       "112. 'P_CBH'\n",
       "113. 'O_CBH'\n",
       "114. 'mO_CBH'\n",
       "115. 'T_CBH'\n",
       "116. 'Cau_CBH'\n",
       "117. 'Put_CBH'\n",
       "118. 'GP_CBH'\n",
       "119. 'CTX_only_CBH'\n",
       "120. 'HIP_only_CBH'\n",
       "121. 'BG_only_CBH'\n",
       "122. 'GP_only_CBH'\n",
       "123. 'CTX_HIP_shared_CBH'\n",
       "124. 'CTX_BG_shared_CBH'\n",
       "125. 'BG_HIP_shared_CBH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"X\"                               \"variant_list\"                   \n",
       "  [3] \"sample_number_presented\"         \"HEART\"                          \n",
       "  [5] \"ADRENAL\"                         \"CB\"                             \n",
       "  [7] \"CTX\"                             \"LIVER\"                          \n",
       "  [9] \"KIDNEY\"                          \"POA\"                            \n",
       " [11] \"BG\"                              \"THAL\"                           \n",
       " [13] \"HIP\"                             \"SKIN\"                           \n",
       " [15] \"OLF\"                             \"L\"                              \n",
       " [17] \"R\"                               \"I\"                              \n",
       " [19] \"F\"                               \"P\"                              \n",
       " [21] \"O\"                               \"T\"                              \n",
       " [23] \"Cau\"                             \"Put\"                            \n",
       " [25] \"Thal\"                            \"GP\"                             \n",
       " [27] \"Hip\"                             \"EC\"                             \n",
       " [29] \"Olf\"                             \"CC\"                             \n",
       " [31] \"mO\"                              \"Cb\"                             \n",
       " [33] \"DLX1\"                            \"TBR1\"                           \n",
       " [35] \"COUPTF2\"                         \"Bulk\"                           \n",
       " [37] \"NeuN\"                            \"Brain_only\"                     \n",
       " [39] \"CTX_only\"                        \"BG_only\"                        \n",
       " [41] \"THAL_only\"                       \"CTX_BG_only\"                    \n",
       " [43] \"Left_only\"                       \"Right_only\"                     \n",
       " [45] \"CTX_BG_shared\"                   \"CTX_THAL_shared\"                \n",
       " [47] \"sample_number_presented_COUPTF2\" \"CTX_COUPTF2\"                    \n",
       " [49] \"THAL_COUPTF2\"                    \"HIP_COUPTF2\"                    \n",
       " [51] \"L_COUPTF2\"                       \"R_COUPTF2\"                      \n",
       " [53] \"I_COUPTF2\"                       \"F_COUPTF2\"                      \n",
       " [55] \"P_COUPTF2\"                       \"O_COUPTF2\"                      \n",
       " [57] \"T_COUPTF2\"                       \"Thal_COUPTF2\"                   \n",
       " [59] \"Hip_COUPTF2\"                     \"CTX_only_COUPTF2\"               \n",
       " [61] \"HIP_only_COUPTF2\"                \"THAL_only_COUPTF2\"              \n",
       " [63] \"CTX_THAL_shared_COUPTF2\"         \"CTX_HIP_shared_COUPTF2\"         \n",
       " [65] \"HIP_THAL_shared_COUPTF2\"         \"Left_only_COUPTF2\"              \n",
       " [67] \"Right_only_COUPTF2\"              \"sample_number_presented_TBR1\"   \n",
       " [69] \"CTX_TBR1\"                        \"HIP_TBR1\"                       \n",
       " [71] \"L_TBR1\"                          \"R_TBR1\"                         \n",
       " [73] \"I_TBR1\"                          \"F_TBR1\"                         \n",
       " [75] \"P_TBR1\"                          \"O_TBR1\"                         \n",
       " [77] \"T_TBR1\"                          \"Hip_TBR1\"                       \n",
       " [79] \"CTX_only_TBR1\"                   \"HIP_only_TBR1\"                  \n",
       " [81] \"Left_only_TBR1\"                  \"Right_only_TBR1\"                \n",
       " [83] \"sample_number_presented_DLX1\"    \"CTX_DLX1\"                       \n",
       " [85] \"BG_DLX1\"                         \"THAL_DLX1\"                      \n",
       " [87] \"HIP_DLX1\"                        \"L_DLX1\"                         \n",
       " [89] \"R_DLX1\"                          \"I_DLX1\"                         \n",
       " [91] \"F_DLX1\"                          \"P_DLX1\"                         \n",
       " [93] \"O_DLX1\"                          \"T_DLX1\"                         \n",
       " [95] \"Cau_DLX1\"                        \"GP_DLX1\"                        \n",
       " [97] \"CTX_only_DLX1\"                   \"HIP_only_DLX1\"                  \n",
       " [99] \"THAL_only_DLX1\"                  \"BG_only_DLX1\"                   \n",
       "[101] \"CTX_HIP_shared_DLX1\"             \"CTX_THAL_shared_DLX1\"           \n",
       "[103] \"CTX_BG_shared_DLX1\"              \"Left_only_DLX1\"                 \n",
       "[105] \"Right_only_DLX1\"                 \"sample_number_presented_CBH\"    \n",
       "[107] \"CTX_CBH\"                         \"BG_CBH\"                         \n",
       "[109] \"HIP_CBH\"                         \"I_CBH\"                          \n",
       "[111] \"F_CBH\"                           \"P_CBH\"                          \n",
       "[113] \"O_CBH\"                           \"mO_CBH\"                         \n",
       "[115] \"T_CBH\"                           \"Cau_CBH\"                        \n",
       "[117] \"Put_CBH\"                         \"GP_CBH\"                         \n",
       "[119] \"CTX_only_CBH\"                    \"HIP_only_CBH\"                   \n",
       "[121] \"BG_only_CBH\"                     \"GP_only_CBH\"                    \n",
       "[123] \"CTX_HIP_shared_CBH\"              \"CTX_BG_shared_CBH\"              \n",
       "[125] \"BG_HIP_shared_CBH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(filterdata)\n",
    "names(annotated)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "394561d0",
   "metadata": {},
   "source": [
    "## Filter for Mosaic Variants Shared by WGS and Amplicon Sequencing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0c519e31",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Confirmed total variants (WGS + Amp):  780\"\n"
     ]
    }
   ],
   "source": [
    "#sort df for only WGS + Ampliseq confirmed (1) -> confirmed variants (excluding JGG controls)\n",
    "totalConfVarsInclCtrl = length(unique( subset(filterdata, Mosaic_shared_btw_WGS_Amp == 1) ))\n",
    "confirmedVars = subset(filterdata, Mosaic_shared_btw_WGS_Amp == 1 & Organ != \"JGG\")\n",
    "\n",
    "numTotalVars = length(unique((confirmedVars$CHROM.POS.REF.ALT)))\n",
    "pretext = \"Confirmed total variants (WGS + Amp): \"\n",
    "print(paste(pretext, length(unique((confirmedVars$CHROM.POS.REF.ALT))))) "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e22360c3",
   "metadata": {},
   "source": [
    "## Sorting of Confirmed Mosaic Variants by Region and MAF_CI within Mosaic Threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8bf55d1f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"780 total confirmed unique variants\"\n",
      "[1] \"428 unique variants in brain\"\n",
      "[1] \"264 unique variants in left hemisphere\"\n",
      "[1] \"282 unique variants in right hemisphere\"\n",
      "[1] \"201 unique variants in cortex\"\n",
      "[1] \"178 unique variants in basal ganglia\"\n",
      "[1] \"190 unique variants in DLX1 cells\"\n",
      "[1] \"156 unique variants in TBR1 cells\"\n",
      "[1] \"194 unique variants in DLX1+COUPTFII cells (GABAergic interneurons)\"\n"
     ]
    }
   ],
   "source": [
    "#Mosaic Variants only in the brain\n",
    "#Brain : CTX, BG, THAL, HIP, POA (preoptic area), OLF, CB (cerebellum)\n",
    "#Kidney : ADRENAL, KIDNEY\n",
    "#Other: HEART, LIVER, SKIN\n",
    "# Exculde JGG variants (control)\n",
    "\n",
    "\n",
    "allOrgans = unique(confirmedVars$Organ)\n",
    "#print(\"All regions: \")\n",
    "#print(allOrgans)\n",
    "\n",
    "##use MAF above threshold\n",
    "##upper and lower bounds for MAF to be a valid variant\n",
    "Mosaic_UPPER = 0.4562841\n",
    "Mosaic_LOWER = 0.002360687\n",
    "\n",
    "brainVars = subset(confirmedVars, (Organ == \"CTX\" | Organ == \"BG\" | Organ == \"THAL\" | Organ == \"HIP\" | Organ == \"POA\" | Organ == \"CB\" | Organ == \"OLF\"))\n",
    "brainVars = subset(brainVars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numBrainUniqueVars = length(unique(brainVars$CHROM.POS.REF.ALT))\n",
    "\n",
    "leftHemisphereVars = subset(brainVars, Hemisphere == \"L\")\n",
    "leftHemisphereVars = subset(leftHemisphereVars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numLeftHVars = length(unique(leftHemisphereVars$CHROM.POS.REF.ALT))\n",
    "\n",
    "rightHemisphereVars = subset(brainVars, Hemisphere == \"R\")\n",
    "rightHemisphereVars = subset(rightHemisphereVars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numRightHVars = length(unique(rightHemisphereVars$CHROM.POS.REF.ALT))\n",
    "\n",
    "cortexVars = subset(brainVars, Organ == \"CTX\")\n",
    "cortexVars = subset(cortexVars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numCortexVars = length(unique(cortexVars$CHROM.POS.REF.ALT))\n",
    "\n",
    "basalGangliaVars = subset(brainVars, Organ == \"BG\")\n",
    "basalGangliaVars = subset(basalGangliaVars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numBgVars = length(unique(basalGangliaVars$CHROM.POS.REF.ALT))\n",
    "\n",
    "dlx1Vars = subset(brainVars, Cell_Type == \"DLX1\")\n",
    "dlx1Vars = subset(dlx1Vars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numDlx1Vars = length(unique(dlx1Vars$CHROM.POS.REF.ALT))\n",
    "\n",
    "tbr1Vars = subset(brainVars, Cell_Type == \"TBR1\")\n",
    "tbr1Vars = subset(tbr1Vars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numTbr1Vars = length(unique(tbr1Vars$CHROM.POS.REF.ALT))\n",
    "\n",
    "dlx1_couptf2Vars = subset(brainVars, Cell_Type == \"DLX1\" | Cell_Type == \"COUPTF2\")\n",
    "dlx1_couptf2Vars = subset(dlx1_couptf2Vars, LOWER_CI > Mosaic_LOWER & UPPER_CI < Mosaic_UPPER & NORMAL_LOWER_CI < Mosaic_LOWER)\n",
    "numdlx1_couptf2Vars = length(unique(dlx1_couptf2Vars$CHROM.POS.REF.ALT))\n",
    "\n",
    "\n",
    "\n",
    "print(paste(numTotalVars, \"total confirmed unique variants\"))\n",
    "print(paste(numBrainUniqueVars, \"unique variants in brain\"))\n",
    "print(paste(numLeftHVars, \"unique variants in left hemisphere\"))\n",
    "print(paste(numRightHVars, \"unique variants in right hemisphere\"))\n",
    "print(paste(numCortexVars, \"unique variants in cortex\"))\n",
    "print(paste(numBgVars, \"unique variants in basal ganglia\"))\n",
    "print(paste(numDlx1Vars, \"unique variants in DLX1 cells\"))\n",
    "print(paste(numTbr1Vars, \"unique variants in TBR1 cells\"))\n",
    "print(paste(numdlx1_couptf2Vars, \"unique variants in DLX1+COUPTFII cells (GABAergic interneurons)\"))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5ed2291",
   "metadata": {},
   "source": [
    "## Bargraph of Number of Unique Mosaic Variants By Region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "variantCounts = c(numBrainUniqueVars, numCortexVars, numTotalVars, numLeftHVars, numRightHVars, numBgVars)\n",
    "regions = c(\"Brain\", \"Cortex\", \"Total\", \"Left H\", \"Right H\", \"Basal Ganglia\")\n",
    "variantCountsByCellType = c(numDlx1Vars, numTbr1Vars, numdlx1_couptf2Vars)\n",
    "cellTypes = c(\"DLX1\", \"TBR1\", \"DLX1+COUPTF2\")\n",
    "\n",
    "variantsByRegion = data.frame(regions, variantCounts)\n",
    "variantsByCellType = data.frame(cellTypes, variantCountsByCellType)\n",
    "\n",
    "regionCount <- ggplot(variantsByRegion, aes(x = regions, y = variantCounts)) +\n",
    "  geom_bar(stat = \"identity\") +\n",
    "  theme(panel.grid.major = element_blank(), \n",
    "        panel.grid.minor = element_blank(),\n",
    "        panel.background = element_rect(fill = \"white\"),\n",
    "        axis.line = element_line()) + \n",
    "  geom_text(aes(label = variantCounts), position = position_dodge(0.9), vjust = -0.5, color = \"#940303\")\n",
    "\n",
    "cellTypeCount <- ggplot(variantsByCellType, aes(x = cellTypes, y = variantCountsByCellType)) +\n",
    "  geom_bar(stat = \"identity\") +\n",
    "  theme(panel.grid.major = element_blank(), \n",
    "        panel.grid.minor = element_blank(),\n",
    "        panel.background = element_rect(fill = \"white\"),\n",
    "        axis.line = element_line()) + \n",
    "  geom_text(aes(label = variantCountsByCellType), position = position_dodge(0.9), vjust = -0.5, color = \"#940303\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ae9ceb6",
   "metadata": {},
   "source": [
    "## Correlation matrix of Variant x Variant using ComplexHeatmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "77cf9080",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#ComplexHeatmap plot (working)\n",
    "#PEARSON\n",
    "\n",
    "varsConfirmedIn2Sample = subset(annotated, sample_number_presented >= 2)\n",
    "\n",
    "confirmed2SampleVars = subset(confirmedVars, CHROM.POS.REF.ALT %in% varsConfirmedIn2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleVars$MAF), confirmed2SampleVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "LR_colors = colorRamp2(c(-1, 0, 1), c(\"#d68426\", \"#ffffff\", \"#2a1abf\"))\n",
    "\n",
    "pdf(\"220_correlation_7669_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "Bulk = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Bulk\")),\n",
    "ADRENAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Adrenal\")),\n",
    "HEART = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Heart\")),\n",
    "LIVER = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Liver\")),\n",
    "KIDNEY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Kidney\")),\n",
    "SKIN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Skin\")),\n",
    "Brain_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CB = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cerebellum\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "POA = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Preoptic Area\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "HIP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Hippocampus\")),\n",
    "OLF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Olfactory Bulb\")),\n",
    "CTX_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "THAL_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUP_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 only\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        Bulk = varsConfirmedIn2Sample$Bulk,\n",
    "                                        ADRENAL = varsConfirmedIn2Sample$ADRENAL,\n",
    "                                        HEART = varsConfirmedIn2Sample$HEART,\n",
    "                                        LIVER = varsConfirmedIn2Sample$LIVER,\n",
    "                                        KIDNEY = varsConfirmedIn2Sample$KIDNEY,\n",
    "                                        SKIN = varsConfirmedIn2Sample$SKIN,\n",
    "                                        Brain_only = varsConfirmedIn2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedIn2Sample$CB,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX,\n",
    "                                        POA = varsConfirmedIn2Sample$POA,\n",
    "                                        BG = varsConfirmedIn2Sample$BG,\n",
    "                                        THAL = varsConfirmedIn2Sample$THAL,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP,\n",
    "                                        OLF = varsConfirmedIn2Sample$OLF,\n",
    "                                        CTX_only = varsConfirmedIn2Sample$CTX_only,\n",
    "                                        CTX_BG_only = varsConfirmedIn2Sample$CTX_BG_only,\n",
    "                                        BG_only = varsConfirmedIn2Sample$BG_only,\n",
    "                                        THAL_only = varsConfirmedIn2Sample$THAL_only,\n",
    "                                        L = varsConfirmedIn2Sample$L,\n",
    "                                        R = varsConfirmedIn2Sample$R,\n",
    "                                        LR_only = varsConfirmedIn2Sample$Left_only - varsConfirmedIn2Sample$Right_only, \n",
    "                                        NeuN = varsConfirmedIn2Sample$NeuN,\n",
    "                                        IN_only = varsConfirmedIn2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedIn2Sample$COUPTF2,\n",
    "                                        COUP_only = varsConfirmedIn2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedIn2Sample$TBR1,\n",
    "                                        TBR_only = varsConfirmedIn2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedIn2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR,\n",
    "                                        col = list(LR_only = LR_colors,\n",
    "                                        COUP_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"black\")),\n",
    "                                        IN_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"darkgreen\")),\n",
    "                                        TBR_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"purple\")), \n",
    "                                        TBR_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"#e58f9d\")),\n",
    "                                        DLX_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"maroon\"))))#,\n",
    "                                        #annotation_legend_param = legendDets) \n",
    "\n",
    "leftMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_7669_Pearson_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "leftMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "afd545f1",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#SPEARMAN\n",
    "\n",
    "\n",
    "pdf(\"220_correlation_7669_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "Bulk = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Bulk\")),\n",
    "ADRENAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Adrenal\")),\n",
    "HEART = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Heart\")),\n",
    "LIVER = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Liver\")),\n",
    "KIDNEY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Kidney\")),\n",
    "SKIN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Skin\")),\n",
    "Brain_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CB = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cerebellum\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "POA = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Preoptic Area\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "HIP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Hippocampus\")),\n",
    "OLF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Olfactory Bulb\")),\n",
    "CTX_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "THAL_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUP_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 only\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        Bulk = varsConfirmedIn2Sample$Bulk,\n",
    "                                        ADRENAL = varsConfirmedIn2Sample$ADRENAL,\n",
    "                                        HEART = varsConfirmedIn2Sample$HEART,\n",
    "                                        LIVER = varsConfirmedIn2Sample$LIVER,\n",
    "                                        KIDNEY = varsConfirmedIn2Sample$KIDNEY,\n",
    "                                        SKIN = varsConfirmedIn2Sample$SKIN,\n",
    "                                        Brain_only = varsConfirmedIn2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedIn2Sample$CB,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX,\n",
    "                                        POA = varsConfirmedIn2Sample$POA,\n",
    "                                        BG = varsConfirmedIn2Sample$BG,\n",
    "                                        THAL = varsConfirmedIn2Sample$THAL,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP,\n",
    "                                        OLF = varsConfirmedIn2Sample$OLF,\n",
    "                                        CTX_only = varsConfirmedIn2Sample$CTX_only,\n",
    "                                        CTX_BG_only = varsConfirmedIn2Sample$CTX_BG_only,\n",
    "                                        BG_only = varsConfirmedIn2Sample$BG_only,\n",
    "                                        THAL_only = varsConfirmedIn2Sample$THAL_only,\n",
    "                                        L = varsConfirmedIn2Sample$L,\n",
    "                                        R = varsConfirmedIn2Sample$R,\n",
    "                                        LR_only = varsConfirmedIn2Sample$Left_only - varsConfirmedIn2Sample$Right_only, \n",
    "                                        NeuN = varsConfirmedIn2Sample$NeuN,\n",
    "                                        IN_only = varsConfirmedIn2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedIn2Sample$COUPTF2,\n",
    "                                        COUP_only = varsConfirmedIn2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedIn2Sample$TBR1,\n",
    "                                        TBR_only = varsConfirmedIn2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedIn2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR,\n",
    "                                        col = list(LR_only = LR_colors,\n",
    "                                        COUP_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"black\")),\n",
    "                                        IN_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"darkgreen\")),\n",
    "                                        TBR_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"purple\")), \n",
    "                                        TBR_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"#e58f9d\")),\n",
    "                                        DLX_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"maroon\"))))#,\n",
    "                                        #annotation_legend_param = legendDets) \n",
    "\n",
    "\n",
    "leftMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "leftMapUpdate\n",
    "dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "75a8a982",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#BULK\n",
    "\n",
    "bulkVars = subset(confirmedVars, Cell_Type == \"Bulk\")\n",
    "varsConfirmedInBulk2Sample = subset(varsConfirmedIn2Sample, Bulk == 1)\n",
    "confirmed2SampleBulkVars = subset(bulkVars, CHROM.POS.REF.ALT %in% varsConfirmedInBulk2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleBulkVars$MAF), confirmed2SampleBulkVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "pdf(\"220_correlation_BULK_Pearson_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "Bulk = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Bulk\")),\n",
    "ADRENAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Adrenal\")),\n",
    "HEART = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Heart\")),\n",
    "LIVER = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Liver\")),\n",
    "KIDNEY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Kidney\")),\n",
    "SKIN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Skin\")),\n",
    "Brain_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CB = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cerebellum\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "POA = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Preoptic Area\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "HIP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Hippocampus\")),\n",
    "OLF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Olfactory Bulb\")),\n",
    "CTX_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "THAL_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUP_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_only = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 only\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedInBulk2Sample = varsConfirmedInBulk2Sample[order(varsConfirmedInBulk2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        Bulk = varsConfirmedInBulk2Sample$Bulk,\n",
    "                                        ADRENAL = varsConfirmedInBulk2Sample$ADRENAL,\n",
    "                                        HEART = varsConfirmedInBulk2Sample$HEART,\n",
    "                                        LIVER = varsConfirmedInBulk2Sample$LIVER,\n",
    "                                        KIDNEY = varsConfirmedInBulk2Sample$KIDNEY,\n",
    "                                        SKIN = varsConfirmedInBulk2Sample$SKIN,\n",
    "                                        Brain_only = varsConfirmedInBulk2Sample$Brain_only,\n",
    "                                        CB = varsConfirmedInBulk2Sample$CB,\n",
    "                                        CTX = varsConfirmedInBulk2Sample$CTX,\n",
    "                                        POA = varsConfirmedInBulk2Sample$POA,\n",
    "                                        BG = varsConfirmedInBulk2Sample$BG,\n",
    "                                        THAL = varsConfirmedInBulk2Sample$THAL,\n",
    "                                        HIP = varsConfirmedInBulk2Sample$HIP,\n",
    "                                        OLF = varsConfirmedInBulk2Sample$OLF,\n",
    "                                        CTX_only = varsConfirmedInBulk2Sample$CTX_only,\n",
    "                                        CTX_BG_only = varsConfirmedInBulk2Sample$CTX_BG_only,\n",
    "                                        BG_only = varsConfirmedInBulk2Sample$BG_only,\n",
    "                                        THAL_only = varsConfirmedInBulk2Sample$THAL_only,\n",
    "                                        L = varsConfirmedInBulk2Sample$L,\n",
    "                                        R = varsConfirmedInBulk2Sample$R,\n",
    "                                        LR_only = varsConfirmedInBulk2Sample$Left_only - varsConfirmedInBulk2Sample$Right_only, \n",
    "                                        NeuN = varsConfirmedInBulk2Sample$NeuN,\n",
    "                                        IN_only = varsConfirmedInBulk2Sample$IN_only,\n",
    "                                        COUPTF2 = varsConfirmedInBulk2Sample$COUPTF2,\n",
    "                                        COUP_only = varsConfirmedInBulk2Sample$COUP_only,\n",
    "                                        TBR1 = varsConfirmedInBulk2Sample$TBR1,\n",
    "                                        TBR_only = varsConfirmedInBulk2Sample$TBR_only,\n",
    "                                        TBR_BR = varsConfirmedInBulk2Sample$TBR_BR,\n",
    "                                        DLX1 = varsConfirmedInBulk2Sample$DLX1,\n",
    "                                        DLX_BR = varsConfirmedInBulk2Sample$DLX_BR,\n",
    "                                        col = list(LR_only = LR_colors,\n",
    "                                        COUP_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"black\")),\n",
    "                                        IN_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"darkgreen\")),\n",
    "                                        TBR_only = colorRamp2(c(0, 1), c(\"#ffffff\", \"purple\")), \n",
    "                                        TBR_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"#e58f9d\")),\n",
    "                                        DLX_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"maroon\"))))#,\n",
    "                                        #annotation_legend_param = legendDets) \n",
    "\n",
    "bulkMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "bulkMapUpdate\n",
    "dev.off()\n",
    "\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "pdf(\"220_correlation_BULK_Spearman_Manhattan_Dist.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInBulk2Sample = varsConfirmedInBulk2Sample[order(varsConfirmedInBulk2Sample$X), ]\n",
    "\n",
    "\n",
    "bulkMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "bulkMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d2cbbdef",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "color_list = list(LR_ONLY = LR_colors,\n",
    "                COUP_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"black\")),\n",
    "                IN_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"darkgreen\")),\n",
    "                TBR_ONLY = colorRamp2(c(0, 1), c(\"#ffffff\", \"purple\")), \n",
    "                TBR_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"#e58f9d\")),\n",
    "                DLX_BR = colorRamp2(c(0, 1), c(\"#ffffff\", \"maroon\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8adcd653",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_DLX1'</li><li>'CTX_DLX1'</li><li>'BG_DLX1'</li><li>'THAL_DLX1'</li><li>'HIP_DLX1'</li><li>'L_DLX1'</li><li>'R_DLX1'</li><li>'I_DLX1'</li><li>'F_DLX1'</li><li>'P_DLX1'</li><li>'O_DLX1'</li><li>'T_DLX1'</li><li>'Cau_DLX1'</li><li>'GP_DLX1'</li><li>'CTX_only_DLX1'</li><li>'BG_only_DLX1'</li><li>'CTX_HIP_shared_DLX1'</li><li>'CTX_BG_shared_DLX1'</li><li>'Left_only_DLX1'</li><li>'Right_only_DLX1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_DLX1'\n",
       "\\item 'CTX\\_DLX1'\n",
       "\\item 'BG\\_DLX1'\n",
       "\\item 'THAL\\_DLX1'\n",
       "\\item 'HIP\\_DLX1'\n",
       "\\item 'L\\_DLX1'\n",
       "\\item 'R\\_DLX1'\n",
       "\\item 'I\\_DLX1'\n",
       "\\item 'F\\_DLX1'\n",
       "\\item 'P\\_DLX1'\n",
       "\\item 'O\\_DLX1'\n",
       "\\item 'T\\_DLX1'\n",
       "\\item 'Cau\\_DLX1'\n",
       "\\item 'GP\\_DLX1'\n",
       "\\item 'CTX\\_only\\_DLX1'\n",
       "\\item 'BG\\_only\\_DLX1'\n",
       "\\item 'CTX\\_HIP\\_shared\\_DLX1'\n",
       "\\item 'CTX\\_BG\\_shared\\_DLX1'\n",
       "\\item 'Left\\_only\\_DLX1'\n",
       "\\item 'Right\\_only\\_DLX1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_DLX1'\n",
       "2. 'CTX_DLX1'\n",
       "3. 'BG_DLX1'\n",
       "4. 'THAL_DLX1'\n",
       "5. 'HIP_DLX1'\n",
       "6. 'L_DLX1'\n",
       "7. 'R_DLX1'\n",
       "8. 'I_DLX1'\n",
       "9. 'F_DLX1'\n",
       "10. 'P_DLX1'\n",
       "11. 'O_DLX1'\n",
       "12. 'T_DLX1'\n",
       "13. 'Cau_DLX1'\n",
       "14. 'GP_DLX1'\n",
       "15. 'CTX_only_DLX1'\n",
       "16. 'BG_only_DLX1'\n",
       "17. 'CTX_HIP_shared_DLX1'\n",
       "18. 'CTX_BG_shared_DLX1'\n",
       "19. 'Left_only_DLX1'\n",
       "20. 'Right_only_DLX1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_DLX1\" \"CTX_DLX1\"                    \n",
       " [3] \"BG_DLX1\"                      \"THAL_DLX1\"                   \n",
       " [5] \"HIP_DLX1\"                     \"L_DLX1\"                      \n",
       " [7] \"R_DLX1\"                       \"I_DLX1\"                      \n",
       " [9] \"F_DLX1\"                       \"P_DLX1\"                      \n",
       "[11] \"O_DLX1\"                       \"T_DLX1\"                      \n",
       "[13] \"Cau_DLX1\"                     \"GP_DLX1\"                     \n",
       "[15] \"CTX_only_DLX1\"                \"BG_only_DLX1\"                \n",
       "[17] \"CTX_HIP_shared_DLX1\"          \"CTX_BG_shared_DLX1\"          \n",
       "[19] \"Left_only_DLX1\"               \"Right_only_DLX1\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#DLX1\n",
    "dlxVars = subset(confirmedVars, Sort_main == 1 & Cell_Type == \"DLX1\")\n",
    "varsConfirmedInDLX1_2Sample = subset(annotated, sample_number_presented_DLX1 >= 2) # using __DLX1 columns\n",
    "confirmed2SampleDLXVars = subset(dlxVars, CHROM.POS.REF.ALT %in% varsConfirmedInDLX1_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleDLXVars$MAF), confirmed2SampleDLXVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[, colSums(varsConfirmedInDLX1_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"220_correlation_DLX_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "names(varsConfirmedInDLX1_2Sample %>% select(ends_with(\"_DLX1\")))\n",
    "\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        #BRAIN_ONLY = varsConfirmedInDLX1_2Sample$Brain_only,\n",
    "                                        CTX = varsConfirmedInDLX1_2Sample$CTX_DLX1,\n",
    "                                        BG = varsConfirmedInDLX1_2Sample$BG_DLX1,\n",
    "                                        THAL = varsConfirmedInDLX1_2Sample$THAL_DLX1,\n",
    "                                        PF = varsConfirmedInDLX1_2Sample$PF_DLX1,\n",
    "                                        F = varsConfirmedInDLX1_2Sample$F_DLX1,\n",
    "                                        P = varsConfirmedInDLX1_2Sample$P_DLX1,\n",
    "                                        O = varsConfirmedInDLX1_2Sample$O_DLX1,\n",
    "                                        T = varsConfirmedInDLX1_2Sample$T_DLX1,\n",
    "                                        INS = varsConfirmedInDLX1_2Sample$Ins_DLX1,\n",
    "                                        PUT = varsConfirmedInDLX1_2Sample$Put_DLX1,\n",
    "                                        GP = varsConfirmedInDLX1_2Sample$GP_DLX1,\n",
    "                                        CTX_ONLY = varsConfirmedInDLX1_2Sample$CTX_only_DLX1,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInDLX1_2Sample$CTX_BG_only_DLX1,\n",
    "                                        BG_ONLY = varsConfirmedInDLX1_2Sample$BG_only_DLX1,\n",
    "                                        LR_ONLY = varsConfirmedInDLX1_2Sample$Left_only_DLX1 - varsConfirmedInDLX1_2Sample$Right_only_DLX1, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInDLX1_2Sample$L_BR_PF_only_DLX1,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInDLX1_2Sample$L_BR_F_only_DLX1,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInDLX1_2Sample$L_BR_O_only_DLX1,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedInDLX1_2Sample$L_BR_Ins_only_DLX1,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInDLX1_2Sample$R_BR_F_only_DLX1,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInDLX1_2Sample$R_BR_O_only_DLX1,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedInDLX1_2Sample$R_BR_Ins_only_DLX1,\n",
    "                                        IN_ONLY = varsConfirmedInDLX1_2Sample$IN_only_DLX1,\n",
    "                                        COUPTF2 = varsConfirmedInDLX1_2Sample$COUPTF2_DLX1,\n",
    "                                        COUP_BR = varsConfirmedInDLX1_2Sample$COUP_BR_DLX1,\n",
    "                                        COUP_ONLY = varsConfirmedInDLX1_2Sample$COUP_only_DLX1,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "dlxMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_DLX_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "\n",
    "dlxMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_DLX_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInDLX1_2Sample = varsConfirmedInDLX1_2Sample[order(varsConfirmedInDLX1_2Sample$X), ]\n",
    "\n",
    "\n",
    "dlxMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "dlxMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "9d393d23",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_TBR1'</li><li>'CTX_TBR1'</li><li>'HIP_TBR1'</li><li>'L_TBR1'</li><li>'R_TBR1'</li><li>'I_TBR1'</li><li>'F_TBR1'</li><li>'P_TBR1'</li><li>'O_TBR1'</li><li>'T_TBR1'</li><li>'Hip_TBR1'</li><li>'CTX_only_TBR1'</li><li>'Left_only_TBR1'</li><li>'Right_only_TBR1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_TBR1'\n",
       "\\item 'CTX\\_TBR1'\n",
       "\\item 'HIP\\_TBR1'\n",
       "\\item 'L\\_TBR1'\n",
       "\\item 'R\\_TBR1'\n",
       "\\item 'I\\_TBR1'\n",
       "\\item 'F\\_TBR1'\n",
       "\\item 'P\\_TBR1'\n",
       "\\item 'O\\_TBR1'\n",
       "\\item 'T\\_TBR1'\n",
       "\\item 'Hip\\_TBR1'\n",
       "\\item 'CTX\\_only\\_TBR1'\n",
       "\\item 'Left\\_only\\_TBR1'\n",
       "\\item 'Right\\_only\\_TBR1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_TBR1'\n",
       "2. 'CTX_TBR1'\n",
       "3. 'HIP_TBR1'\n",
       "4. 'L_TBR1'\n",
       "5. 'R_TBR1'\n",
       "6. 'I_TBR1'\n",
       "7. 'F_TBR1'\n",
       "8. 'P_TBR1'\n",
       "9. 'O_TBR1'\n",
       "10. 'T_TBR1'\n",
       "11. 'Hip_TBR1'\n",
       "12. 'CTX_only_TBR1'\n",
       "13. 'Left_only_TBR1'\n",
       "14. 'Right_only_TBR1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_TBR1\" \"CTX_TBR1\"                    \n",
       " [3] \"HIP_TBR1\"                     \"L_TBR1\"                      \n",
       " [5] \"R_TBR1\"                       \"I_TBR1\"                      \n",
       " [7] \"F_TBR1\"                       \"P_TBR1\"                      \n",
       " [9] \"O_TBR1\"                       \"T_TBR1\"                      \n",
       "[11] \"Hip_TBR1\"                     \"CTX_only_TBR1\"               \n",
       "[13] \"Left_only_TBR1\"               \"Right_only_TBR1\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#TBR1\n",
    "tbrVars = subset(confirmedVars, Sort_main == 1 & Cell_Type == \"TBR1\")\n",
    "varsConfirmedInTBR1_2Sample = subset(annotated, sample_number_presented_TBR1 >= 2) # using __TBR1 columns\n",
    "confirmed2SampleTBRVars = subset(tbrVars, CHROM.POS.REF.ALT %in% varsConfirmedInTBR1_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleTBRVars$MAF), confirmed2SampleTBRVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[, colSums(varsConfirmedInTBR1_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "names(varsConfirmedInTBR1_2Sample %>% select(ends_with(\"_TBR1\")))\n",
    "pdf(\"220_correlation_TBR_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BRAIN_ONLY = varsConfirmedInTBR1_2Sample$Brain_only_TBR1,\n",
    "                                        CTX = varsConfirmedInTBR1_2Sample$CTX_TBR1,\n",
    "                                        PF = varsConfirmedInTBR1_2Sample$PF_TBR1,\n",
    "                                        F = varsConfirmedInTBR1_2Sample$F_TBR1,\n",
    "                                        P = varsConfirmedInTBR1_2Sample$P_TBR1,\n",
    "                                        O = varsConfirmedInTBR1_2Sample$O_TBR1,\n",
    "                                        T = varsConfirmedInTBR1_2Sample$T_TBR1,\n",
    "                                        INS = varsConfirmedInTBR1_2Sample$Ins_TBR1,\n",
    "                                        CTX_ONLY = varsConfirmedInTBR1_2Sample$CTX_only_TBR1,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInTBR1_2Sample$CTX_BG_only_TBR1,\n",
    "                                        L = varsConfirmedInTBR1_2Sample$L_TBR1,\n",
    "                                        R = varsConfirmedInTBR1_2Sample$R_TBR1,\n",
    "                                        LR_ONLY = varsConfirmedInTBR1_2Sample$Left_only_TBR1 - varsConfirmedInTBR1_2Sample$Right_only_TBR1, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInTBR1_2Sample$L_BR_PF_only_TBR1,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInTBR1_2Sample$L_BR_F_only_TBR1,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInTBR1_2Sample$L_BR_O_only_TBR1,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInTBR1_2Sample$R_BR_F_only_TBR1,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInTBR1_2Sample$R_BR_O_only_TBR1,\n",
    "                                        TBR1 = varsConfirmedInTBR1_2Sample$TBR1_TBR1,\n",
    "                                        TBR_ONLY = varsConfirmedInTBR1_2Sample$TBR_only_TBR1,\n",
    "                                        TBR_BR = varsConfirmedInTBR1_2Sample$TBR_BR_TBR1,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "tbrMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_TBR_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "\n",
    "tbrMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_TBR_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInTBR1_2Sample = varsConfirmedInTBR1_2Sample[order(varsConfirmedInTBR1_2Sample$X), ]\n",
    "\n",
    "\n",
    "tbrMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "tbrMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_COUPTF2'</li><li>'CTX_COUPTF2'</li><li>'THAL_COUPTF2'</li><li>'HIP_COUPTF2'</li><li>'L_COUPTF2'</li><li>'R_COUPTF2'</li><li>'I_COUPTF2'</li><li>'F_COUPTF2'</li><li>'P_COUPTF2'</li><li>'O_COUPTF2'</li><li>'T_COUPTF2'</li><li>'Thal_COUPTF2'</li><li>'Hip_COUPTF2'</li><li>'CTX_only_COUPTF2'</li><li>'CTX_THAL_shared_COUPTF2'</li><li>'CTX_HIP_shared_COUPTF2'</li><li>'Left_only_COUPTF2'</li><li>'Right_only_COUPTF2'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_COUPTF2'\n",
       "\\item 'CTX\\_COUPTF2'\n",
       "\\item 'THAL\\_COUPTF2'\n",
       "\\item 'HIP\\_COUPTF2'\n",
       "\\item 'L\\_COUPTF2'\n",
       "\\item 'R\\_COUPTF2'\n",
       "\\item 'I\\_COUPTF2'\n",
       "\\item 'F\\_COUPTF2'\n",
       "\\item 'P\\_COUPTF2'\n",
       "\\item 'O\\_COUPTF2'\n",
       "\\item 'T\\_COUPTF2'\n",
       "\\item 'Thal\\_COUPTF2'\n",
       "\\item 'Hip\\_COUPTF2'\n",
       "\\item 'CTX\\_only\\_COUPTF2'\n",
       "\\item 'CTX\\_THAL\\_shared\\_COUPTF2'\n",
       "\\item 'CTX\\_HIP\\_shared\\_COUPTF2'\n",
       "\\item 'Left\\_only\\_COUPTF2'\n",
       "\\item 'Right\\_only\\_COUPTF2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_COUPTF2'\n",
       "2. 'CTX_COUPTF2'\n",
       "3. 'THAL_COUPTF2'\n",
       "4. 'HIP_COUPTF2'\n",
       "5. 'L_COUPTF2'\n",
       "6. 'R_COUPTF2'\n",
       "7. 'I_COUPTF2'\n",
       "8. 'F_COUPTF2'\n",
       "9. 'P_COUPTF2'\n",
       "10. 'O_COUPTF2'\n",
       "11. 'T_COUPTF2'\n",
       "12. 'Thal_COUPTF2'\n",
       "13. 'Hip_COUPTF2'\n",
       "14. 'CTX_only_COUPTF2'\n",
       "15. 'CTX_THAL_shared_COUPTF2'\n",
       "16. 'CTX_HIP_shared_COUPTF2'\n",
       "17. 'Left_only_COUPTF2'\n",
       "18. 'Right_only_COUPTF2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_COUPTF2\" \"CTX_COUPTF2\"                    \n",
       " [3] \"THAL_COUPTF2\"                    \"HIP_COUPTF2\"                    \n",
       " [5] \"L_COUPTF2\"                       \"R_COUPTF2\"                      \n",
       " [7] \"I_COUPTF2\"                       \"F_COUPTF2\"                      \n",
       " [9] \"P_COUPTF2\"                       \"O_COUPTF2\"                      \n",
       "[11] \"T_COUPTF2\"                       \"Thal_COUPTF2\"                   \n",
       "[13] \"Hip_COUPTF2\"                     \"CTX_only_COUPTF2\"               \n",
       "[15] \"CTX_THAL_shared_COUPTF2\"         \"CTX_HIP_shared_COUPTF2\"         \n",
       "[17] \"Left_only_COUPTF2\"               \"Right_only_COUPTF2\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#COUPTF2\n",
    "coupVars = subset(confirmedVars, Cell_Type == \"COUPTF2\")\n",
    "varsConfirmedInCOUPTF2_2Sample = subset(annotated, sample_number_presented_COUPTF2 >= 2) # using __COUPTF2 columns\n",
    "confirmed2SampleCOUPVars = subset(coupVars, CHROM.POS.REF.ALT %in% varsConfirmedInCOUPTF2_2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmed2SampleCOUPVars$MAF), confirmed2SampleCOUPVars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[, colSums(varsConfirmedInCOUPTF2_2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "names(varsConfirmedInCOUPTF2_2Sample %>% select(ends_with(\"_COUPTF2\")))\n",
    "pdf(\"220_correlation_COUP_Pearson.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.2, \"cm\"), \n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 4),\n",
    "                                        BRAIN_ONLY = varsConfirmedInCOUPTF2_2Sample$Brain_only_COUPTF2,\n",
    "                                        CTX = varsConfirmedInCOUPTF2_2Sample$CTX_COUPTF2,\n",
    "                                        BG = varsConfirmedInCOUPTF2_2Sample$BG_COUPTF2,\n",
    "                                        THAL = varsConfirmedInCOUPTF2_2Sample$THAL_COUPTF2,\n",
    "                                        PF = varsConfirmedInCOUPTF2_2Sample$PF_COUPTF2,\n",
    "                                        F = varsConfirmedInCOUPTF2_2Sample$F_COUPTF2,\n",
    "                                        P = varsConfirmedInCOUPTF2_2Sample$P_COUPTF2,\n",
    "                                        O = varsConfirmedInCOUPTF2_2Sample$O_COUPTF2,\n",
    "                                        T = varsConfirmedInCOUPTF2_2Sample$T_COUPTF2,\n",
    "                                        INS = varsConfirmedInCOUPTF2_2Sample$Ins_COUPTF2,\n",
    "                                        PUT = varsConfirmedInCOUPTF2_2Sample$Put_COUPTF2,\n",
    "                                        GP = varsConfirmedInCOUPTF2_2Sample$GP_COUPTF2,\n",
    "                                        CTX_ONLY = varsConfirmedInCOUPTF2_2Sample$CTX_only_COUPTF2,\n",
    "                                        CTX_BG_ONLY = varsConfirmedInCOUPTF2_2Sample$CTX_BG_only_COUPTF2,\n",
    "                                        BG_ONLY = varsConfirmedInCOUPTF2_2Sample$BG_only_COUPTF2,\n",
    "                                        L = varsConfirmedInCOUPTF2_2Sample$L_COUPTF2,\n",
    "                                        R = varsConfirmedInCOUPTF2_2Sample$R_COUPTF2,\n",
    "                                        LR_ONLY = varsConfirmedInCOUPTF2_2Sample$Left_only_COUPTF2 - varsConfirmedInCOUPTF2_2Sample$Right_only_COUPTF2, \n",
    "                                        L_BR_PF_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_PF_only_COUPTF2,\n",
    "                                        L_BR_F_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_F_only_COUPTF2,\n",
    "                                        L_BR_O_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_O_only_COUPTF2,\n",
    "                                        L_BR_INS_ONLY = varsConfirmedInCOUPTF2_2Sample$L_BR_Ins_only_COUPTF2,\n",
    "                                        R_BR_F_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_F_only_COUPTF2,\n",
    "                                        R_BR_O_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_O_only_COUPTF2,\n",
    "                                        R_BR_INS_ONLY = varsConfirmedInCOUPTF2_2Sample$R_BR_Ins_only_COUPTF2,\n",
    "                                        IN_ONLY = varsConfirmedInCOUPTF2_2Sample$IN_only_COUPTF2,\n",
    "                                        COUPTF2 = varsConfirmedInCOUPTF2_2Sample$COUPTF2_COUPTF2,\n",
    "                                        COUP_ONLY = varsConfirmedInCOUPTF2_2Sample$COUP_only_COUPTF2,\n",
    "                                        COUP_BR = varsConfirmedInCOUPTF2_2Sample$COUP_BR_COUPTF2,\n",
    "                                        COUPTF2_ENRICHED = varsConfirmedInCOUPTF2_2Sample$COUPTF2_enriched_COUPTF2,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "coupMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_COUP_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "coupMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_COUP_Spearman.pdf\", width = unit(16, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "varsConfirmedInCOUPTF2_2Sample = varsConfirmedInCOUPTF2_2Sample[order(varsConfirmedInCOUPTF2_2Sample$X), ]\n",
    "\n",
    "coupMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "coupMapUpdate\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "212f7f0d",
   "metadata": {},
   "source": [
    "# CTX TBR1 vs BG DLX1 vs HIP TBR1 Variant Matrix for Determination of Earlier Divergence of HIP from CTX+BG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fbf94471",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'sample_number_presented_CBH'</li><li>'CTX_CBH'</li><li>'BG_CBH'</li><li>'HIP_CBH'</li><li>'I_CBH'</li><li>'F_CBH'</li><li>'P_CBH'</li><li>'O_CBH'</li><li>'mO_CBH'</li><li>'T_CBH'</li><li>'Cau_CBH'</li><li>'Put_CBH'</li><li>'GP_CBH'</li><li>'CTX_only_CBH'</li><li>'BG_only_CBH'</li><li>'CTX_HIP_shared_CBH'</li><li>'CTX_BG_shared_CBH'</li><li>'BG_HIP_shared_CBH'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'sample\\_number\\_presented\\_CBH'\n",
       "\\item 'CTX\\_CBH'\n",
       "\\item 'BG\\_CBH'\n",
       "\\item 'HIP\\_CBH'\n",
       "\\item 'I\\_CBH'\n",
       "\\item 'F\\_CBH'\n",
       "\\item 'P\\_CBH'\n",
       "\\item 'O\\_CBH'\n",
       "\\item 'mO\\_CBH'\n",
       "\\item 'T\\_CBH'\n",
       "\\item 'Cau\\_CBH'\n",
       "\\item 'Put\\_CBH'\n",
       "\\item 'GP\\_CBH'\n",
       "\\item 'CTX\\_only\\_CBH'\n",
       "\\item 'BG\\_only\\_CBH'\n",
       "\\item 'CTX\\_HIP\\_shared\\_CBH'\n",
       "\\item 'CTX\\_BG\\_shared\\_CBH'\n",
       "\\item 'BG\\_HIP\\_shared\\_CBH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'sample_number_presented_CBH'\n",
       "2. 'CTX_CBH'\n",
       "3. 'BG_CBH'\n",
       "4. 'HIP_CBH'\n",
       "5. 'I_CBH'\n",
       "6. 'F_CBH'\n",
       "7. 'P_CBH'\n",
       "8. 'O_CBH'\n",
       "9. 'mO_CBH'\n",
       "10. 'T_CBH'\n",
       "11. 'Cau_CBH'\n",
       "12. 'Put_CBH'\n",
       "13. 'GP_CBH'\n",
       "14. 'CTX_only_CBH'\n",
       "15. 'BG_only_CBH'\n",
       "16. 'CTX_HIP_shared_CBH'\n",
       "17. 'CTX_BG_shared_CBH'\n",
       "18. 'BG_HIP_shared_CBH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"sample_number_presented_CBH\" \"CTX_CBH\"                    \n",
       " [3] \"BG_CBH\"                      \"HIP_CBH\"                    \n",
       " [5] \"I_CBH\"                       \"F_CBH\"                      \n",
       " [7] \"P_CBH\"                       \"O_CBH\"                      \n",
       " [9] \"mO_CBH\"                      \"T_CBH\"                      \n",
       "[11] \"Cau_CBH\"                     \"Put_CBH\"                    \n",
       "[13] \"GP_CBH\"                      \"CTX_only_CBH\"               \n",
       "[15] \"BG_only_CBH\"                 \"CTX_HIP_shared_CBH\"         \n",
       "[17] \"CTX_BG_shared_CBH\"           \"BG_HIP_shared_CBH\"          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#CBH\n",
    "# CTX  |  BG  |  HIP |\n",
    "# TBR1 | DLX1 | TBR1 |\n",
    "\n",
    "cbhVars = subset(confirmedVars, CTX_BG_HIP_relationship == 1)\n",
    "varsConfirmedIn2Sample = subset(annotated, sample_number_presented_CBH >= 2) # using CBH columns\n",
    "#varsConfirmedIn2Sample = subset(varsConfirmedIn2Sample, (CTX_CBH == 1 & TBR_only_CBH == 1) | (BG_CBH == 1 & DLX1_CBH == 1) | (HIP_CBH == 1 & TBR_only_CBH == 1) )\n",
    "varsConfirmedIn2Sample = subset(varsConfirmedIn2Sample, (CTX_CBH == 1) | (BG_CBH == 1) | (HIP_CBH == 1) )\n",
    "confirmedCBH_vars = subset(cbhVars, CHROM.POS.REF.ALT %in% varsConfirmedIn2Sample$variant_list)\n",
    "\n",
    "onlyMAFofVars = split(sqrt(confirmedCBH_vars$MAF), confirmedCBH_vars$CHROM.POS.REF.ALT) #now using sqrt(MAF) 4/14/23\n",
    "df_encoded <- model.matrix(~ . - 1, data = onlyMAFofVars) #converts to matrix format with col variants, row MAF\n",
    "pearson_corr_matrix = round(cor(df_encoded, method = 'pearson', use = 'pairwise.complete.obs'), 4)\n",
    "spearman_corr_matrix = round(cor(df_encoded, method = 'spearman', use = 'pairwise.complete.obs'), 4)\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[, colSums(varsConfirmedIn2Sample != 0) > 0] #removes all 0 value columns\n",
    "\n",
    "pdf(\"220_correlation_CBH_Pearson.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "\n",
    "legendDets = list(\n",
    "BRAIN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Brain Only\")),\n",
    "CTX = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Cortex\")),\n",
    "BG = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Basal Ganglia\")),\n",
    "THAL = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Thalamus\")),\n",
    "PF = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PF\")),\n",
    "F = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"F\")),\n",
    "P = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"P\")),\n",
    "O = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"O\")),\n",
    "T = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"T\")),\n",
    "INS = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"INS\")),\n",
    "PUT = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PUT\")),\n",
    "GP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"GP\")),\n",
    "CTX_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX Only\")),\n",
    "CTX_BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"CTX+BG Only\")),\n",
    "BG_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"BG Only\")),\n",
    "L = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Left\")),\n",
    "R = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right\")),\n",
    "LR_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(-1, 1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Right Only\", \"Left Only\")),\n",
    "DARPP32M.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32M.NeuNP\")),\n",
    "DARPP32P.NeuNP = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DARPP32P.NeuNP\")),\n",
    "LHX2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"LHX2\")),\n",
    "OLIG2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"OLIG2\")),\n",
    "PU1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"PU1\")),\n",
    "NeuN = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"NeuN\")),\n",
    "IN_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"Interneuron Only\")),\n",
    "COUPTF2 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2\")),\n",
    "COUPTF2_ENRICHED = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Enriched\")),\n",
    "COUP_ONLY = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 only\")),\n",
    "COUP_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"COUPTF2 Brain only\")),\n",
    "TBR1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1\")),\n",
    "TBR_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"TBR1 Brain only\")),\n",
    "DLX1 = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1\")),\n",
    "DLX_BR = list(\n",
    "        title = \"\",\n",
    "        at = c(1),\n",
    "        color_bar = \"discrete\",\n",
    "        labels = c(\"DLX1 Brain only\")))\n",
    "\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "names(varsConfirmedIn2Sample %>% select(ends_with(\"_CBH\")))\n",
    "\n",
    "left_right_cell_col_ann = HeatmapAnnotation(simple_anno_size = unit(0.5, \"cm\"), #previously 0.2\n",
    "                                        which = \"row\", \n",
    "                                        annotation_name_gp = gpar(fontsize = 6), #previously 4 fontsize\n",
    "                                        #BRAIN_ONLY = varsConfirmedIn2Sample$Brain_only_CBH,\n",
    "                                        #INS = varsConfirmedIn2Sample$Ins_CBH,\n",
    "                                        #CAU = varsConfirmedIn2Sample$Cau_CBH,\n",
    "                                        #PUT = varsConfirmedIn2Sample$Put_CBH,\n",
    "                                        #GP = varsConfirmedIn2Sample$GP_CBH,\n",
    "                                        #GP_ONLY = varsConfirmedIn2Sample$GP_only_CBH,\n",
    "                                        #PF = varsConfirmedIn2Sample$PF_CBH,\n",
    "                                        #F = varsConfirmedIn2Sample$F_CBH,\n",
    "                                        #P = varsConfirmedIn2Sample$P_CBH,\n",
    "                                        #O = varsConfirmedIn2Sample$O_CBH,\n",
    "                                        #T = varsConfirmedIn2Sample$T_CBH,\n",
    "                                        #CA1 = varsConfirmedIn2Sample$CA1_CBH,\n",
    "                                        #CA3 = varsConfirmedIn2Sample$CA3_CBH,\n",
    "                                        #DG = varsConfirmedIn2Sample$DG_CBH,\n",
    "                                        IN_ONLY = varsConfirmedIn2Sample$IN_only_CBH,\n",
    "                                        #DLX1 = varsConfirmedIn2Sample$DLX1_CBH,\n",
    "                                        DLX_BR = varsConfirmedIn2Sample$DLX_BR_CBH,\n",
    "                                        #TBR1 = varsConfirmedIn2Sample$TBR1_CBH,\n",
    "                                        #TBR_ONLY = varsConfirmedIn2Sample$TBR_only_CBH,\n",
    "                                        TBR_BR = varsConfirmedIn2Sample$TBR_BR_CBH,\n",
    "                                        CTX = varsConfirmedIn2Sample$CTX_CBH,\n",
    "                                        BG = varsConfirmedIn2Sample$BG_CBH,\n",
    "                                        HIP = varsConfirmedIn2Sample$HIP_CBH,\n",
    "                                        CTX_ONLY = varsConfirmedIn2Sample$CTX_only_CBH,\n",
    "                                        BG_ONLY = varsConfirmedIn2Sample$BG_only_CBH,\n",
    "                                        HIP_ONLY = varsConfirmedIn2Sample$HIP_only_CBH,\n",
    "                                        CTX_BG = varsConfirmedIn2Sample$CTX_BG_shared_CBH,\n",
    "                                        BG_HIP = varsConfirmedIn2Sample$BG_HIP_shared_CBH,\n",
    "                                        CTX_HIP = varsConfirmedIn2Sample$CTX_HIP_shared_CBH,\n",
    "                                        col = color_list)#,\n",
    "                                        #annotation_legend_param = legendDets)\n",
    "\n",
    "cbhMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_CBH_Pearson_Manhattan_Dist.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "cbhMapUpdate = Heatmap(pearson_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"manhattan\", clustering_distance_columns = \"manhattan\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n",
    "\n",
    "pdf(\"220_correlation_CBH_Spearman.pdf\", width = unit(20, \"cm\"), height = unit(8, \"cm\"))\n",
    "varsConfirmedIn2Sample = varsConfirmedIn2Sample[order(varsConfirmedIn2Sample$X), ]\n",
    "cbhMapUpdate = Heatmap(spearman_corr_matrix,\n",
    "        col = colorRamp2(c(-1, 0, 1), c(\"#0d0dad\", \"white\", \"firebrick3\")),\n",
    "        row_names_gp = gpar(fontsize = \"1\"), column_names_gp = gpar(fontsize = \"1\"), name = \"Correlation\",\n",
    "        left_annotation = left_right_cell_col_ann,\n",
    "        clustering_method_rows = \"complete\", clustering_method_columns = \"complete\", \n",
    "        clustering_distance_rows = \"euclidean\", clustering_distance_columns = \"euclidean\",\n",
    "        cluster_rows = TRUE, cluster_columns = TRUE,\n",
    "        width = unit(16, \"cm\"),\n",
    "        heatmap_height = unit(18, \"cm\")) \n",
    "cbhMapUpdate\n",
    "dev.off()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
