{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "74efffe5-17bc-489d-8830-a2a589900411",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘gplots’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(gplots)\n",
    "library(reshape2)\n",
    "library(umap)\n",
    "library(ggExtra) \n",
    "library(pals)\n",
    "library(RColorBrewer)\n",
    "\n",
    "filtered <- read.table(\"../20230101_QC/final_summary_anno3.txt\",header=T,sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e385256-62ae-4ec0-b1b6-72dcdd04d824",
   "metadata": {
    "tags": []
   },
   "source": [
    "# COUPTF2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "08c31827-7572-4cb9-955d-ceff0ae09762",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"A total of variant calls:  353682\"\n",
      "[1] \"The number of variants included in ampliseq:  2406\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ID'</li><li>'CHROM'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'CHROM.POS.REF.ALT'</li><li>'ANNO'</li><li>'GENE'</li><li>'GNOMAD_FREQ'</li><li>'REPEAT_MASKER'</li><li>'SEGDUP'</li><li>'HOMOPOLYMER'</li><li>'REF_SEQ'</li><li>'DINUCLEOTIDE'</li><li>'NEAR_INDEL'</li><li>'UCSC_RPMSK'</li><li>'REF_COUNT'</li><li>'ALT_COUNT'</li><li>'MAF'</li><li>'LOWER_CI'</li><li>'UPPER_CI'</li><li>'CI_IS_GREATER'</li><li>'NORMAL_REF_COUNT'</li><li>'NORMAL_ALT_COUNT'</li><li>'NORMAL_MAF'</li><li>'NORMAL_LOWER_CI'</li><li>'NORMAL_UPPER_CI'</li><li>'NORMAL_CI_IS_GREATER'</li><li>'Mosaic'</li><li>'Mosaic_shared_btw_WGS_Amp'</li><li>'Hemisphere'</li><li>'Region'</li><li>'Punch'</li><li>'Cell_Type'</li><li>'Location'</li><li>'AntPost'</li><li>'DorVen'</li><li>'Organ'</li><li>'Note'</li><li>'Bulk_main'</li><li>'Sort_main'</li><li>'CTX_BG_HIP_relationship'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ID'\n",
       "\\item 'CHROM'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'CHROM.POS.REF.ALT'\n",
       "\\item 'ANNO'\n",
       "\\item 'GENE'\n",
       "\\item 'GNOMAD\\_FREQ'\n",
       "\\item 'REPEAT\\_MASKER'\n",
       "\\item 'SEGDUP'\n",
       "\\item 'HOMOPOLYMER'\n",
       "\\item 'REF\\_SEQ'\n",
       "\\item 'DINUCLEOTIDE'\n",
       "\\item 'NEAR\\_INDEL'\n",
       "\\item 'UCSC\\_RPMSK'\n",
       "\\item 'REF\\_COUNT'\n",
       "\\item 'ALT\\_COUNT'\n",
       "\\item 'MAF'\n",
       "\\item 'LOWER\\_CI'\n",
       "\\item 'UPPER\\_CI'\n",
       "\\item 'CI\\_IS\\_GREATER'\n",
       "\\item 'NORMAL\\_REF\\_COUNT'\n",
       "\\item 'NORMAL\\_ALT\\_COUNT'\n",
       "\\item 'NORMAL\\_MAF'\n",
       "\\item 'NORMAL\\_LOWER\\_CI'\n",
       "\\item 'NORMAL\\_UPPER\\_CI'\n",
       "\\item 'NORMAL\\_CI\\_IS\\_GREATER'\n",
       "\\item 'Mosaic'\n",
       "\\item 'Mosaic\\_shared\\_btw\\_WGS\\_Amp'\n",
       "\\item 'Hemisphere'\n",
       "\\item 'Region'\n",
       "\\item 'Punch'\n",
       "\\item 'Cell\\_Type'\n",
       "\\item 'Location'\n",
       "\\item 'AntPost'\n",
       "\\item 'DorVen'\n",
       "\\item 'Organ'\n",
       "\\item 'Note'\n",
       "\\item 'Bulk\\_main'\n",
       "\\item 'Sort\\_main'\n",
       "\\item 'CTX\\_BG\\_HIP\\_relationship'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ID'\n",
       "2. 'CHROM'\n",
       "3. 'POS'\n",
       "4. 'REF'\n",
       "5. 'ALT'\n",
       "6. 'CHROM.POS.REF.ALT'\n",
       "7. 'ANNO'\n",
       "8. 'GENE'\n",
       "9. 'GNOMAD_FREQ'\n",
       "10. 'REPEAT_MASKER'\n",
       "11. 'SEGDUP'\n",
       "12. 'HOMOPOLYMER'\n",
       "13. 'REF_SEQ'\n",
       "14. 'DINUCLEOTIDE'\n",
       "15. 'NEAR_INDEL'\n",
       "16. 'UCSC_RPMSK'\n",
       "17. 'REF_COUNT'\n",
       "18. 'ALT_COUNT'\n",
       "19. 'MAF'\n",
       "20. 'LOWER_CI'\n",
       "21. 'UPPER_CI'\n",
       "22. 'CI_IS_GREATER'\n",
       "23. 'NORMAL_REF_COUNT'\n",
       "24. 'NORMAL_ALT_COUNT'\n",
       "25. 'NORMAL_MAF'\n",
       "26. 'NORMAL_LOWER_CI'\n",
       "27. 'NORMAL_UPPER_CI'\n",
       "28. 'NORMAL_CI_IS_GREATER'\n",
       "29. 'Mosaic'\n",
       "30. 'Mosaic_shared_btw_WGS_Amp'\n",
       "31. 'Hemisphere'\n",
       "32. 'Region'\n",
       "33. 'Punch'\n",
       "34. 'Cell_Type'\n",
       "35. 'Location'\n",
       "36. 'AntPost'\n",
       "37. 'DorVen'\n",
       "38. 'Organ'\n",
       "39. 'Note'\n",
       "40. 'Bulk_main'\n",
       "41. 'Sort_main'\n",
       "42. 'CTX_BG_HIP_relationship'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ID\"                        \"CHROM\"                    \n",
       " [3] \"POS\"                       \"REF\"                      \n",
       " [5] \"ALT\"                       \"CHROM.POS.REF.ALT\"        \n",
       " [7] \"ANNO\"                      \"GENE\"                     \n",
       " [9] \"GNOMAD_FREQ\"               \"REPEAT_MASKER\"            \n",
       "[11] \"SEGDUP\"                    \"HOMOPOLYMER\"              \n",
       "[13] \"REF_SEQ\"                   \"DINUCLEOTIDE\"             \n",
       "[15] \"NEAR_INDEL\"                \"UCSC_RPMSK\"               \n",
       "[17] \"REF_COUNT\"                 \"ALT_COUNT\"                \n",
       "[19] \"MAF\"                       \"LOWER_CI\"                 \n",
       "[21] \"UPPER_CI\"                  \"CI_IS_GREATER\"            \n",
       "[23] \"NORMAL_REF_COUNT\"          \"NORMAL_ALT_COUNT\"         \n",
       "[25] \"NORMAL_MAF\"                \"NORMAL_LOWER_CI\"          \n",
       "[27] \"NORMAL_UPPER_CI\"           \"NORMAL_CI_IS_GREATER\"     \n",
       "[29] \"Mosaic\"                    \"Mosaic_shared_btw_WGS_Amp\"\n",
       "[31] \"Hemisphere\"                \"Region\"                   \n",
       "[33] \"Punch\"                     \"Cell_Type\"                \n",
       "[35] \"Location\"                  \"AntPost\"                  \n",
       "[37] \"DorVen\"                    \"Organ\"                    \n",
       "[39] \"Note\"                      \"Bulk_main\"                \n",
       "[41] \"Sort_main\"                 \"CTX_BG_HIP_relationship\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"The number of validated MVs:  780\"\n",
      "[1] \"The number of validated MV calls:  1453\"\n",
      "[1] \"The number of validated MVs (in the variant_list variable):  780\"\n",
      "[1] \"The number of validated MVs and detected in this 2023 Ampliseq:  135\"\n",
      "[1] \"The number of shared validated MVs :  123\"\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(paste(\"A total of variant calls: \",nrow(filtered)))\n",
    "print(paste(\"The number of variants included in ampliseq: \", length(unique(filtered$CHROM.POS.REF.ALT))))\n",
    "colnames(filtered)\n",
    "filtered2 <- subset(filtered, ID %in% c(\n",
    "    '7669_R_Ant_Cau_DLX1','7669_R_sec3_Cau_DLX1',\n",
    "    '7669_R_Ant_Put_DLX1','7669_R_sec3_Put_DLX1',\n",
    "    #'7669_L_sec1_Cau_DLX1','7669_L_sec1_Put_DLX1',\n",
    "    '7669_R_O_1_COUPTF2','7669_R_P_1_COUPTF2',\n",
    "    '7669_R_T_1_COUPTF2', \n",
    "    '7669_R_I_1_COUPTF2','7669_R_F_1_COUPTF2',\n",
    "    '7669_R_O_1_TBR1','7669_R_P_1_TBR1','7669_R_T_1_TBR1','7669_R_I_1_TBR1','7669_R_F_1_TBR1',\n",
    "    '7669_R_O_1_DLX1','7669_R_P_1_DLX1','7669_R_T_1_DLX1','7669_R_I_1_DLX1','7669_R_F_1_DLX1'))\n",
    "filtered2$Punch<-as.character(filtered2$Punch)\n",
    "filtered2$Note<-as.character(filtered2$Note)\n",
    "\n",
    "mosaic_filtered <- filtered2[filtered2$Mosaic_shared_btw_WGS_Amp==1,]\n",
    "mosaic_filtered_cutoff <- subset(mosaic_filtered, LOWER_CI>0.002360687 & UPPER_CI<0.4562841 & NORMAL_LOWER_CI<0.002360687)\n",
    "\n",
    "print(paste(\"The number of validated MVs: \", length(unique(mosaic_filtered$CHROM.POS.REF.ALT))))\n",
    "print(paste(\"The number of validated MV calls: \", length(mosaic_filtered_cutoff$CHROM.POS.REF.ALT)))\n",
    "\n",
    "# statistics\n",
    "\n",
    "variant_list<-unique(mosaic_filtered$CHROM.POS.REF.ALT)\n",
    "variant_list<-as.data.frame(variant_list, row.names = variant_list) # assign rownames as CHROM.POS.REF.ALT\n",
    "print(paste(\"The number of validated MVs (in the variant_list variable): \", nrow(variant_list)))\n",
    "for (i in 1:nrow(variant_list)){\n",
    "    xx <- filtered2[filtered2$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "    xx <- subset(xx, LOWER_CI>0.002360687 & UPPER_CI<0.4562841 & NORMAL_LOWER_CI<0.002360687 )\n",
    "    n_row <- nrow(xx)\n",
    "    variant_list$sample_number_presented[i] <- n_row\n",
    "    }\n",
    "print(paste(\"The number of validated MVs and detected in this 2023 Ampliseq: \", nrow(variant_list[variant_list$sample_number_presented>0,])))\n",
    "# Non-private Mosaic variants\n",
    "print(paste(\"The number of shared validated MVs : \",nrow(variant_list[variant_list$sample_number_presented>1,])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "abebb81f-606d-4bba-a538-60b76bf006eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "for (j in c('HEART','ADRENAL','CB','CTX','LIVER','KIDNEY','POA','BG','THAL','SKIN','OLF','HIP')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Organ)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('L','R')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Hemisphere)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('I','F','P','O','T','Cau','Put','Thal','GP','Hip','POA','EC','Olf','CC','mO','Cb')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Region)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('DLX1','TBR1','COUPTF2','Bulk','NeuN')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Cell_Type)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1f8c4777-54b4-4cbf-be54-1d97259da57c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'variant_list'</li><li>'sample_number_presented'</li><li>'HEART'</li><li>'ADRENAL'</li><li>'CB'</li><li>'CTX'</li><li>'LIVER'</li><li>'KIDNEY'</li><li>'POA'</li><li>'BG'</li><li>'THAL'</li><li>'SKIN'</li><li>'OLF'</li><li>'HIP'</li><li>'L'</li><li>'R'</li><li>'I'</li><li>'F'</li><li>'P'</li><li>'O'</li><li>'T'</li><li>'Cau'</li><li>'Put'</li><li>'Thal'</li><li>'GP'</li><li>'Hip'</li><li>'EC'</li><li>'Olf'</li><li>'CC'</li><li>'mO'</li><li>'Cb'</li><li>'DLX1'</li><li>'TBR1'</li><li>'COUPTF2'</li><li>'Bulk'</li><li>'NeuN'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'variant\\_list'\n",
       "\\item 'sample\\_number\\_presented'\n",
       "\\item 'HEART'\n",
       "\\item 'ADRENAL'\n",
       "\\item 'CB'\n",
       "\\item 'CTX'\n",
       "\\item 'LIVER'\n",
       "\\item 'KIDNEY'\n",
       "\\item 'POA'\n",
       "\\item 'BG'\n",
       "\\item 'THAL'\n",
       "\\item 'SKIN'\n",
       "\\item 'OLF'\n",
       "\\item 'HIP'\n",
       "\\item 'L'\n",
       "\\item 'R'\n",
       "\\item 'I'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'O'\n",
       "\\item 'T'\n",
       "\\item 'Cau'\n",
       "\\item 'Put'\n",
       "\\item 'Thal'\n",
       "\\item 'GP'\n",
       "\\item 'Hip'\n",
       "\\item 'EC'\n",
       "\\item 'Olf'\n",
       "\\item 'CC'\n",
       "\\item 'mO'\n",
       "\\item 'Cb'\n",
       "\\item 'DLX1'\n",
       "\\item 'TBR1'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'Bulk'\n",
       "\\item 'NeuN'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'variant_list'\n",
       "2. 'sample_number_presented'\n",
       "3. 'HEART'\n",
       "4. 'ADRENAL'\n",
       "5. 'CB'\n",
       "6. 'CTX'\n",
       "7. 'LIVER'\n",
       "8. 'KIDNEY'\n",
       "9. 'POA'\n",
       "10. 'BG'\n",
       "11. 'THAL'\n",
       "12. 'SKIN'\n",
       "13. 'OLF'\n",
       "14. 'HIP'\n",
       "15. 'L'\n",
       "16. 'R'\n",
       "17. 'I'\n",
       "18. 'F'\n",
       "19. 'P'\n",
       "20. 'O'\n",
       "21. 'T'\n",
       "22. 'Cau'\n",
       "23. 'Put'\n",
       "24. 'Thal'\n",
       "25. 'GP'\n",
       "26. 'Hip'\n",
       "27. 'EC'\n",
       "28. 'Olf'\n",
       "29. 'CC'\n",
       "30. 'mO'\n",
       "31. 'Cb'\n",
       "32. 'DLX1'\n",
       "33. 'TBR1'\n",
       "34. 'COUPTF2'\n",
       "35. 'Bulk'\n",
       "36. 'NeuN'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"variant_list\"            \"sample_number_presented\"\n",
       " [3] \"HEART\"                   \"ADRENAL\"                \n",
       " [5] \"CB\"                      \"CTX\"                    \n",
       " [7] \"LIVER\"                   \"KIDNEY\"                 \n",
       " [9] \"POA\"                     \"BG\"                     \n",
       "[11] \"THAL\"                    \"SKIN\"                   \n",
       "[13] \"OLF\"                     \"HIP\"                    \n",
       "[15] \"L\"                       \"R\"                      \n",
       "[17] \"I\"                       \"F\"                      \n",
       "[19] \"P\"                       \"O\"                      \n",
       "[21] \"T\"                       \"Cau\"                    \n",
       "[23] \"Put\"                     \"Thal\"                   \n",
       "[25] \"GP\"                      \"Hip\"                    \n",
       "[27] \"EC\"                      \"Olf\"                    \n",
       "[29] \"CC\"                      \"mO\"                     \n",
       "[31] \"Cb\"                      \"DLX1\"                   \n",
       "[33] \"TBR1\"                    \"COUPTF2\"                \n",
       "[35] \"Bulk\"                    \"NeuN\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(variant_list)\n",
    "#'HEART','ADRENAL','LIVER','KIDNEY','SKIN','POA','BG','THAL','HIP','OLF''CB','CTX',\n",
    "Brain_only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0) & (POA==1 | BG==1 | THAL==1 | HIP==1 | OLF==1 | CB==1 | CTX==1))\n",
    "variant_list$Brain_only[variant_list$variant_list %in% rownames(Brain_only_variants)]<-1\n",
    "variant_list$Brain_only[!variant_list$variant_list %in% rownames(Brain_only_variants)]<-0\n",
    "\n",
    "CTX_only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & BG==0 & THAL==0 & HIP==0 & OLF==0 & CB==0) & CTX==1)\n",
    "variant_list$CTX_only[variant_list$variant_list %in% rownames(CTX_only_variants)]<-1\n",
    "variant_list$CTX_only[!variant_list$variant_list %in% rownames(CTX_only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & THAL==0 & HIP==0 & OLF==0 & CB==0 & CTX==0) & (BG==1))\n",
    "variant_list$BG_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$BG_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & BG==0 & HIP==0 & OLF==0 & CB==0 & CTX==0) & (THAL==1))\n",
    "variant_list$THAL_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$THAL_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & THAL==0 & HIP==0 & OLF==0 & CB==0)& (CTX==1 | BG==1))\n",
    "variant_list$CTX_BG_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$CTX_BG_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & THAL==0 & HIP==0 & OLF==0 & CB==0)& (CTX==1 & BG==1))\n",
    "variant_list$CTX_BG_shared[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$CTX_BG_shared[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & BG==0 & HIP==0 & OLF==0 & CB==0)& (CTX==1 & THAL==1))\n",
    "variant_list$CTX_THAL_shared[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$CTX_THAL_shared[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "# only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & THAL==0 & CTX==0 & OLF==0 & CB==0)& (HIP==1 | BG==1))\n",
    "# variant_list$BG_HIP_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "# variant_list$BG_HIP_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "# only_variants<-subset(variant_list, (HEART==0 & ADRENAL==0 & LIVER==0 & KIDNEY==0 & SKIN==0 & POA==0 & THAL==0 & BG==0 & OLF==0 & CB==0)& (CTX==1 | HIP==1))\n",
    "# variant_list$CTX_HIP_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "# variant_list$CTX_HIP_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "\n",
    "\n",
    "only_variants<-subset(variant_list, (L==1) & (R==0) & (Brain_only==1))\n",
    "variant_list$Left_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$Left_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (R==1) & (L==0) & (Brain_only==1))\n",
    "variant_list$Right_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$Right_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==0) & (DLX1==1 | COUPTF2==1) & (Brain_only==1))\n",
    "variant_list$IN_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$IN_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==0) & (COUPTF2==1) & (Brain_only==1))\n",
    "variant_list$COUP_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$COUP_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==0 & COUPTF2==0) & (TBR1==1) & (Brain_only==1))\n",
    "variant_list$TBR_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==1) & (Brain_only==1))\n",
    "variant_list$TBR_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==1) & (Brain_only==1))\n",
    "variant_list$DLX_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1692919c-c05a-4e65-805d-0377d63749f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'variant_list'</li><li>'sample_number_presented'</li><li>'HEART'</li><li>'ADRENAL'</li><li>'CB'</li><li>'CTX'</li><li>'LIVER'</li><li>'KIDNEY'</li><li>'POA'</li><li>'BG'</li><li>'THAL'</li><li>'SKIN'</li><li>'OLF'</li><li>'HIP'</li><li>'L'</li><li>'R'</li><li>'I'</li><li>'F'</li><li>'P'</li><li>'O'</li><li>'T'</li><li>'Cau'</li><li>'Put'</li><li>'Thal'</li><li>'GP'</li><li>'Hip'</li><li>'EC'</li><li>'Olf'</li><li>'CC'</li><li>'mO'</li><li>'Cb'</li><li>'DLX1'</li><li>'TBR1'</li><li>'COUPTF2'</li><li>'Bulk'</li><li>'NeuN'</li><li>'Brain_only'</li><li>'CTX_only'</li><li>'BG_only'</li><li>'THAL_only'</li><li>'CTX_BG_only'</li><li>'CTX_BG_shared'</li><li>'CTX_THAL_shared'</li><li>'Left_only'</li><li>'Right_only'</li><li>'IN_only'</li><li>'COUP_only'</li><li>'TBR_only'</li><li>'TBR_BR'</li><li>'DLX_BR'</li><li>'TBR_BR_CTX'</li><li>'DLX_BR_BG'</li><li>'COUP_BR_CTX'</li><li>'TBR_BR_CTXnDLX_BR_BG'</li><li>'TBR_BR_CTXnCOUP_BR_CTX'</li><li>'DLX_BR_BGnCOUP_BR_CTX'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'variant\\_list'\n",
       "\\item 'sample\\_number\\_presented'\n",
       "\\item 'HEART'\n",
       "\\item 'ADRENAL'\n",
       "\\item 'CB'\n",
       "\\item 'CTX'\n",
       "\\item 'LIVER'\n",
       "\\item 'KIDNEY'\n",
       "\\item 'POA'\n",
       "\\item 'BG'\n",
       "\\item 'THAL'\n",
       "\\item 'SKIN'\n",
       "\\item 'OLF'\n",
       "\\item 'HIP'\n",
       "\\item 'L'\n",
       "\\item 'R'\n",
       "\\item 'I'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'O'\n",
       "\\item 'T'\n",
       "\\item 'Cau'\n",
       "\\item 'Put'\n",
       "\\item 'Thal'\n",
       "\\item 'GP'\n",
       "\\item 'Hip'\n",
       "\\item 'EC'\n",
       "\\item 'Olf'\n",
       "\\item 'CC'\n",
       "\\item 'mO'\n",
       "\\item 'Cb'\n",
       "\\item 'DLX1'\n",
       "\\item 'TBR1'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'Bulk'\n",
       "\\item 'NeuN'\n",
       "\\item 'Brain\\_only'\n",
       "\\item 'CTX\\_only'\n",
       "\\item 'BG\\_only'\n",
       "\\item 'THAL\\_only'\n",
       "\\item 'CTX\\_BG\\_only'\n",
       "\\item 'CTX\\_BG\\_shared'\n",
       "\\item 'CTX\\_THAL\\_shared'\n",
       "\\item 'Left\\_only'\n",
       "\\item 'Right\\_only'\n",
       "\\item 'IN\\_only'\n",
       "\\item 'COUP\\_only'\n",
       "\\item 'TBR\\_only'\n",
       "\\item 'TBR\\_BR'\n",
       "\\item 'DLX\\_BR'\n",
       "\\item 'TBR\\_BR\\_CTX'\n",
       "\\item 'DLX\\_BR\\_BG'\n",
       "\\item 'COUP\\_BR\\_CTX'\n",
       "\\item 'TBR\\_BR\\_CTXnDLX\\_BR\\_BG'\n",
       "\\item 'TBR\\_BR\\_CTXnCOUP\\_BR\\_CTX'\n",
       "\\item 'DLX\\_BR\\_BGnCOUP\\_BR\\_CTX'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'variant_list'\n",
       "2. 'sample_number_presented'\n",
       "3. 'HEART'\n",
       "4. 'ADRENAL'\n",
       "5. 'CB'\n",
       "6. 'CTX'\n",
       "7. 'LIVER'\n",
       "8. 'KIDNEY'\n",
       "9. 'POA'\n",
       "10. 'BG'\n",
       "11. 'THAL'\n",
       "12. 'SKIN'\n",
       "13. 'OLF'\n",
       "14. 'HIP'\n",
       "15. 'L'\n",
       "16. 'R'\n",
       "17. 'I'\n",
       "18. 'F'\n",
       "19. 'P'\n",
       "20. 'O'\n",
       "21. 'T'\n",
       "22. 'Cau'\n",
       "23. 'Put'\n",
       "24. 'Thal'\n",
       "25. 'GP'\n",
       "26. 'Hip'\n",
       "27. 'EC'\n",
       "28. 'Olf'\n",
       "29. 'CC'\n",
       "30. 'mO'\n",
       "31. 'Cb'\n",
       "32. 'DLX1'\n",
       "33. 'TBR1'\n",
       "34. 'COUPTF2'\n",
       "35. 'Bulk'\n",
       "36. 'NeuN'\n",
       "37. 'Brain_only'\n",
       "38. 'CTX_only'\n",
       "39. 'BG_only'\n",
       "40. 'THAL_only'\n",
       "41. 'CTX_BG_only'\n",
       "42. 'CTX_BG_shared'\n",
       "43. 'CTX_THAL_shared'\n",
       "44. 'Left_only'\n",
       "45. 'Right_only'\n",
       "46. 'IN_only'\n",
       "47. 'COUP_only'\n",
       "48. 'TBR_only'\n",
       "49. 'TBR_BR'\n",
       "50. 'DLX_BR'\n",
       "51. 'TBR_BR_CTX'\n",
       "52. 'DLX_BR_BG'\n",
       "53. 'COUP_BR_CTX'\n",
       "54. 'TBR_BR_CTXnDLX_BR_BG'\n",
       "55. 'TBR_BR_CTXnCOUP_BR_CTX'\n",
       "56. 'DLX_BR_BGnCOUP_BR_CTX'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"variant_list\"            \"sample_number_presented\"\n",
       " [3] \"HEART\"                   \"ADRENAL\"                \n",
       " [5] \"CB\"                      \"CTX\"                    \n",
       " [7] \"LIVER\"                   \"KIDNEY\"                 \n",
       " [9] \"POA\"                     \"BG\"                     \n",
       "[11] \"THAL\"                    \"SKIN\"                   \n",
       "[13] \"OLF\"                     \"HIP\"                    \n",
       "[15] \"L\"                       \"R\"                      \n",
       "[17] \"I\"                       \"F\"                      \n",
       "[19] \"P\"                       \"O\"                      \n",
       "[21] \"T\"                       \"Cau\"                    \n",
       "[23] \"Put\"                     \"Thal\"                   \n",
       "[25] \"GP\"                      \"Hip\"                    \n",
       "[27] \"EC\"                      \"Olf\"                    \n",
       "[29] \"CC\"                      \"mO\"                     \n",
       "[31] \"Cb\"                      \"DLX1\"                   \n",
       "[33] \"TBR1\"                    \"COUPTF2\"                \n",
       "[35] \"Bulk\"                    \"NeuN\"                   \n",
       "[37] \"Brain_only\"              \"CTX_only\"               \n",
       "[39] \"BG_only\"                 \"THAL_only\"              \n",
       "[41] \"CTX_BG_only\"             \"CTX_BG_shared\"          \n",
       "[43] \"CTX_THAL_shared\"         \"Left_only\"              \n",
       "[45] \"Right_only\"              \"IN_only\"                \n",
       "[47] \"COUP_only\"               \"TBR_only\"               \n",
       "[49] \"TBR_BR\"                  \"DLX_BR\"                 \n",
       "[51] \"TBR_BR_CTX\"              \"DLX_BR_BG\"              \n",
       "[53] \"COUP_BR_CTX\"             \"TBR_BR_CTXnDLX_BR_BG\"   \n",
       "[55] \"TBR_BR_CTXnCOUP_BR_CTX\"  \"DLX_BR_BGnCOUP_BR_CTX\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#---------------------------BG-DLX1 vs CTX-TBR1 vs CTX-COUPTFII-------------------\n",
    "only_variants<-subset(variant_list, (TBR1==1))\n",
    "variant_list$TBR_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==1))\n",
    "variant_list$DLX_BR_BG[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR_BG[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (COUPTF2==1))\n",
    "variant_list$COUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$COUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR_BR_CTX==1) & (DLX_BR_BG==1) & (COUP_BR_CTX==0))\n",
    "variant_list$TBR_BR_CTXnDLX_BR_BG[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTXnDLX_BR_BG[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR_BR_CTX==1) & (DLX_BR_BG==0) & (COUP_BR_CTX==1))\n",
    "variant_list$TBR_BR_CTXnCOUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTXnCOUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR_BR_CTX==0) & (DLX_BR_BG==1) & (COUP_BR_CTX==1))\n",
    "variant_list$DLX_BR_BGnCOUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR_BGnCOUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "names(variant_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "574fe5b3-f4d1-4b2a-a69d-64bd0cf37d9a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"TBR_BR_CTX variants:  116\"\n",
      "[1] \"DLX_BR_BG variants:  130\"\n",
      "[1] \"COUP_BR_CTX variants:  101\"\n",
      "[1] \"TBR_BR_CTXnDLX_BR_BG variants:  21\"\n",
      "[1] \"TBR_BR_CTXnCOUP_BR_CTX variants:  1\"\n",
      "[1] \"DLX_BR_BGnCOUP_BR_CTX variants:  2\"\n"
     ]
    }
   ],
   "source": [
    "print(paste(\"TBR_BR_CTX variants: \",nrow(variant_list[variant_list$TBR_BR_CTX==1,])))\n",
    "print(paste(\"DLX_BR_BG variants: \",nrow(variant_list[variant_list$DLX_BR_BG==1,])))\n",
    "print(paste(\"COUP_BR_CTX variants: \",nrow(variant_list[variant_list$COUP_BR_CTX==1,])))\n",
    "print(paste(\"TBR_BR_CTXnDLX_BR_BG variants: \",nrow(variant_list[variant_list$TBR_BR_CTXnDLX_BR_BG==1,])))\n",
    "print(paste(\"TBR_BR_CTXnCOUP_BR_CTX variants: \",nrow(variant_list[variant_list$TBR_BR_CTXnCOUP_BR_CTX==1,])))\n",
    "print(paste(\"DLX_BR_BGnCOUP_BR_CTX variants: \",nrow(variant_list[variant_list$DLX_BR_BGnCOUP_BR_CTX==1,])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "01ec4b6d-b3c6-43b5-ac82-3df37233faf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'7669_R_Ant_Cau_DLX1'</li><li>'7669_R_Ant_Put_DLX1'</li><li>'7669_R_F_1_COUPTF2'</li><li>'7669_R_F_1_DLX1'</li><li>'7669_R_F_1_TBR1'</li><li>'7669_R_I_1_COUPTF2'</li><li>'7669_R_I_1_DLX1'</li><li>'7669_R_I_1_TBR1'</li><li>'7669_R_O_1_COUPTF2'</li><li>'7669_R_O_1_DLX1'</li><li>'7669_R_O_1_TBR1'</li><li>'7669_R_P_1_COUPTF2'</li><li>'7669_R_P_1_DLX1'</li><li>'7669_R_P_1_TBR1'</li><li>'7669_R_sec3_Cau_DLX1'</li><li>'7669_R_sec3_Put_DLX1'</li><li>'7669_R_T_1_COUPTF2'</li><li>'7669_R_T_1_DLX1'</li><li>'7669_R_T_1_TBR1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '7669\\_R\\_Ant\\_Cau\\_DLX1'\n",
       "\\item '7669\\_R\\_Ant\\_Put\\_DLX1'\n",
       "\\item '7669\\_R\\_F\\_1\\_COUPTF2'\n",
       "\\item '7669\\_R\\_F\\_1\\_DLX1'\n",
       "\\item '7669\\_R\\_F\\_1\\_TBR1'\n",
       "\\item '7669\\_R\\_I\\_1\\_COUPTF2'\n",
       "\\item '7669\\_R\\_I\\_1\\_DLX1'\n",
       "\\item '7669\\_R\\_I\\_1\\_TBR1'\n",
       "\\item '7669\\_R\\_O\\_1\\_COUPTF2'\n",
       "\\item '7669\\_R\\_O\\_1\\_DLX1'\n",
       "\\item '7669\\_R\\_O\\_1\\_TBR1'\n",
       "\\item '7669\\_R\\_P\\_1\\_COUPTF2'\n",
       "\\item '7669\\_R\\_P\\_1\\_DLX1'\n",
       "\\item '7669\\_R\\_P\\_1\\_TBR1'\n",
       "\\item '7669\\_R\\_sec3\\_Cau\\_DLX1'\n",
       "\\item '7669\\_R\\_sec3\\_Put\\_DLX1'\n",
       "\\item '7669\\_R\\_T\\_1\\_COUPTF2'\n",
       "\\item '7669\\_R\\_T\\_1\\_DLX1'\n",
       "\\item '7669\\_R\\_T\\_1\\_TBR1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '7669_R_Ant_Cau_DLX1'\n",
       "2. '7669_R_Ant_Put_DLX1'\n",
       "3. '7669_R_F_1_COUPTF2'\n",
       "4. '7669_R_F_1_DLX1'\n",
       "5. '7669_R_F_1_TBR1'\n",
       "6. '7669_R_I_1_COUPTF2'\n",
       "7. '7669_R_I_1_DLX1'\n",
       "8. '7669_R_I_1_TBR1'\n",
       "9. '7669_R_O_1_COUPTF2'\n",
       "10. '7669_R_O_1_DLX1'\n",
       "11. '7669_R_O_1_TBR1'\n",
       "12. '7669_R_P_1_COUPTF2'\n",
       "13. '7669_R_P_1_DLX1'\n",
       "14. '7669_R_P_1_TBR1'\n",
       "15. '7669_R_sec3_Cau_DLX1'\n",
       "16. '7669_R_sec3_Put_DLX1'\n",
       "17. '7669_R_T_1_COUPTF2'\n",
       "18. '7669_R_T_1_DLX1'\n",
       "19. '7669_R_T_1_TBR1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"7669_R_Ant_Cau_DLX1\"  \"7669_R_Ant_Put_DLX1\"  \"7669_R_F_1_COUPTF2\"  \n",
       " [4] \"7669_R_F_1_DLX1\"      \"7669_R_F_1_TBR1\"      \"7669_R_I_1_COUPTF2\"  \n",
       " [7] \"7669_R_I_1_DLX1\"      \"7669_R_I_1_TBR1\"      \"7669_R_O_1_COUPTF2\"  \n",
       "[10] \"7669_R_O_1_DLX1\"      \"7669_R_O_1_TBR1\"      \"7669_R_P_1_COUPTF2\"  \n",
       "[13] \"7669_R_P_1_DLX1\"      \"7669_R_P_1_TBR1\"      \"7669_R_sec3_Cau_DLX1\"\n",
       "[16] \"7669_R_sec3_Put_DLX1\" \"7669_R_T_1_COUPTF2\"   \"7669_R_T_1_DLX1\"     \n",
       "[19] \"7669_R_T_1_TBR1\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sort(unique(filtered2$ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "33495679-22f8-426c-a902-b53a1411eb01",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(variant_list,\"./780_variant_annotation_COUPTF2_origin2.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb8d2688-8324-4173-ab6b-e3a259f9e62c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
