{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d883195b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘gplots’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    lowess\n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'ID'</li><li>'CHROM'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'CHROM.POS.REF.ALT'</li><li>'ANNO'</li><li>'GENE'</li><li>'GNOMAD_FREQ'</li><li>'REPEAT_MASKER'</li><li>'SEGDUP'</li><li>'HOMOPOLYMER'</li><li>'REF_SEQ'</li><li>'DINUCLEOTIDE'</li><li>'NEAR_INDEL'</li><li>'UCSC_RPMSK'</li><li>'REF_COUNT'</li><li>'ALT_COUNT'</li><li>'MAF'</li><li>'LOWER_CI'</li><li>'UPPER_CI'</li><li>'CI_IS_GREATER'</li><li>'NORMAL_REF_COUNT'</li><li>'NORMAL_ALT_COUNT'</li><li>'NORMAL_MAF'</li><li>'NORMAL_LOWER_CI'</li><li>'NORMAL_UPPER_CI'</li><li>'NORMAL_CI_IS_GREATER'</li><li>'Mosaic'</li><li>'Mosaic_shared_btw_WGS_Amp'</li><li>'Hemisphere'</li><li>'Region'</li><li>'Punch'</li><li>'Cell_Type'</li><li>'Location'</li><li>'AntPost'</li><li>'DorVen'</li><li>'Organ'</li><li>'Note'</li><li>'Bulk_main'</li><li>'Sort_main'</li><li>'CTX_BG_HIP_relationship'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ID'\n",
       "\\item 'CHROM'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'CHROM.POS.REF.ALT'\n",
       "\\item 'ANNO'\n",
       "\\item 'GENE'\n",
       "\\item 'GNOMAD\\_FREQ'\n",
       "\\item 'REPEAT\\_MASKER'\n",
       "\\item 'SEGDUP'\n",
       "\\item 'HOMOPOLYMER'\n",
       "\\item 'REF\\_SEQ'\n",
       "\\item 'DINUCLEOTIDE'\n",
       "\\item 'NEAR\\_INDEL'\n",
       "\\item 'UCSC\\_RPMSK'\n",
       "\\item 'REF\\_COUNT'\n",
       "\\item 'ALT\\_COUNT'\n",
       "\\item 'MAF'\n",
       "\\item 'LOWER\\_CI'\n",
       "\\item 'UPPER\\_CI'\n",
       "\\item 'CI\\_IS\\_GREATER'\n",
       "\\item 'NORMAL\\_REF\\_COUNT'\n",
       "\\item 'NORMAL\\_ALT\\_COUNT'\n",
       "\\item 'NORMAL\\_MAF'\n",
       "\\item 'NORMAL\\_LOWER\\_CI'\n",
       "\\item 'NORMAL\\_UPPER\\_CI'\n",
       "\\item 'NORMAL\\_CI\\_IS\\_GREATER'\n",
       "\\item 'Mosaic'\n",
       "\\item 'Mosaic\\_shared\\_btw\\_WGS\\_Amp'\n",
       "\\item 'Hemisphere'\n",
       "\\item 'Region'\n",
       "\\item 'Punch'\n",
       "\\item 'Cell\\_Type'\n",
       "\\item 'Location'\n",
       "\\item 'AntPost'\n",
       "\\item 'DorVen'\n",
       "\\item 'Organ'\n",
       "\\item 'Note'\n",
       "\\item 'Bulk\\_main'\n",
       "\\item 'Sort\\_main'\n",
       "\\item 'CTX\\_BG\\_HIP\\_relationship'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ID'\n",
       "2. 'CHROM'\n",
       "3. 'POS'\n",
       "4. 'REF'\n",
       "5. 'ALT'\n",
       "6. 'CHROM.POS.REF.ALT'\n",
       "7. 'ANNO'\n",
       "8. 'GENE'\n",
       "9. 'GNOMAD_FREQ'\n",
       "10. 'REPEAT_MASKER'\n",
       "11. 'SEGDUP'\n",
       "12. 'HOMOPOLYMER'\n",
       "13. 'REF_SEQ'\n",
       "14. 'DINUCLEOTIDE'\n",
       "15. 'NEAR_INDEL'\n",
       "16. 'UCSC_RPMSK'\n",
       "17. 'REF_COUNT'\n",
       "18. 'ALT_COUNT'\n",
       "19. 'MAF'\n",
       "20. 'LOWER_CI'\n",
       "21. 'UPPER_CI'\n",
       "22. 'CI_IS_GREATER'\n",
       "23. 'NORMAL_REF_COUNT'\n",
       "24. 'NORMAL_ALT_COUNT'\n",
       "25. 'NORMAL_MAF'\n",
       "26. 'NORMAL_LOWER_CI'\n",
       "27. 'NORMAL_UPPER_CI'\n",
       "28. 'NORMAL_CI_IS_GREATER'\n",
       "29. 'Mosaic'\n",
       "30. 'Mosaic_shared_btw_WGS_Amp'\n",
       "31. 'Hemisphere'\n",
       "32. 'Region'\n",
       "33. 'Punch'\n",
       "34. 'Cell_Type'\n",
       "35. 'Location'\n",
       "36. 'AntPost'\n",
       "37. 'DorVen'\n",
       "38. 'Organ'\n",
       "39. 'Note'\n",
       "40. 'Bulk_main'\n",
       "41. 'Sort_main'\n",
       "42. 'CTX_BG_HIP_relationship'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ID\"                        \"CHROM\"                    \n",
       " [3] \"POS\"                       \"REF\"                      \n",
       " [5] \"ALT\"                       \"CHROM.POS.REF.ALT\"        \n",
       " [7] \"ANNO\"                      \"GENE\"                     \n",
       " [9] \"GNOMAD_FREQ\"               \"REPEAT_MASKER\"            \n",
       "[11] \"SEGDUP\"                    \"HOMOPOLYMER\"              \n",
       "[13] \"REF_SEQ\"                   \"DINUCLEOTIDE\"             \n",
       "[15] \"NEAR_INDEL\"                \"UCSC_RPMSK\"               \n",
       "[17] \"REF_COUNT\"                 \"ALT_COUNT\"                \n",
       "[19] \"MAF\"                       \"LOWER_CI\"                 \n",
       "[21] \"UPPER_CI\"                  \"CI_IS_GREATER\"            \n",
       "[23] \"NORMAL_REF_COUNT\"          \"NORMAL_ALT_COUNT\"         \n",
       "[25] \"NORMAL_MAF\"                \"NORMAL_LOWER_CI\"          \n",
       "[27] \"NORMAL_UPPER_CI\"           \"NORMAL_CI_IS_GREATER\"     \n",
       "[29] \"Mosaic\"                    \"Mosaic_shared_btw_WGS_Amp\"\n",
       "[31] \"Hemisphere\"                \"Region\"                   \n",
       "[33] \"Punch\"                     \"Cell_Type\"                \n",
       "[35] \"Location\"                  \"AntPost\"                  \n",
       "[37] \"DorVen\"                    \"Organ\"                    \n",
       "[39] \"Note\"                      \"Bulk_main\"                \n",
       "[41] \"Sort_main\"                 \"CTX_BG_HIP_relationship\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'L'</li><li>NA</li><li>'R'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'L'\n",
       "\\item NA\n",
       "\\item 'R'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'L'\n",
       "2. NA\n",
       "3. 'R'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"L\" NA  \"R\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'T'</li><li>'Cb'</li><li>NA</li><li>'Cau'</li><li>'Amg'</li><li>'GP'</li><li>'Put'</li><li>'Hip'</li><li>'Thal'</li><li>'CA1'</li><li>'CA3'</li><li>'DG'</li><li>'Sub'</li><li>'Ins'</li><li>'F'</li><li>'O'</li><li>'P'</li><li>'PF'</li><li>'Clo'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'T'\n",
       "\\item 'Cb'\n",
       "\\item NA\n",
       "\\item 'Cau'\n",
       "\\item 'Amg'\n",
       "\\item 'GP'\n",
       "\\item 'Put'\n",
       "\\item 'Hip'\n",
       "\\item 'Thal'\n",
       "\\item 'CA1'\n",
       "\\item 'CA3'\n",
       "\\item 'DG'\n",
       "\\item 'Sub'\n",
       "\\item 'Ins'\n",
       "\\item 'F'\n",
       "\\item 'O'\n",
       "\\item 'P'\n",
       "\\item 'PF'\n",
       "\\item 'Clo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'T'\n",
       "2. 'Cb'\n",
       "3. NA\n",
       "4. 'Cau'\n",
       "5. 'Amg'\n",
       "6. 'GP'\n",
       "7. 'Put'\n",
       "8. 'Hip'\n",
       "9. 'Thal'\n",
       "10. 'CA1'\n",
       "11. 'CA3'\n",
       "12. 'DG'\n",
       "13. 'Sub'\n",
       "14. 'Ins'\n",
       "15. 'F'\n",
       "16. 'O'\n",
       "17. 'P'\n",
       "18. 'PF'\n",
       "19. 'Clo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"T\"    \"Cb\"   NA     \"Cau\"  \"Amg\"  \"GP\"   \"Put\"  \"Hip\"  \"Thal\" \"CA1\" \n",
       "[11] \"CA3\"  \"DG\"   \"Sub\"  \"Ins\"  \"F\"    \"O\"    \"P\"    \"PF\"   \"Clo\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>&lt;NA&gt;</li><li>141</li><li>2</li><li>3</li><li>1</li><li>4</li><li>5</li><li>6</li><li>10</li><li>7</li><li>8</li><li>9</li><li>144</li><li>145</li><li>34</li><li>35</li><li>36</li><li>37</li><li>72</li><li>73</li><li>74</li><li>75</li><li>29</li><li>57</li><li>43</li><li>15</li><li>99</li><li>127</li><li>113</li><li>85</li><li>71</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item 141\n",
       "\\item 2\n",
       "\\item 3\n",
       "\\item 1\n",
       "\\item 4\n",
       "\\item 5\n",
       "\\item 6\n",
       "\\item 10\n",
       "\\item 7\n",
       "\\item 8\n",
       "\\item 9\n",
       "\\item 144\n",
       "\\item 145\n",
       "\\item 34\n",
       "\\item 35\n",
       "\\item 36\n",
       "\\item 37\n",
       "\\item 72\n",
       "\\item 73\n",
       "\\item 74\n",
       "\\item 75\n",
       "\\item 29\n",
       "\\item 57\n",
       "\\item 43\n",
       "\\item 15\n",
       "\\item 99\n",
       "\\item 127\n",
       "\\item 113\n",
       "\\item 85\n",
       "\\item 71\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. 141\n",
       "3. 2\n",
       "4. 3\n",
       "5. 1\n",
       "6. 4\n",
       "7. 5\n",
       "8. 6\n",
       "9. 10\n",
       "10. 7\n",
       "11. 8\n",
       "12. 9\n",
       "13. 144\n",
       "14. 145\n",
       "15. 34\n",
       "16. 35\n",
       "17. 36\n",
       "18. 37\n",
       "19. 72\n",
       "20. 73\n",
       "21. 74\n",
       "22. 75\n",
       "23. 29\n",
       "24. 57\n",
       "25. 43\n",
       "26. 15\n",
       "27. 99\n",
       "28. 127\n",
       "29. 113\n",
       "30. 85\n",
       "31. 71\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]  NA 141   2   3   1   4   5   6  10   7   8   9 144 145  34  35  36  37  72\n",
       "[20]  73  74  75  29  57  43  15  99 127 113  85  71"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Bulk'</li><li>'Cardiomyocyte'</li><li>'COUPTF2'</li><li>'DLX1'</li><li>'DARPP32M-NeuNP'</li><li>'DARPP32P-NeuNP'</li><li>'TBR1'</li><li>'NeuN'</li><li>'LHX2'</li><li>'Olig2'</li><li>'PU1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Bulk'\n",
       "\\item 'Cardiomyocyte'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'DLX1'\n",
       "\\item 'DARPP32M-NeuNP'\n",
       "\\item 'DARPP32P-NeuNP'\n",
       "\\item 'TBR1'\n",
       "\\item 'NeuN'\n",
       "\\item 'LHX2'\n",
       "\\item 'Olig2'\n",
       "\\item 'PU1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Bulk'\n",
       "2. 'Cardiomyocyte'\n",
       "3. 'COUPTF2'\n",
       "4. 'DLX1'\n",
       "5. 'DARPP32M-NeuNP'\n",
       "6. 'DARPP32P-NeuNP'\n",
       "7. 'TBR1'\n",
       "8. 'NeuN'\n",
       "9. 'LHX2'\n",
       "10. 'Olig2'\n",
       "11. 'PU1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Bulk\"           \"Cardiomyocyte\"  \"COUPTF2\"        \"DLX1\"          \n",
       " [5] \"DARPP32M-NeuNP\" \"DARPP32P-NeuNP\" \"TBR1\"           \"NeuN\"          \n",
       " [9] \"LHX2\"           \"Olig2\"          \"PU1\"           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>NA</li><li>'C'</li><li>'P'</li><li>'V'</li><li>'A'</li><li>'D'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item NA\n",
       "\\item 'C'\n",
       "\\item 'P'\n",
       "\\item 'V'\n",
       "\\item 'A'\n",
       "\\item 'D'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. NA\n",
       "2. 'C'\n",
       "3. 'P'\n",
       "4. 'V'\n",
       "5. 'A'\n",
       "6. 'D'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] NA  \"C\" \"P\" \"V\" \"A\" \"D\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Post'</li><li>NA</li><li>'Ant'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Post'\n",
       "\\item NA\n",
       "\\item 'Ant'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Post'\n",
       "2. NA\n",
       "3. 'Ant'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Post\" NA     \"Ant\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Ven'</li><li>NA</li><li>'Dor'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Ven'\n",
       "\\item NA\n",
       "\\item 'Dor'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Ven'\n",
       "2. NA\n",
       "3. 'Dor'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Ven\" NA    \"Dor\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CTX'</li><li>'CB'</li><li>'Heart'</li><li>'Kidney'</li><li>'BG'</li><li>'AMG'</li><li>'HIP'</li><li>'THAL'</li><li>'SUB'</li><li>'Liver'</li><li>'CLO'</li><li>'JGG'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CTX'\n",
       "\\item 'CB'\n",
       "\\item 'Heart'\n",
       "\\item 'Kidney'\n",
       "\\item 'BG'\n",
       "\\item 'AMG'\n",
       "\\item 'HIP'\n",
       "\\item 'THAL'\n",
       "\\item 'SUB'\n",
       "\\item 'Liver'\n",
       "\\item 'CLO'\n",
       "\\item 'JGG'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CTX'\n",
       "2. 'CB'\n",
       "3. 'Heart'\n",
       "4. 'Kidney'\n",
       "5. 'BG'\n",
       "6. 'AMG'\n",
       "7. 'HIP'\n",
       "8. 'THAL'\n",
       "9. 'SUB'\n",
       "10. 'Liver'\n",
       "11. 'CLO'\n",
       "12. 'JGG'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"CTX\"    \"CB\"     \"Heart\"  \"Kidney\" \"BG\"     \"AMG\"    \"HIP\"    \"THAL\"  \n",
       " [9] \"SUB\"    \"Liver\"  \"CLO\"    \"JGG\"   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'LARGE'</li><li>'SMALL'</li><li>NA</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'LARGE'\n",
       "\\item 'SMALL'\n",
       "\\item NA\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'LARGE'\n",
       "2. 'SMALL'\n",
       "3. NA\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"LARGE\" \"SMALL\" NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"A total of variant calls:  335124\"\n",
      "[1] \"The number of variants included in ampliseq:  1044\"\n",
      "[1] \"The number of validated MVs:  287\"\n",
      "[1] \"The number of validated MV calls:  1409\"\n",
      "[1] \"The number of validated MVs (in the variant_list variable):  287\"\n",
      "[1] \"The number of validated MVs and detected in this 2023 Ampliseq:  124\"\n",
      "[1] \"The number of shared validated MVs :  114\"\n"
     ]
    }
   ],
   "source": [
    "library(ggplot2)\n",
    "library(dplyr)\n",
    "library(gplots)\n",
    "library(reshape2)\n",
    "library(umap)\n",
    "library(ggExtra) \n",
    "library(pals)\n",
    "library(RColorBrewer)\n",
    "\n",
    "filtered <- read.table(\"../7614_final_summary_anno3.txt\",header=T,sep=\"\\t\")\n",
    "\n",
    "names(filtered)\n",
    "\n",
    "unique(filtered$Hemisphere)\n",
    "unique(filtered$Region)\n",
    "unique(filtered$Punch)\n",
    "unique(filtered$Cell_Type)\n",
    "unique(filtered$Location)\n",
    "unique(filtered$AntPost)\n",
    "unique(filtered$DorVen)\n",
    "unique(filtered$Organ)\n",
    "unique(filtered$Note)\n",
    "\n",
    "print(paste(\"A total of variant calls: \",nrow(filtered)))\n",
    "print(paste(\"The number of variants included in ampliseq: \", length(unique(filtered$CHROM.POS.REF.ALT))))\n",
    "\n",
    "filtered2 <- subset(filtered, ID %in% c(\n",
    "'7614-L-2-Cau-DLX1','7614-L-3-Cau-DLX1','7614-L-4-Put-DLX1','7614-L-5-Cau-DLX1','7614-L-5-Put-DLX1','7614-L-7-Cau-DLX1','7614-L-7-Put-DLX1',\n",
    "                                               '7614-L-PF-TBR1', '7614-L-F-TBR1','7614-L-9-Ins-TBR1','7614-L-T-TBR1','7614-L-O-TBR1',\n",
    "                                               '7614-L-PF-COUPTF2','7614-L-F-COUPTF2','7614-L-9-Ins-COUPTF2','7614-L-T-COUPTF2','7614-L-O-COUPTF2',\n",
    "                                               '7614-L-PF-DLX1','7614-L-F-34-DLX1','7614-L-9-Ins-DLX1','7614-L-T-DLX1','7614-L-O-DLX1'\n",
    ")) # restrict annotation of variants based on the only selected sorted sample information\n",
    "filtered2$Punch<-as.character(filtered2$Punch)\n",
    "filtered2$Note<-as.character(filtered2$Note)\n",
    "\n",
    "mosaic_filtered <- filtered2[filtered2$Mosaic_shared_btw_WGS_Amp==1,]\n",
    "mosaic_filtered_cutoff <- subset(mosaic_filtered, LOWER_CI>0.001677998 & UPPER_CI<0.3923302 & NORMAL_LOWER_CI<0.001677998)\n",
    "\n",
    "print(paste(\"The number of validated MVs: \", length(unique(mosaic_filtered$CHROM.POS.REF.ALT))))\n",
    "print(paste(\"The number of validated MV calls: \", length(mosaic_filtered_cutoff$CHROM.POS.REF.ALT)))\n",
    "\n",
    "# statistics\n",
    "\n",
    "variant_list<-unique(mosaic_filtered$CHROM.POS.REF.ALT)\n",
    "variant_list<-as.data.frame(variant_list, row.names = variant_list) # assign rownames as CHROM.POS.REF.ALT\n",
    "print(paste(\"The number of validated MVs (in the variant_list variable): \", nrow(variant_list)))\n",
    "for (i in 1:nrow(variant_list)){\n",
    "    xx <- filtered2[filtered2$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "    xx <- subset(xx, LOWER_CI>0.001677998 & UPPER_CI<0.3923302 & NORMAL_LOWER_CI<0.001677998 )\n",
    "    n_row <- nrow(xx)\n",
    "    variant_list$sample_number_presented[i] <- n_row\n",
    "    }\n",
    "print(paste(\"The number of validated MVs and detected in this 2023 Ampliseq: \", nrow(variant_list[variant_list$sample_number_presented>0,])))\n",
    "# Non-private Mosaic variants\n",
    "print(paste(\"The number of shared validated MVs : \",nrow(variant_list[variant_list$sample_number_presented>1,])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2987ac6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for (j in c('CTX','CB','Heart','Kidney','BG','THAL','Liver')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Organ)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('L','R')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Hemisphere)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('PF','F','P','O','T','Ins')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Region)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('Cau','Put','GP')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Region)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}\n",
    "\n",
    "for (j in c('DLX1','TBR1','COUPTF2','Bulk')){\n",
    "    for (i in 1:nrow(variant_list)){\n",
    "        xx <- mosaic_filtered_cutoff[mosaic_filtered_cutoff$CHROM.POS.REF.ALT == variant_list$variant_list[i],]\n",
    "        if(j %in% unique(xx$Cell_Type)){variant_list[i,j] <- 1}\n",
    "        else{variant_list[i,j] <- 0}\n",
    "        }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b8b87b2e-69d0-418d-86f7-c1c6f507aa3b",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "Brain_only_variants<-subset(variant_list, (Liver==0 & Heart==0 & Kidney==0) & (CTX==1 | CB==1 | BG==1 |  THAL==1))\n",
    "variant_list$Brain_only[variant_list$variant_list %in% rownames(Brain_only_variants)]<-1\n",
    "variant_list$Brain_only[!variant_list$variant_list %in% rownames(Brain_only_variants)]<-0\n",
    "\n",
    "CTX_only_variants<-subset(variant_list, (Liver==0 & Heart==0 & Kidney==0 & CB==0 & BG==0 & THAL==0) & (CTX==1))\n",
    "variant_list$CTX_only[variant_list$variant_list %in% rownames(CTX_only_variants)]<-1\n",
    "variant_list$CTX_only[!variant_list$variant_list %in% rownames(CTX_only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (Liver==0 & Heart==0 & Kidney==0 & CB==0 & THAL==0) & (CTX==1 | BG==1))\n",
    "variant_list$CTX_BG_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$CTX_BG_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (Liver==0 & Heart==0 & Kidney==0 & CB==0 & THAL==0 & CTX==0) & (BG==1))\n",
    "variant_list$BG_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$BG_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (Liver==0 & Heart==0 & Kidney==0 & CB==0 & BG==0 & CTX==0) & (THAL==1))\n",
    "variant_list$THAL_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$THAL_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "#--------------------------------hemisphere only----------------------------------------\n",
    "only_variants<-subset(variant_list, (L==1) & (R==0) & (Brain_only==1))\n",
    "variant_list$Left_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$Left_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (R==1) & (L==0) & (Brain_only==1))\n",
    "variant_list$Right_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$Right_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "#--------------------------------lobe only----------------------------------------\n",
    "only_variants<-subset(variant_list, (PF==1) & (F==0) & (P==0) & (O==0) & (Ins==0) & (Left_only==1) & (Brain_only==1))\n",
    "variant_list$L_BR_PF_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$L_BR_PF_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==1) & (P==0) & (O==0) & (Ins==0) & (Left_only==1) & (Brain_only==1))\n",
    "variant_list$L_BR_F_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$L_BR_F_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==1) & (O==0) & (Ins==0) & (Left_only==1) & (Brain_only==1))\n",
    "variant_list$L_BR_P_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$L_BR_P_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==0) & (O==1) & (Ins==0) & (Left_only==1) & (Brain_only==1))\n",
    "variant_list$L_BR_O_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$L_BR_O_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==0) & (O==0) & (Ins==1) & (Left_only==1) & (Brain_only==1))\n",
    "variant_list$L_BR_Ins_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$L_BR_Ins_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==1) & (F==0) & (P==0) & (O==0) & (Ins==0) & (Right_only==1) & (Brain_only==1))\n",
    "variant_list$R_BR_PF_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$R_BR_PF_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==1) & (P==0) & (O==0) & (Ins==0) & (Right_only==1) & (Brain_only==1))\n",
    "variant_list$R_BR_F_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$R_BR_F_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==1) & (O==0) & (Ins==0) & (Right_only==1) & (Brain_only==1))\n",
    "variant_list$R_BR_P_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$R_BR_P_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==0) & (O==1) & (Ins==0) & (Right_only==1) & (Brain_only==1))\n",
    "variant_list$R_BR_O_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$R_BR_O_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (PF==0) & (F==0) & (P==0) & (O==0) & (Ins==1) & (Right_only==1) & (Brain_only==1))\n",
    "variant_list$R_BR_Ins_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$R_BR_Ins_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "#----------------------------------------------------------------------------------------------------------------------\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==0) & (DLX1==1 | COUPTF2==1) & (Brain_only==1))\n",
    "variant_list$IN_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$IN_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==0) & (COUPTF2==1) & (Brain_only==1))\n",
    "variant_list$COUP_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$COUP_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==0 & COUPTF2==0) & (TBR1==1) & (Brain_only==1))\n",
    "variant_list$TBR_only[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_only[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR1==1) & (Brain_only==1))\n",
    "variant_list$TBR_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==1) & (Brain_only==1))\n",
    "variant_list$DLX_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (COUPTF2==1) & (Brain_only==1))\n",
    "variant_list$COUP_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$COUP_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, ('DARPP32P-NeuNP'==1) & (Brain_only==1))\n",
    "variant_list$MSN_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$MSN_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, ('DARPP32M-NeuNP'==1) & (Brain_only==1))\n",
    "variant_list$nonMSN_BR[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$nonMSN_BR[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cfd2bab1-07ea-4592-a132-3fd4b89c8fe4",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'variant_list'</li><li>'sample_number_presented'</li><li>'CTX'</li><li>'CB'</li><li>'Heart'</li><li>'Kidney'</li><li>'BG'</li><li>'THAL'</li><li>'Liver'</li><li>'L'</li><li>'R'</li><li>'PF'</li><li>'F'</li><li>'P'</li><li>'O'</li><li>'T'</li><li>'Ins'</li><li>'Cau'</li><li>'Put'</li><li>'GP'</li><li>'DLX1'</li><li>'TBR1'</li><li>'COUPTF2'</li><li>'Bulk'</li><li>'Brain_only'</li><li>'CTX_only'</li><li>'CTX_BG_only'</li><li>'BG_only'</li><li>'THAL_only'</li><li>'Left_only'</li><li>'Right_only'</li><li>'L_BR_PF_only'</li><li>'L_BR_F_only'</li><li>'L_BR_P_only'</li><li>'L_BR_O_only'</li><li>'L_BR_Ins_only'</li><li>'R_BR_PF_only'</li><li>'R_BR_F_only'</li><li>'R_BR_P_only'</li><li>'R_BR_O_only'</li><li>'R_BR_Ins_only'</li><li>'IN_only'</li><li>'COUP_only'</li><li>'TBR_only'</li><li>'TBR_BR'</li><li>'DLX_BR'</li><li>'COUP_BR'</li><li>'MSN_BR'</li><li>'nonMSN_BR'</li><li>'TBR_BR_CTX'</li><li>'DLX_BR_BG'</li><li>'COUP_BR_CTX'</li><li>'TBR_BR_CTXnDLX_BR_BG'</li><li>'TBR_BR_CTXnCOUP_BR_CTX'</li><li>'DLX_BR_BGnCOUP_BR_CTX'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'variant\\_list'\n",
       "\\item 'sample\\_number\\_presented'\n",
       "\\item 'CTX'\n",
       "\\item 'CB'\n",
       "\\item 'Heart'\n",
       "\\item 'Kidney'\n",
       "\\item 'BG'\n",
       "\\item 'THAL'\n",
       "\\item 'Liver'\n",
       "\\item 'L'\n",
       "\\item 'R'\n",
       "\\item 'PF'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'O'\n",
       "\\item 'T'\n",
       "\\item 'Ins'\n",
       "\\item 'Cau'\n",
       "\\item 'Put'\n",
       "\\item 'GP'\n",
       "\\item 'DLX1'\n",
       "\\item 'TBR1'\n",
       "\\item 'COUPTF2'\n",
       "\\item 'Bulk'\n",
       "\\item 'Brain\\_only'\n",
       "\\item 'CTX\\_only'\n",
       "\\item 'CTX\\_BG\\_only'\n",
       "\\item 'BG\\_only'\n",
       "\\item 'THAL\\_only'\n",
       "\\item 'Left\\_only'\n",
       "\\item 'Right\\_only'\n",
       "\\item 'L\\_BR\\_PF\\_only'\n",
       "\\item 'L\\_BR\\_F\\_only'\n",
       "\\item 'L\\_BR\\_P\\_only'\n",
       "\\item 'L\\_BR\\_O\\_only'\n",
       "\\item 'L\\_BR\\_Ins\\_only'\n",
       "\\item 'R\\_BR\\_PF\\_only'\n",
       "\\item 'R\\_BR\\_F\\_only'\n",
       "\\item 'R\\_BR\\_P\\_only'\n",
       "\\item 'R\\_BR\\_O\\_only'\n",
       "\\item 'R\\_BR\\_Ins\\_only'\n",
       "\\item 'IN\\_only'\n",
       "\\item 'COUP\\_only'\n",
       "\\item 'TBR\\_only'\n",
       "\\item 'TBR\\_BR'\n",
       "\\item 'DLX\\_BR'\n",
       "\\item 'COUP\\_BR'\n",
       "\\item 'MSN\\_BR'\n",
       "\\item 'nonMSN\\_BR'\n",
       "\\item 'TBR\\_BR\\_CTX'\n",
       "\\item 'DLX\\_BR\\_BG'\n",
       "\\item 'COUP\\_BR\\_CTX'\n",
       "\\item 'TBR\\_BR\\_CTXnDLX\\_BR\\_BG'\n",
       "\\item 'TBR\\_BR\\_CTXnCOUP\\_BR\\_CTX'\n",
       "\\item 'DLX\\_BR\\_BGnCOUP\\_BR\\_CTX'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'variant_list'\n",
       "2. 'sample_number_presented'\n",
       "3. 'CTX'\n",
       "4. 'CB'\n",
       "5. 'Heart'\n",
       "6. 'Kidney'\n",
       "7. 'BG'\n",
       "8. 'THAL'\n",
       "9. 'Liver'\n",
       "10. 'L'\n",
       "11. 'R'\n",
       "12. 'PF'\n",
       "13. 'F'\n",
       "14. 'P'\n",
       "15. 'O'\n",
       "16. 'T'\n",
       "17. 'Ins'\n",
       "18. 'Cau'\n",
       "19. 'Put'\n",
       "20. 'GP'\n",
       "21. 'DLX1'\n",
       "22. 'TBR1'\n",
       "23. 'COUPTF2'\n",
       "24. 'Bulk'\n",
       "25. 'Brain_only'\n",
       "26. 'CTX_only'\n",
       "27. 'CTX_BG_only'\n",
       "28. 'BG_only'\n",
       "29. 'THAL_only'\n",
       "30. 'Left_only'\n",
       "31. 'Right_only'\n",
       "32. 'L_BR_PF_only'\n",
       "33. 'L_BR_F_only'\n",
       "34. 'L_BR_P_only'\n",
       "35. 'L_BR_O_only'\n",
       "36. 'L_BR_Ins_only'\n",
       "37. 'R_BR_PF_only'\n",
       "38. 'R_BR_F_only'\n",
       "39. 'R_BR_P_only'\n",
       "40. 'R_BR_O_only'\n",
       "41. 'R_BR_Ins_only'\n",
       "42. 'IN_only'\n",
       "43. 'COUP_only'\n",
       "44. 'TBR_only'\n",
       "45. 'TBR_BR'\n",
       "46. 'DLX_BR'\n",
       "47. 'COUP_BR'\n",
       "48. 'MSN_BR'\n",
       "49. 'nonMSN_BR'\n",
       "50. 'TBR_BR_CTX'\n",
       "51. 'DLX_BR_BG'\n",
       "52. 'COUP_BR_CTX'\n",
       "53. 'TBR_BR_CTXnDLX_BR_BG'\n",
       "54. 'TBR_BR_CTXnCOUP_BR_CTX'\n",
       "55. 'DLX_BR_BGnCOUP_BR_CTX'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"variant_list\"            \"sample_number_presented\"\n",
       " [3] \"CTX\"                     \"CB\"                     \n",
       " [5] \"Heart\"                   \"Kidney\"                 \n",
       " [7] \"BG\"                      \"THAL\"                   \n",
       " [9] \"Liver\"                   \"L\"                      \n",
       "[11] \"R\"                       \"PF\"                     \n",
       "[13] \"F\"                       \"P\"                      \n",
       "[15] \"O\"                       \"T\"                      \n",
       "[17] \"Ins\"                     \"Cau\"                    \n",
       "[19] \"Put\"                     \"GP\"                     \n",
       "[21] \"DLX1\"                    \"TBR1\"                   \n",
       "[23] \"COUPTF2\"                 \"Bulk\"                   \n",
       "[25] \"Brain_only\"              \"CTX_only\"               \n",
       "[27] \"CTX_BG_only\"             \"BG_only\"                \n",
       "[29] \"THAL_only\"               \"Left_only\"              \n",
       "[31] \"Right_only\"              \"L_BR_PF_only\"           \n",
       "[33] \"L_BR_F_only\"             \"L_BR_P_only\"            \n",
       "[35] \"L_BR_O_only\"             \"L_BR_Ins_only\"          \n",
       "[37] \"R_BR_PF_only\"            \"R_BR_F_only\"            \n",
       "[39] \"R_BR_P_only\"             \"R_BR_O_only\"            \n",
       "[41] \"R_BR_Ins_only\"           \"IN_only\"                \n",
       "[43] \"COUP_only\"               \"TBR_only\"               \n",
       "[45] \"TBR_BR\"                  \"DLX_BR\"                 \n",
       "[47] \"COUP_BR\"                 \"MSN_BR\"                 \n",
       "[49] \"nonMSN_BR\"               \"TBR_BR_CTX\"             \n",
       "[51] \"DLX_BR_BG\"               \"COUP_BR_CTX\"            \n",
       "[53] \"TBR_BR_CTXnDLX_BR_BG\"    \"TBR_BR_CTXnCOUP_BR_CTX\" \n",
       "[55] \"DLX_BR_BGnCOUP_BR_CTX\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#---------------------------BG-DLX1 vs CTX-TBR1 vs CTX-COUPTFII-------------------\n",
    "only_variants<-subset(variant_list, (TBR1==1) )\n",
    "variant_list$TBR_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX1==1) )\n",
    "variant_list$DLX_BR_BG[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR_BG[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (COUPTF2==1) )\n",
    "variant_list$COUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$COUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR_BR_CTX==1) & (DLX_BR_BG==1) & !(COUP_BR_CTX==1))\n",
    "variant_list$TBR_BR_CTXnDLX_BR_BG[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTXnDLX_BR_BG[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (TBR_BR_CTX==1) & (COUP_BR_CTX==1) & !(DLX_BR_BG==1))\n",
    "variant_list$TBR_BR_CTXnCOUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$TBR_BR_CTXnCOUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "only_variants<-subset(variant_list, (DLX_BR_BG==1) & (COUP_BR_CTX==1) & !(TBR_BR_CTX==1))\n",
    "variant_list$DLX_BR_BGnCOUP_BR_CTX[variant_list$variant_list %in% rownames(only_variants)]<-1\n",
    "variant_list$DLX_BR_BGnCOUP_BR_CTX[!variant_list$variant_list %in% rownames(only_variants)]<-0\n",
    "\n",
    "names(variant_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b9f597bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"TBR_BR_CTX variants:  106\"\n",
      "[1] \"DLX_BR_BG variants:  117\"\n",
      "[1] \"COUP_BR_CTX variants:  90\"\n",
      "[1] \"TBR_BR_CTXnDLX_BR_BG variants:  20\"\n",
      "[1] \"TBR_BR_CTXnCOUP_BR_CTX variants:  1\"\n",
      "[1] \"DLX_BR_BGnCOUP_BR_CTX variants:  8\"\n"
     ]
    }
   ],
   "source": [
    "print(paste(\"TBR_BR_CTX variants: \",nrow(variant_list[variant_list$TBR_BR_CTX==1,])))\n",
    "print(paste(\"DLX_BR_BG variants: \",nrow(variant_list[variant_list$DLX_BR_BG==1,])))\n",
    "print(paste(\"COUP_BR_CTX variants: \",nrow(variant_list[variant_list$COUP_BR_CTX==1,])))\n",
    "print(paste(\"TBR_BR_CTXnDLX_BR_BG variants: \",nrow(variant_list[variant_list$TBR_BR_CTXnDLX_BR_BG==1,])))\n",
    "print(paste(\"TBR_BR_CTXnCOUP_BR_CTX variants: \",nrow(variant_list[variant_list$TBR_BR_CTXnCOUP_BR_CTX==1,])))\n",
    "print(paste(\"DLX_BR_BGnCOUP_BR_CTX variants: \",nrow(variant_list[variant_list$DLX_BR_BGnCOUP_BR_CTX==1,])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9ea7e526-00a0-4527-bf5f-1aba60db23bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'7614-L-2-Cau-DLX1'</li><li>'7614-L-3-Cau-DLX1'</li><li>'7614-L-4-Put-DLX1'</li><li>'7614-L-5-Cau-DLX1'</li><li>'7614-L-5-Put-DLX1'</li><li>'7614-L-7-Cau-DLX1'</li><li>'7614-L-7-Put-DLX1'</li><li>'7614-L-9-Ins-COUPTF2'</li><li>'7614-L-9-Ins-DLX1'</li><li>'7614-L-9-Ins-TBR1'</li><li>'7614-L-F-34-DLX1'</li><li>'7614-L-F-COUPTF2'</li><li>'7614-L-F-TBR1'</li><li>'7614-L-O-COUPTF2'</li><li>'7614-L-O-DLX1'</li><li>'7614-L-O-TBR1'</li><li>'7614-L-PF-COUPTF2'</li><li>'7614-L-PF-DLX1'</li><li>'7614-L-PF-TBR1'</li><li>'7614-L-T-COUPTF2'</li><li>'7614-L-T-DLX1'</li><li>'7614-L-T-TBR1'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '7614-L-2-Cau-DLX1'\n",
       "\\item '7614-L-3-Cau-DLX1'\n",
       "\\item '7614-L-4-Put-DLX1'\n",
       "\\item '7614-L-5-Cau-DLX1'\n",
       "\\item '7614-L-5-Put-DLX1'\n",
       "\\item '7614-L-7-Cau-DLX1'\n",
       "\\item '7614-L-7-Put-DLX1'\n",
       "\\item '7614-L-9-Ins-COUPTF2'\n",
       "\\item '7614-L-9-Ins-DLX1'\n",
       "\\item '7614-L-9-Ins-TBR1'\n",
       "\\item '7614-L-F-34-DLX1'\n",
       "\\item '7614-L-F-COUPTF2'\n",
       "\\item '7614-L-F-TBR1'\n",
       "\\item '7614-L-O-COUPTF2'\n",
       "\\item '7614-L-O-DLX1'\n",
       "\\item '7614-L-O-TBR1'\n",
       "\\item '7614-L-PF-COUPTF2'\n",
       "\\item '7614-L-PF-DLX1'\n",
       "\\item '7614-L-PF-TBR1'\n",
       "\\item '7614-L-T-COUPTF2'\n",
       "\\item '7614-L-T-DLX1'\n",
       "\\item '7614-L-T-TBR1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '7614-L-2-Cau-DLX1'\n",
       "2. '7614-L-3-Cau-DLX1'\n",
       "3. '7614-L-4-Put-DLX1'\n",
       "4. '7614-L-5-Cau-DLX1'\n",
       "5. '7614-L-5-Put-DLX1'\n",
       "6. '7614-L-7-Cau-DLX1'\n",
       "7. '7614-L-7-Put-DLX1'\n",
       "8. '7614-L-9-Ins-COUPTF2'\n",
       "9. '7614-L-9-Ins-DLX1'\n",
       "10. '7614-L-9-Ins-TBR1'\n",
       "11. '7614-L-F-34-DLX1'\n",
       "12. '7614-L-F-COUPTF2'\n",
       "13. '7614-L-F-TBR1'\n",
       "14. '7614-L-O-COUPTF2'\n",
       "15. '7614-L-O-DLX1'\n",
       "16. '7614-L-O-TBR1'\n",
       "17. '7614-L-PF-COUPTF2'\n",
       "18. '7614-L-PF-DLX1'\n",
       "19. '7614-L-PF-TBR1'\n",
       "20. '7614-L-T-COUPTF2'\n",
       "21. '7614-L-T-DLX1'\n",
       "22. '7614-L-T-TBR1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"7614-L-2-Cau-DLX1\"    \"7614-L-3-Cau-DLX1\"    \"7614-L-4-Put-DLX1\"   \n",
       " [4] \"7614-L-5-Cau-DLX1\"    \"7614-L-5-Put-DLX1\"    \"7614-L-7-Cau-DLX1\"   \n",
       " [7] \"7614-L-7-Put-DLX1\"    \"7614-L-9-Ins-COUPTF2\" \"7614-L-9-Ins-DLX1\"   \n",
       "[10] \"7614-L-9-Ins-TBR1\"    \"7614-L-F-34-DLX1\"     \"7614-L-F-COUPTF2\"    \n",
       "[13] \"7614-L-F-TBR1\"        \"7614-L-O-COUPTF2\"     \"7614-L-O-DLX1\"       \n",
       "[16] \"7614-L-O-TBR1\"        \"7614-L-PF-COUPTF2\"    \"7614-L-PF-DLX1\"      \n",
       "[19] \"7614-L-PF-TBR1\"       \"7614-L-T-COUPTF2\"     \"7614-L-T-DLX1\"       \n",
       "[22] \"7614-L-T-TBR1\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sort(unique(filtered2$ID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5ed9d492-5001-4df4-8bfc-c93a78f3ba2e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "variant_list2 <- variant_list[,-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "19c2c204-8d81-4f13-ad37-13d9dd2c624d",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(variant_list2,\"./287_variant_annotation_COUPTF2_origin2.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c7787aa-c0e5-424f-915f-71d315a782f8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
